<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Ludger Goeminne, Kris Gevaert and Lieven Clement" />

<meta name="date" content="2017-03-03" />

<title>MSqRob Vignette</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">MSqRob Vignette</h1>
<h4 class="author"><em>Ludger Goeminne, Kris Gevaert and Lieven Clement</em></h4>
<h4 class="date"><em>2017-03-03</em></h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The MSqRob package is designed for the analysis of differential abundance in label-free mass spectrometry-based shotgun proteomics experiments. These experiments typically output large spectral files (e.g. “.mzML” files) that need to be searched for identifications by specialized peak detection and peptide identification software such as Andromeda (used in the MaxQuant software package) and Mascot Distiller. These algorithms output lists of putative peptides accompanied by their corresponding overall intensity estimates.</p>
<p>Often, researchers further summarize these peptide intensities to protein intensities. However, as we have shown before, this summarization-based approaches tend to perform suboptimal compared to peptide-based models, wherein the peptide intensities are directly provided as input to the statistical model used for differential quantification (Goeminne et al., 2015). Nonetheless, most of these models still suffer from overfitting issues.</p>
<p>This package implements the robust ridge approach that prevents overfitting in sparse data but also accurately fits highly abundant proteins as described in Goeminne et al. (2016).</p>
<p>This vignette will guide you through the main features of the MSqRob package. We will highlight two examples in order to help you familiarize with its main features. MSqRob is completely free and open source, but please make a reference to Goeminne et al. (2016) when using our package in any kind of publication.</p>
</div>
<div id="downloading-installing-and-loading-msqrob" class="section level2">
<h2>Downloading, installing and loading MSqRob</h2>
</div>
<div id="installing-the-package" class="section level2">
<h2>Installing the package</h2>
<p>On Windows, make sure that RTools is installed. Go to: <a href="https://cran.r-project.org/bin/windows/Rtools/" class="uri">https://cran.r-project.org/bin/windows/Rtools/</a> to download RTools. A user guide on how to install RTools on Windows can be found at: <a href="https://github.com/stan-dev/rstan/wiki/Install-Rtools-for-Windows" class="uri">https://github.com/stan-dev/rstan/wiki/Install-Rtools-for-Windows</a>. Errors in MSqRob on Windows related to unable to zip the results Excel file might be related to errors in configuring RTools.</p>
<p>First, we need to install the package devtools:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
<span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>, <span class="dt">repos =</span> <span class="st">&quot;http://cran.us.r-project.org&quot;</span>)
<span class="kw">library</span>(devtools)</code></pre></div>
<p>Then, we call this to install MSqRob from GitHub:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
devtools::<span class="kw">install_github</span>(<span class="st">&quot;statOmics/MSqRob&quot;</span>)
<span class="kw">library</span>(MSqRob)</code></pre></div>
<p>Note that the latest version of MSqRob can always be found at: <a href="https://github.com/statOmics/MSqRob" class="uri">https://github.com/statOmics/MSqRob</a>.</p>
<p>The latest version of this vignette can be found at: <a href="https://github.com/statOmics/MSqRob/blob/master/vignettes/MSqRob.Rmd" class="uri">https://github.com/statOmics/MSqRob/blob/master/vignettes/MSqRob.Rmd</a>.</p>
</div>
<div id="loading-msqrob" class="section level2">
<h2>Loading MSqRob</h2>
<p>Loading the package is done via the <code>library</code> command, as with any other package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(MSqRob)</code></pre></div>
<pre><code>## Loading required package: lme4</code></pre>
<pre><code>## Loading required package: Matrix</code></pre>
<pre><code>## Warning: package 'Matrix' was built under R version 3.3.2</code></pre>
<pre><code>## 
## Attaching package: 'lme4'</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     sigma</code></pre>
<p>Additionally, we load the packages <code>Biobase</code> and <code>MSnbase</code>, as we will use a few of their functions in this vignette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Biobase)</code></pre></div>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following object is masked from 'package:Matrix':
## 
##     which</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, cbind, colnames,
##     do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, lengths, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
##     Position, rank, rbind, Reduce, rownames, sapply, setdiff,
##     sort, table, tapply, union, unique, unsplit, which, which.max,
##     which.min</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation(&quot;Biobase&quot;)', and for packages 'citation(&quot;pkgname&quot;)'.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(MSnbase)</code></pre></div>
<pre><code>## Warning: package 'MSnbase' was built under R version 3.3.2</code></pre>
<pre><code>## Loading required package: mzR</code></pre>
<pre><code>## Warning: package 'mzR' was built under R version 3.3.2</code></pre>
<pre><code>## Loading required package: Rcpp</code></pre>
<pre><code>## Warning: package 'Rcpp' was built under R version 3.3.2</code></pre>
<pre><code>## Warning in fun(libname, pkgname): mzR has been built against a different Rcpp version (0.12.8)
## than is installed on your system (0.12.9). This might lead to errors
## when loading mzR. If you encounter such issues, please send a report,
## including the output of sessionInfo() to the Bioc support forum at 
## https://support.bioconductor.org/. For details see also
## https://github.com/sneumann/mzR/wiki/mzR-Rcpp-compiler-linker-issue.</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## Loading required package: ProtGenerics</code></pre>
<pre><code>## 
## This is MSnbase version 2.0.2 
##   Read '?MSnbase' and references therein for information
##   about the package and how to get started.</code></pre>
<pre><code>## 
## Attaching package: 'MSnbase'</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     smooth</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     trimws</code></pre>
</div>
<div id="the-msqrobdata-package" class="section level2">
<h2>The MSqRobData package</h2>
<p>R packages uploaded on CRAN have a maximum size limit of 5 MB. As proteomics datasets are typically several hunderds of GB in size and even an output file of a peptide search can be tens of MBs in size, we uploaded our example data in a separate data package called MSqRobData. If you want to run the examples in this vignette, you also need the data package. This can also be downloaded from GitHub in the following way:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
devtools::<span class="kw">install_github</span>(<span class="st">&quot;statOmics/MSqRobData&quot;</span>)
<span class="kw">library</span>(MSqRobData)</code></pre></div>
</div>
<div id="the-shiny-app" class="section level2">
<h2>The Shiny App</h2>
<p>If you want to work with the Shiny graphical user interface (MaxQuant output only), use the following command:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
shiny::<span class="kw">runApp</span>(<span class="kw">system.file</span>(<span class="st">'App-MSqRob'</span>, <span class="dt">package=</span><span class="st">'MSqRob'</span>))</code></pre></div>
</div>
<div id="francisella-tularensis-dataset-design-with-pseudoreplication" class="section level2">
<h2><em>Francisella tularensis</em> dataset (design with pseudoreplication)</h2>
<p>The bacterium <em>Francisella tularensis</em> is a facultative intracellular parasite of macrophages and must import host cell arginine in order to be able to multiply inside the host cell. Ramond et al. (2015) investigated the effect of gene deletion of a newly identified arginine transporter called ArgP. Their data is deposited in the PRIDE repository at <a href="http://www.ebi.ac.uk/pride/archive/projects/PXD001584" class="uri">http://www.ebi.ac.uk/pride/archive/projects/PXD001584</a>. We used the authors’ supplied peptides.txt file which is the result of preprocessing with MaxQuant version 1.4.1.2. As the authors used now outdated GI numbers, we modified this file to incorporate NCBI accessions and protein names (September 25, 2016).</p>
<div id="importing-data" class="section level3">
<h3>1. Importing data</h3>
<p>First, specify the location of the peptides.txt MaxQuant output file. By default, MaxQuant creates a “combined” folder in the same folder as the raw files are (or if the raw files are in different directories, in the directory of the first raw file). The peptides.txt file can then be found in “path_to_raw_files/combined/txt/peptides.txt”. Just provide the path to where the file is located on your computer. The <code>system.file</code> function is used here only to access the example tab-delimited peptides.txt file that is delivered with our package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file_peptides_txt &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/Francisella&quot;</span>, <span class="st">&quot;peptides.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MSqRobData&quot;</span>)
file_peptides_txt</code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Versions/3.3/Resources/library/MSqRobData/extdata/Francisella/peptides.txt&quot;</code></pre>
<p>Import the data using the <code>read_MaxQuant</code> function. The <code>pattern</code> argument should contain a text string that is present in all column names that contain peptide intensities but not in the other column names. This defaults to <code>&quot;Intensity\ &quot;</code>. The default setting <code>remove_pattern=TRUE</code> then removes the pattern from these column names so that the names equal the names that were entered in MaxQuant’s “Experiment” column when conducting the search. In this particular <em>Francisella</em> experiment, our column names start with a number (either “1” or “3”), which is not a valid column name in R. Therefore, an “X” is prepended to these names.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peptidesFranc &lt;-<span class="st"> </span><span class="kw">read_MaxQuant</span>(file_peptides_txt, <span class="dt">pattern=</span><span class="st">&quot;Intensity &quot;</span>, <span class="dt">remove_pattern=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><code>read_MaxQuant</code> creates an <code>MSnSet</code> object with 3 slots:</p>
<ul>
<li><p><code>exprs</code>: contains a matrix of peptide intensities of which the rows correspond to the peptides present in the dataset and the columns correspond to the different mass spec runs.</p></li>
<li><p><code>fData</code>: contains a data frame containing all additional information (such as peptide sequences, protein names, scores, number of missed cleavages, etc.) present in the input file.</p></li>
<li><p><code>pData</code>: contains a data frame with 0 columns and as many rows as there are mass spec runs. This slot will be used to store the experimental annotation.</p></li>
</ul>
<p>If you want to assess whether importing happened in a correct way, you can always check the <code>exprs</code>, <code>fData</code> and <code>pData</code> slots.</p>
<p>The <code>exprs</code> slot should contain all the peptide intensities grouped per column.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">exprs</span>(peptidesFranc))</code></pre></div>
<pre><code>##   X1WT_20_2h_n3_1 X1WT_20_2h_n3_2 X1WT_20_2h_n3_3 X1WT_20_2h_n4_1
## 1               0               0         9290400        92996000
## 2               0        20679000               0               0
## 3        28853000        28091000        30436000        19476000
## 4       339830000       420390000       393930000       235060000
## 5       240280000               0       241890000       222250000
## 6               0        31069000        31448000               0
##   X1WT_20_2h_n4_2 X1WT_20_2h_n4_3 X1WT_20_2h_n5_1 X1WT_20_2h_n5_2
## 1               0        98059000               0        80803000
## 2               0               0               0        17358000
## 3        22050000        20687000        36588000        28654000
## 4       381090000       360270000       255710000       311370000
## 5       189930000               0       160440000       147230000
## 6        27721000        28657000               0               0
##   X1WT_20_2h_n5_3 X3D8_20_2h_n3_1 X3D8_20_2h_n3_2 X3D8_20_2h_n3_3
## 1               0               0        95433000               0
## 2        13841000        12278000        10712000               0
## 3         7009100        14174000         8548000         7859200
## 4       271440000       225140000       289880000       282500000
## 5       143340000       135910000       185740000       168620000
## 6               0               0               0               0
##   X3D8_20_2h_n4_1 X3D8_20_2h_n4_2 X3D8_20_2h_n4_3 X3D8_20_2h_n5_1
## 1               0               0               0               0
## 2         9500700        11111000         8723500         8125200
## 3        14871000        11690000         4470500               0
## 4       209490000       223610000       219700000       169120000
## 5       160240000       158000000               0        92542000
## 6               0        18253000               0               0
##   X3D8_20_2h_n5_2 X3D8_20_2h_n5_3
## 1               0               0
## 2               0               0
## 3         4048100         3422300
## 4       178150000       170400000
## 5       102350000       100090000
## 6               0               0</code></pre>
<p>The <code>fData</code> slot contains all other columns.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">fData</span>(peptidesFranc))</code></pre></div>
<pre><code>##      GI.number                Sequence Amino.acid.before First.amino.acid
## 1 gi|118497196               AAAEELDTR                 K                A
## 2 gi|118498194           AAAGFVITASHNK                 R                A
## 3 gi|118497331 AAANEYELALAYSIEEVAPDLHK                 K                A
## 4 gi|118496879            AAANNPQLEAFK                 K                A
## 5 gi|118497152             AAASAGLVDEK                 K                A
## 6 gi|118497492            AAASLDLYSYPK                 K                A
##   Second.amino.acid Second.last.amino.acid Last.amino.acid
## 1                 A                      T               R
## 2                 A                      N               K
## 3                 A                      H               K
## 4                 A                      F               K
## 5                 A                      E               K
## 6                 A                      P               K
##   Amino.acid.after A.Count R.Count N.Count D.Count C.Count Q.Count E.Count
## 1                K       3       1       0       1       0       0       2
## 2                F       4       0       1       0       0       0       0
## 3                Y       6       0       1       1       0       0       4
## 4                K       4       0       2       0       0       1       1
## 5                A       4       0       0       1       0       0       1
## 6                V       3       0       0       1       0       0       0
##   G.Count H.Count I.Count L.Count K.Count M.Count F.Count P.Count S.Count
## 1       0       0       0       1       0       0       0       0       0
## 2       1       1       1       0       1       0       1       0       1
## 3       0       1       1       3       1       0       0       1       1
## 4       0       0       0       1       1       0       1       1       0
## 5       1       0       0       1       1       0       0       0       1
## 6       0       0       0       2       1       0       0       1       2
##   T.Count W.Count Y.Count V.Count U.Count Length Missed.cleavages
## 1       1       0       0       0       0      9                0
## 2       1       0       0       1       0     13                0
## 3       0       0       2       1       0     23                0
## 4       0       0       0       0       0     12                0
## 5       0       0       0       1       0     11                0
## 6       0       0       2       0       0     12                0
##         Mass     Proteins
## 1 97.446.689 WP_003038655
## 2 12.856.779 WP_003041237
## 3 25.162.435 WP_003038915
## 4 12.726.463 WP_003038264
## 5 10.305.295 WP_003035781
## 6 12.976.554 WP_003039212
##                                                    Protein.names
## 1                  hypothetical protein [Francisella tularensis]
## 2             phosphoglucosamine mutase [Francisella tularensis]
## 3     glycine--tRNA ligase subunit beta [Francisella tularensis]
## 4              molecular chaperone HtpG [Francisella tularensis]
## 5 delta-aminolevulinic acid dehydratase [Francisella tularensis]
## 6                         phosphorylase [Francisella tularensis]
##   Leading.razor.protein Unique..Groups. Unique..Proteins. Charges
## 1          gi|118497196             yes               yes     1,2
## 2          gi|118498194             yes               yes       2
## 3          gi|118497331             yes               yes       3
## 4          gi|118496879             yes               yes       2
## 5          gi|118497152             yes               yes       2
## 6          gi|118497492             yes               yes       2
##          PEP  Score Experiment.1WT_20_2h_n3_1 Experiment.1WT_20_2h_n3_2
## 1 0.00012531  59.35                        NA                        NA
## 2   5,30E-02 45.825                        NA                         1
## 3   1,05E-41 84.327                         1                         1
## 4   8,75E-20 108.56                         1                         1
## 5   9,51E-01 67.385                         1                        NA
## 6   0.012563 31.675                        NA                         1
##   Experiment.1WT_20_2h_n3_3 Experiment.1WT_20_2h_n4_1
## 1                         1                         1
## 2                        NA                        NA
## 3                         2                         1
## 4                         1                         1
## 5                         1                         1
## 6                         1                        NA
##   Experiment.1WT_20_2h_n4_2 Experiment.1WT_20_2h_n4_3
## 1                        NA                         1
## 2                        NA                        NA
## 3                         1                         1
## 4                         1                         1
## 5                         1                        NA
## 6                         1                         1
##   Experiment.1WT_20_2h_n5_1 Experiment.1WT_20_2h_n5_2
## 1                        NA                         1
## 2                        NA                         1
## 3                         2                         2
## 4                         1                         1
## 5                         1                         1
## 6                        NA                        NA
##   Experiment.1WT_20_2h_n5_3 Experiment.3D8_20_2h_n3_1
## 1                        NA                        NA
## 2                         1                         1
## 3                         1                         1
## 4                         1                         1
## 5                         1                         1
## 6                        NA                        NA
##   Experiment.3D8_20_2h_n3_2 Experiment.3D8_20_2h_n3_3
## 1                         1                        NA
## 2                         1                        NA
## 3                         2                         1
## 4                         1                         1
## 5                         1                         1
## 6                        NA                        NA
##   Experiment.3D8_20_2h_n4_1 Experiment.3D8_20_2h_n4_2
## 1                        NA                        NA
## 2                         1                         1
## 3                         1                         1
## 4                         1                         1
## 5                         1                         1
## 6                        NA                         1
##   Experiment.3D8_20_2h_n4_3 Experiment.3D8_20_2h_n5_1
## 1                        NA                        NA
## 2                         1                         1
## 3                         1                        NA
## 4                         1                         1
## 5                        NA                         1
## 6                        NA                        NA
##   Experiment.3D8_20_2h_n5_2 Experiment.3D8_20_2h_n5_3   Intensity Reverse
## 1                        NA                        NA  1329500000        
## 2                        NA                        NA   167190000        
## 3                         1                         1   556750000        
## 4                         1                         1 14830000000        
## 5                         1                         1  4820900000        
## 6                        NA                        NA   137150000        
##   Contaminant id Protein.group.IDs Mod..peptide.IDs
## 1              0               419                0
## 2              1              1150                1
## 3              2               513                2
## 4              3               199                3
## 5              4               388                4
## 6              5               624                5
##                                                                                                                                                               Evidence.IDs
## 1                                                                                                                           0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18
## 2                                                                                                                             19;20;21;22;23;24;25;26;27;28;29;30;31;32;33
## 3                                         34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76
## 4 77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124
## 5                                              125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155
## 6                                                                                                                                                      156;157;158;159;160
##                                                                                                                                                                                                                                                                                                                                                         MS.MS.IDs
## 1                                                                                                                                                                                                                                                                                                                  0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18
## 2                                                                                                                                                                                                                                                                                                                    19;20;21;22;23;24;25;26;27;28;29;30;31;32;33
## 3                                                                                                                                                                                                                       34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79
## 4 80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172
## 5                                                                                                                                                                                                                                     173;174;175;176;177;178;179;180;181;182;183;184;185;186;187;188;189;190;191;192;193;194;195;196;197;198;199;200;201;202;203
## 6                                                                                                                                                                                                                                                                                                                                             204;205;206;207;208
##   Best.MS.MS Oxidation..M..site.IDs
## 1         15                       
## 2         26                       
## 3         39                       
## 4        159                       
## 5        191                       
## 6        205</code></pre>
<p>The <code>pData</code> slot will contain the experimental annotation. Right now, it has 18 rows (equal to the number of mass spec runs) and 0 columns (as the experimental annotation is not yet loaded).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pData</span>(peptidesFranc)</code></pre></div>
<pre><code>## data frame with 0 columns and 18 rows</code></pre>
</div>
<div id="preprocessing" class="section level3">
<h3>2. Preprocessing</h3>
</div>
<div id="create-an-experimental-annotation-data-frame." class="section level3">
<h3>2.1. Create an experimental annotation data frame.</h3>
<p>Now, we are going to create a data frame that contains the experimental annotation. This data frame should contain all explanatory variables that groups different runs together. This is important in order to be able to include factors on which you will do inference (such as treatment effects or effects over time, or, in our case: the effect of the genetic knock-out) as well as other factors that might have an influence (such as biological repeats, technical repeats and potential confounders such as batch effects).</p>
<p><strong>Note that the “experimental annotation” can also be given as an Excel file or a tab delimited file!</strong> This file should have the column names as a header (i.e. the first row in the file) and the same structure as the <code>exp_annotation</code> data frame we create below. If you have your experimental annotation in a file, just provide the filepath to this annotation file in the preprocess_MaxQuant function.</p>
<p>Also note that exactly one column in the experimental annotation should contain the names of the mass spec runs. These names should be equal to the names you entered in the “Experiment” column in MaxQuant (i.e. the column names of the <code>exprs</code> slot of the peptidesFranc <code>MSnSet</code> object). In our case, we will call this column “run” (see below). By default, MSqRob guesses this column by taking the only column of which its elements correspond to the column names of the <code>exprs</code> slot. If there is no such a column, or there are multiple such columns, MSqRob throws an error.</p>
<p>In this example, we create the experimental annotation data frame ourselves.</p>
<p>Extract the names of the mass spec runs. The mass spec run names are equal to the column names of the <code>exprs</code> slot of the peptidesFranc <code>MSnSet</code> object. Other columns in the experimental annotation data frame will typically be derived from this column.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">runs &lt;-<span class="st"> </span><span class="kw">sampleNames</span>(peptidesFranc)
runs</code></pre></div>
<pre><code>##  [1] &quot;X1WT_20_2h_n3_1&quot; &quot;X1WT_20_2h_n3_2&quot; &quot;X1WT_20_2h_n3_3&quot;
##  [4] &quot;X1WT_20_2h_n4_1&quot; &quot;X1WT_20_2h_n4_2&quot; &quot;X1WT_20_2h_n4_3&quot;
##  [7] &quot;X1WT_20_2h_n5_1&quot; &quot;X1WT_20_2h_n5_2&quot; &quot;X1WT_20_2h_n5_3&quot;
## [10] &quot;X3D8_20_2h_n3_1&quot; &quot;X3D8_20_2h_n3_2&quot; &quot;X3D8_20_2h_n3_3&quot;
## [13] &quot;X3D8_20_2h_n4_1&quot; &quot;X3D8_20_2h_n4_2&quot; &quot;X3D8_20_2h_n4_3&quot;
## [16] &quot;X3D8_20_2h_n5_1&quot; &quot;X3D8_20_2h_n5_2&quot; &quot;X3D8_20_2h_n5_3&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colnames</span>(<span class="kw">exprs</span>(peptidesFranc))</code></pre></div>
<pre><code>##  [1] &quot;X1WT_20_2h_n3_1&quot; &quot;X1WT_20_2h_n3_2&quot; &quot;X1WT_20_2h_n3_3&quot;
##  [4] &quot;X1WT_20_2h_n4_1&quot; &quot;X1WT_20_2h_n4_2&quot; &quot;X1WT_20_2h_n4_3&quot;
##  [7] &quot;X1WT_20_2h_n5_1&quot; &quot;X1WT_20_2h_n5_2&quot; &quot;X1WT_20_2h_n5_3&quot;
## [10] &quot;X3D8_20_2h_n3_1&quot; &quot;X3D8_20_2h_n3_2&quot; &quot;X3D8_20_2h_n3_3&quot;
## [13] &quot;X3D8_20_2h_n4_1&quot; &quot;X3D8_20_2h_n4_2&quot; &quot;X3D8_20_2h_n4_3&quot;
## [16] &quot;X3D8_20_2h_n5_1&quot; &quot;X3D8_20_2h_n5_2&quot; &quot;X3D8_20_2h_n5_3&quot;</code></pre>
<p>Note again that the sample names have a prepended “X” because a syntactically valid name in R cannot start with a number.</p>
<p>Add a factor for the genetic knock-out, which we will call “genotype” (i.e. wild type (WT) or knock-out (KO)).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Create the &quot;genotype&quot; vector</span>
genotype &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;WT&quot;</span>,<span class="dv">9</span>),<span class="kw">rep</span>(<span class="st">&quot;KO&quot;</span>,<span class="dv">9</span>)), <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;KO&quot;</span>))
genotype</code></pre></div>
<pre><code>##  [1] WT WT WT WT WT WT WT WT WT KO KO KO KO KO KO KO KO KO
## Levels: WT KO</code></pre>
<p>Add a factor for each of the 6 biological replicates (i.e. 3 biological repeats for each genotype).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Create a factor for the biological replicates</span>
n &lt;-<span class="st"> </span><span class="kw">nchar</span>(runs)
biorep &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">paste0</span>(<span class="st">&quot;b_&quot;</span>,<span class="kw">rep</span>(<span class="dv">1</span>:<span class="dv">6</span>,<span class="dt">each=</span><span class="dv">3</span>)))
biorep</code></pre></div>
<pre><code>##  [1] b_1 b_1 b_1 b_2 b_2 b_2 b_3 b_3 b_3 b_4 b_4 b_4 b_5 b_5 b_5 b_6 b_6
## [18] b_6
## Levels: b_1 b_2 b_3 b_4 b_5 b_6</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Alternative: extract the biological repeat based on the mass spec run names.</span>
<span class="co"># biorep &lt;- as.factor(paste0(&quot;b_&quot;,factor(as.numeric(factor(paste0(substr(runs, 1,1),substr(runs, n-2,n-2))))+c(rep(0,9),rep(3,9)))))</span></code></pre></div>
<p>Each of the 18 technical repeats are here represented by the 18 mass spec runs.</p>
<p>Create an experimental annotation data frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">exp_annotation &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">run=</span>runs, <span class="dt">genotype=</span>genotype, <span class="dt">biorep=</span>biorep)
exp_annotation</code></pre></div>
<pre><code>##                run genotype biorep
## 1  X1WT_20_2h_n3_1       WT    b_1
## 2  X1WT_20_2h_n3_2       WT    b_1
## 3  X1WT_20_2h_n3_3       WT    b_1
## 4  X1WT_20_2h_n4_1       WT    b_2
## 5  X1WT_20_2h_n4_2       WT    b_2
## 6  X1WT_20_2h_n4_3       WT    b_2
## 7  X1WT_20_2h_n5_1       WT    b_3
## 8  X1WT_20_2h_n5_2       WT    b_3
## 9  X1WT_20_2h_n5_3       WT    b_3
## 10 X3D8_20_2h_n3_1       KO    b_4
## 11 X3D8_20_2h_n3_2       KO    b_4
## 12 X3D8_20_2h_n3_3       KO    b_4
## 13 X3D8_20_2h_n4_1       KO    b_5
## 14 X3D8_20_2h_n4_2       KO    b_5
## 15 X3D8_20_2h_n4_3       KO    b_5
## 16 X3D8_20_2h_n5_1       KO    b_6
## 17 X3D8_20_2h_n5_2       KO    b_6
## 18 X3D8_20_2h_n5_3       KO    b_6</code></pre>
<p>Note that you can also import the experimental annotation from a tab-delimited or Excel file. As an example, we show you here how you can import data from an experimental annotation file delivered with our package. First, provide the path to where the data is saved.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file_exp_annotation &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/Francisella&quot;</span>, <span class="st">&quot;label-free_Francisella_annotation.xlsx&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MSqRobData&quot;</span>)
file_exp_annotation</code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Versions/3.3/Resources/library/MSqRobData/extdata/Francisella/label-free_Francisella_annotation.xlsx&quot;</code></pre>
<p>Then, import the data using the <code>read.xlsx</code> function from the <code>openxlsx</code> package and convert the columns of the data frame to factors.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Import Excel file</span>
exp_annotation_xlsx &lt;-<span class="st"> </span>openxlsx::<span class="kw">read.xlsx</span>(file_exp_annotation)
<span class="co">#Convert columns of data frame to factors</span>
exp_annotation_xlsx &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">unclass</span>(exp_annotation_xlsx))
exp_annotation_xlsx</code></pre></div>
<pre><code>##               run genotype biorep
## 1  1WT_20_2h_n3_1       WT    b_1
## 2  1WT_20_2h_n3_2       WT    b_1
## 3  1WT_20_2h_n3_3       WT    b_1
## 4  1WT_20_2h_n4_1       WT    b_2
## 5  1WT_20_2h_n4_2       WT    b_2
## 6  1WT_20_2h_n4_3       WT    b_2
## 7  1WT_20_2h_n5_1       WT    b_3
## 8  1WT_20_2h_n5_2       WT    b_3
## 9  1WT_20_2h_n5_3       WT    b_3
## 10 3D8_20_2h_n3_1       KO    b_4
## 11 3D8_20_2h_n3_2       KO    b_4
## 12 3D8_20_2h_n3_3       KO    b_4
## 13 3D8_20_2h_n4_1       KO    b_5
## 14 3D8_20_2h_n4_2       KO    b_5
## 15 3D8_20_2h_n4_3       KO    b_5
## 16 3D8_20_2h_n5_1       KO    b_6
## 17 3D8_20_2h_n5_2       KO    b_6
## 18 3D8_20_2h_n5_3       KO    b_6</code></pre>
<p>Note that here, there is no “X” prepended to the run names. However, names will be changed automatically to syntactically valid names by MSqRob during preprocessing.</p>
</div>
<div id="preprocess-the-data-using-the-preprocess_maxquant-function." class="section level3">
<h3>2.2. Preprocess the data using the preprocess_MaxQuant function.</h3>
<p>The identification of peptides that carry one or more chemical modifications is often not as reliable as the identification of unmodified peptides. Therefore, a common step during the preprocessing of proteomics data consists of removing the proteins of which all identified peptides have one or more modification sites. If we want to remove these proteins that are only identified by modified peptides from the dataset, we also need the proteinGroups.txt file. This file can be found in the same location as the peptides.txt file (“path_to_raw_files/combined/txt/proteinGroups.txt”). Here, we make use of the proteinGroups.txt file that is delivered with the package.</p>
<p><strong>Note:</strong> if you don’t want to remove proteins that are only identified by modified peptides, set the <code>remove_only_site</code> argument of the <code>preprocess_MaxQuant</code> function to <code>FALSE</code> and leave the <code>file_proteinGroups</code> argument at its default value <code>NULL</code>.</p>
<p>Define the path to the proteinGroups.txt file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file_proteinGroups &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/Francisella&quot;</span>, <span class="st">&quot;proteinGroups.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MSqRobData&quot;</span>)
file_proteinGroups</code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Versions/3.3/Resources/library/MSqRobData/extdata/Francisella/proteinGroups.txt&quot;</code></pre>
<p>Add to the <code>useful_properties</code> argument all column names of the <code>fData</code> slot of the peptidesFranc <code>MSnSet</code> object that you would like to keep (such as gene names, protein names, ontologies, etc.). Other columns will be removed for improved efficiency and memory usage. Here, we would like to keep the GI number and the protein names. These 2 columns are present in our peptides.txt file, but note that you can use basic R manipulations to add other columns to the fData slot if you want to. We certainly want to keep the “Sequence” slot, as peptide-specific effects must be included in our model.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peptidesFranc2 &lt;-<span class="st"> </span><span class="kw">preprocess_MaxQuant</span>(peptidesFranc, <span class="dt">accession=</span><span class="st">&quot;Proteins&quot;</span>, <span class="dt">exp_annotation=</span>exp_annotation, <span class="dt">logtransform=</span><span class="ot">TRUE</span>, <span class="dt">base=</span><span class="dv">2</span>, <span class="dt">normalisation=</span><span class="st">&quot;quantiles&quot;</span>, <span class="dt">smallestUniqueGroups=</span><span class="ot">TRUE</span>, <span class="dt">useful_properties=</span><span class="kw">c</span>(<span class="st">&quot;GI number&quot;</span>,<span class="st">&quot;Protein.names&quot;</span>,<span class="st">&quot;Sequence&quot;</span>), <span class="dt">filter=</span><span class="kw">c</span>(<span class="st">&quot;Contaminant&quot;</span>,<span class="st">&quot;Reverse&quot;</span>), <span class="dt">remove_only_site=</span><span class="ot">TRUE</span>, <span class="dt">file_proteinGroups=</span>file_proteinGroups,  <span class="dt">filter_symbol=</span><span class="st">&quot;+&quot;</span>, <span class="dt">minIdentified=</span><span class="dv">2</span>)</code></pre></div>
<p><code>preprocess_MaxQuant</code> internally executes the following preprocessing steps in this order:</p>
<ol style="list-style-type: decimal">
<li><p>Peptide intensities are (log2-)transformed (<code>logtransform=TRUE, base=2</code>).</p></li>
<li><p>As a normalization approach, we choose to quantile normalize the peptide intensities (<code>normalisation=&quot;quantiles&quot;</code>). Other options are <code>&quot;sum&quot;</code>, <code>&quot;max&quot;</code>, <code>&quot;center.mean&quot;</code>, <code>&quot;center.median&quot;</code>, <code>&quot;quantiles.robust&quot;</code>, <code>&quot;vsn&quot;</code> or <code>&quot;none&quot;</code> (see <code>MSnbase::normalise</code> for more information).</p></li>
<li><p>Handling overlapping protein groups (<code>smallestUniqueGroups=TRUE</code>): in our approach a peptide can map to multiple accessions (<code>accession=&quot;Proteins&quot;</code>), as long as there is none of these proteins present in a smaller subgroup. Peptides that map to protein groups for wich also subgroups are present in the dataset are removed from the dataset. (Proteins in a protein group are separated by a <code>;</code>).</p></li>
<li><p>Remove contaminants and reverse sequences (<code>filter=c(&quot;Contaminant&quot;,&quot;Reverse&quot;), filter_symbol=&quot;+&quot;</code>): each row with the filter symbol (“+” in our case) in the columns “Contaminant” and “Reverse” will be removed from the data. Note that the “Contaminant” column in older MaxQuant peptides.txt files can also be called “Potential.contaminant”.</p></li>
<li><p>Remove all proteins that are only identified by peptides carrying a modification site (<code>remove_only_site=TRUE</code>, <code>file_proteinGroups=file_proteinGroups</code>).</p></li>
<li><p>Remove all superfluous columns in the <code>fData</code> slot, except the “Proteins”, “GI.number”, “Protein.names” and “Sequence” columns (<code>useful_properties=c(&quot;GI.number&quot;,&quot;Protein.names&quot;,&quot;Sequence&quot;)</code>). Note that the “Proteins” column does not have to be provided: the <code>accession</code> column is always retained.</p></li>
<li><p>Remove all peptides that are only identified once in the dataset (<code>minIdentified=2</code>).</p></li>
<li><p>Add experimental annotation (<code>exp_annotation=exp_annotation</code>).</p></li>
</ol>
</div>
<div id="convert-the-msnset-object-peptidesfranc2-into-a-protdata-object-proteinsfranc." class="section level3">
<h3>3. Convert the MSnSet object peptidesFranc2 into a protdata object proteinsFranc.</h3>
<p><strong>Caution:</strong> this step might take some time! (almost 2 minutes on our system). The <code>accession</code> argument denotes by which column the individual peptide observations should be grouped. The <code>annotations</code> argument indicates which columns in peptidesFranc2 are annotations linked to the accessions (in our case: the GI number and the protein names). These annotations will be added to a separate <code>annotation</code> slot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(proteinsFranc &lt;-<span class="st"> </span><span class="kw">MSnSet2protdata</span>(peptidesFranc2, <span class="dt">accession=</span><span class="st">&quot;Proteins&quot;</span>, <span class="dt">annotations =</span> <span class="kw">c</span>(<span class="st">&quot;GI.number&quot;</span>, <span class="st">&quot;Protein.names&quot;</span>)))</code></pre></div>
<pre><code>##    user  system elapsed 
## 124.075   4.173 145.849</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">proteinsFranc</code></pre></div>
<pre><code>## @accession 
## 
## [1] &quot;WP_003038655&quot;
## 
##  @annotation 
## 
##                                       GI.number 
##                                  &quot;gi|118497196&quot; 
##                                   Protein.names 
## &quot;hypothetical protein [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value           Sequence             run genotype biorep
## 1.1    25.11978       AVVVFFDYGCGK X1WT_20_2h_n3_1       WT    b_1
## 1.2    25.34586 DLLSDPATPTVGPQDANK X1WT_20_2h_n3_1       WT    b_1
## 1.3    24.11491   ETNGELTVQDVDNVVK X1WT_20_2h_n3_1       WT    b_1
## 1.4    26.50674          IIAIPDIVK X1WT_20_2h_n3_1       WT    b_1
## 1.5    24.69046     QTNVITDNYQQAAK X1WT_20_2h_n3_1       WT    b_1
## 2.1    24.75196       AVVVFFDYGCGK X1WT_20_2h_n3_2       WT    b_1
## 
##  96 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003041237&quot;
## 
##  @annotation 
## 
##                                            GI.number 
##                                       &quot;gi|118498194&quot; 
##                                        Protein.names 
## &quot;phosphoglucosamine mutase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value         Sequence             run genotype biorep
## 1.1    25.81732    ADLGISLDGDADR X1WT_20_2h_n3_1       WT    b_1
## 1.2    27.36716        FVIVGQDTR X1WT_20_2h_n3_1       WT    b_1
## 1.3    24.91571  GEVANSTITAEFTQK X1WT_20_2h_n3_1       WT    b_1
## 1.4    27.38539 ILANAIDEYIESIHSR X1WT_20_2h_n3_1       WT    b_1
## 1.6    27.93398    SLATNEAEYLVEK X1WT_20_2h_n3_1       WT    b_1
## 1.7    26.59163       VASDVNDVEK X1WT_20_2h_n3_1       WT    b_1
## 
##  187 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003038915&quot;
## 
##  @annotation 
## 
##                                                    GI.number 
##                                               &quot;gi|118497331&quot; 
##                                                Protein.names 
## &quot;glycine--tRNA ligase subunit beta [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value                Sequence             run genotype biorep
## 1      24.37301 AAANEYELALAYSIEEVAPDLHK X1WT_20_2h_n3_1       WT    b_1
## 1.3    22.60222   EEGIAVDIFEAINNTNYDSIK X1WT_20_2h_n3_1       WT    b_1
## 1.4    25.34002           EGEPTQVGLGFAK X1WT_20_2h_n3_1       WT    b_1
## 1.5    25.13163          EYSYALELLTCLDK X1WT_20_2h_n3_1       WT    b_1
## 1.6    24.09977       FHHPQAIVIDNISDYIK X1WT_20_2h_n3_1       WT    b_1
## 1.8    25.75712              IIDITLESYK X1WT_20_2h_n3_1       WT    b_1
## 
##  215 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003038264&quot;
## 
##  @annotation 
## 
##                                           GI.number 
##                                      &quot;gi|118496879&quot; 
##                                       Protein.names 
## &quot;molecular chaperone HtpG [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value        Sequence             run genotype biorep
## 1      27.92076    AAANNPQLEAFK X1WT_20_2h_n3_1       WT    b_1
## 1.2    26.48741 EGQDTIYYITSDSYK X1WT_20_2h_n3_1       WT    b_1
## 1.4    27.33299    EHLDLLEYHVLK X1WT_20_2h_n3_1       WT    b_1
## 1.5    26.37656         EILQHNK X1WT_20_2h_n3_1       WT    b_1
## 1.6    28.11488        ELLPPYLR X1WT_20_2h_n3_1       WT    b_1
## 1.7    28.77850    ELVSNSSDAIEK X1WT_20_2h_n3_1       WT    b_1
## 
##  338 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003035781&quot;
## 
##  @annotation 
## 
##                                                        GI.number 
##                                                   &quot;gi|118497152&quot; 
##                                                    Protein.names 
## &quot;delta-aminolevulinic acid dehydratase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value        Sequence             run genotype biorep
## 1      27.47327     AAASAGLVDEK X1WT_20_2h_n3_1       WT    b_1
## 1.1    24.05872 VILTYTALDVANWLK X1WT_20_2h_n3_1       WT    b_1
## 2.1    23.88890 VILTYTALDVANWLK X1WT_20_2h_n3_2       WT    b_1
## 2.2    25.37749      YASAYYGPFR X1WT_20_2h_n3_2       WT    b_1
## 3      27.47269     AAASAGLVDEK X1WT_20_2h_n3_3       WT    b_1
## 3.1    24.93174 VILTYTALDVANWLK X1WT_20_2h_n3_3       WT    b_1
## 
##  41 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003039212&quot;
## 
##  @annotation 
## 
##                                GI.number 
##                           &quot;gi|118497492&quot; 
##                            Protein.names 
## &quot;phosphorylase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value             Sequence             run genotype biorep
## 1.2    22.96510 FPAGLNDVEFVAEHIFQHSK X1WT_20_2h_n3_1       WT    b_1
## 1.3    25.10257           LQDLLTFIGK X1WT_20_2h_n3_1       WT    b_1
## 1.6    24.55598             VVFSVDYR X1WT_20_2h_n3_1       WT    b_1
## 2      24.53369         AAASLDLYSYPK X1WT_20_2h_n3_2       WT    b_1
## 2.1    24.68029              EVLFLNR X1WT_20_2h_n3_2       WT    b_1
## 2.2    23.16038 FPAGLNDVEFVAEHIFQHSK X1WT_20_2h_n3_2       WT    b_1
## 
##  56 more rows... 
## 
## 1182 more elements...</code></pre>
<p>The <code>system.time</code> function is only used here to print out how long the operation lasted. If you are not interested in that, you can of course omit this.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
proteinsFranc &lt;-<span class="st"> </span><span class="kw">MSnSet2protdata</span>(peptidesFranc2, <span class="dt">accession=</span><span class="st">&quot;Proteins&quot;</span>, <span class="dt">annotations =</span> <span class="kw">c</span>(<span class="st">&quot;GI.number&quot;</span>, <span class="st">&quot;Protein.names&quot;</span>))</code></pre></div>
</div>
<div id="advanced-add-information-on-the-design-after-creating-your-protdata-object" class="section level3">
<h3>Advanced: add information on the design after creating your protdata object</h3>
<p>Imagine you forgot a factor in your experimental annotation. Should you redo the whole preprocessing part? No, often, you can still add this factor thanks to the <code>addVarFromVar</code> function. We will now show how we can add an extra (redundant) factor called <code>techrep</code>.</p>
<p>Create a factor for each of the 18 technical replicates (each sample was analyzed in triplicate). This factor is completely redundant, as the “run” column in the experimental annotation already has 18 different levels and is thus already equal to the factor for the technical replicates.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">techrep &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">paste0</span>(<span class="st">&quot;t_&quot;</span>,<span class="dv">1</span>:<span class="dv">18</span>))
<span class="kw">names</span>(techrep) &lt;-<span class="st"> </span><span class="kw">levels</span>(exp_annotation$run)
techrep</code></pre></div>
<pre><code>## X1WT_20_2h_n3_1 X1WT_20_2h_n3_2 X1WT_20_2h_n3_3 X1WT_20_2h_n4_1 
##             t_1             t_2             t_3             t_4 
## X1WT_20_2h_n4_2 X1WT_20_2h_n4_3 X1WT_20_2h_n5_1 X1WT_20_2h_n5_2 
##             t_5             t_6             t_7             t_8 
## X1WT_20_2h_n5_3 X3D8_20_2h_n3_1 X3D8_20_2h_n3_2 X3D8_20_2h_n3_3 
##             t_9            t_10            t_11            t_12 
## X3D8_20_2h_n4_1 X3D8_20_2h_n4_2 X3D8_20_2h_n4_3 X3D8_20_2h_n5_1 
##            t_13            t_14            t_15            t_16 
## X3D8_20_2h_n5_2 X3D8_20_2h_n5_3 
##            t_17            t_18 
## 18 Levels: t_1 t_10 t_11 t_12 t_13 t_14 t_15 t_16 t_17 t_18 t_2 ... t_9</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">proteinsFranc2 &lt;-<span class="st"> </span><span class="kw">addVarFromVar</span>(proteinsFranc, <span class="dt">basecol=</span><span class="st">&quot;run&quot;</span>, <span class="dt">name=</span><span class="st">&quot;techrep&quot;</span>, <span class="dt">vector=</span>techrep)
proteinsFranc2</code></pre></div>
<pre><code>## @accession 
## 
## [1] &quot;WP_003038655&quot;
## 
##  @annotation 
## 
##                                       GI.number 
##                                  &quot;gi|118497196&quot; 
##                                   Protein.names 
## &quot;hypothetical protein [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value           Sequence             run genotype biorep techrep
## 1.1    25.11978       AVVVFFDYGCGK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.2    25.34586 DLLSDPATPTVGPQDANK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.3    24.11491   ETNGELTVQDVDNVVK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.4    26.50674          IIAIPDIVK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.5    24.69046     QTNVITDNYQQAAK X1WT_20_2h_n3_1       WT    b_1     t_1
## 2.1    24.75196       AVVVFFDYGCGK X1WT_20_2h_n3_2       WT    b_1     t_2
## 
##  96 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003041237&quot;
## 
##  @annotation 
## 
##                                            GI.number 
##                                       &quot;gi|118498194&quot; 
##                                        Protein.names 
## &quot;phosphoglucosamine mutase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value         Sequence             run genotype biorep techrep
## 1.1    25.81732    ADLGISLDGDADR X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.2    27.36716        FVIVGQDTR X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.3    24.91571  GEVANSTITAEFTQK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.4    27.38539 ILANAIDEYIESIHSR X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.6    27.93398    SLATNEAEYLVEK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.7    26.59163       VASDVNDVEK X1WT_20_2h_n3_1       WT    b_1     t_1
## 
##  187 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003038915&quot;
## 
##  @annotation 
## 
##                                                    GI.number 
##                                               &quot;gi|118497331&quot; 
##                                                Protein.names 
## &quot;glycine--tRNA ligase subunit beta [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value                Sequence             run genotype biorep
## 1      24.37301 AAANEYELALAYSIEEVAPDLHK X1WT_20_2h_n3_1       WT    b_1
## 1.3    22.60222   EEGIAVDIFEAINNTNYDSIK X1WT_20_2h_n3_1       WT    b_1
## 1.4    25.34002           EGEPTQVGLGFAK X1WT_20_2h_n3_1       WT    b_1
## 1.5    25.13163          EYSYALELLTCLDK X1WT_20_2h_n3_1       WT    b_1
## 1.6    24.09977       FHHPQAIVIDNISDYIK X1WT_20_2h_n3_1       WT    b_1
## 1.8    25.75712              IIDITLESYK X1WT_20_2h_n3_1       WT    b_1
##     techrep
## 1       t_1
## 1.3     t_1
## 1.4     t_1
## 1.5     t_1
## 1.6     t_1
## 1.8     t_1
## 
##  215 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003038264&quot;
## 
##  @annotation 
## 
##                                           GI.number 
##                                      &quot;gi|118496879&quot; 
##                                       Protein.names 
## &quot;molecular chaperone HtpG [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value        Sequence             run genotype biorep techrep
## 1      27.92076    AAANNPQLEAFK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.2    26.48741 EGQDTIYYITSDSYK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.4    27.33299    EHLDLLEYHVLK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.5    26.37656         EILQHNK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.6    28.11488        ELLPPYLR X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.7    28.77850    ELVSNSSDAIEK X1WT_20_2h_n3_1       WT    b_1     t_1
## 
##  338 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003035781&quot;
## 
##  @annotation 
## 
##                                                        GI.number 
##                                                   &quot;gi|118497152&quot; 
##                                                    Protein.names 
## &quot;delta-aminolevulinic acid dehydratase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value        Sequence             run genotype biorep techrep
## 1      27.47327     AAASAGLVDEK X1WT_20_2h_n3_1       WT    b_1     t_1
## 1.1    24.05872 VILTYTALDVANWLK X1WT_20_2h_n3_1       WT    b_1     t_1
## 2.1    23.88890 VILTYTALDVANWLK X1WT_20_2h_n3_2       WT    b_1     t_2
## 2.2    25.37749      YASAYYGPFR X1WT_20_2h_n3_2       WT    b_1     t_2
## 3      27.47269     AAASAGLVDEK X1WT_20_2h_n3_3       WT    b_1     t_3
## 3.1    24.93174 VILTYTALDVANWLK X1WT_20_2h_n3_3       WT    b_1     t_3
## 
##  41 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003039212&quot;
## 
##  @annotation 
## 
##                                GI.number 
##                           &quot;gi|118497492&quot; 
##                            Protein.names 
## &quot;phosphorylase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##     quant_value             Sequence             run genotype biorep
## 1.2    22.96510 FPAGLNDVEFVAEHIFQHSK X1WT_20_2h_n3_1       WT    b_1
## 1.3    25.10257           LQDLLTFIGK X1WT_20_2h_n3_1       WT    b_1
## 1.6    24.55598             VVFSVDYR X1WT_20_2h_n3_1       WT    b_1
## 2      24.53369         AAASLDLYSYPK X1WT_20_2h_n3_2       WT    b_1
## 2.1    24.68029              EVLFLNR X1WT_20_2h_n3_2       WT    b_1
## 2.2    23.16038 FPAGLNDVEFVAEHIFQHSK X1WT_20_2h_n3_2       WT    b_1
##     techrep
## 1.2     t_1
## 1.3     t_1
## 1.6     t_1
## 2       t_2
## 2.1     t_2
## 2.2     t_2
## 
##  56 more rows... 
## 
## 1182 more elements...</code></pre>
</div>
<div id="fit-the-model-and-squeeze-residual-the-variances" class="section level3">
<h3>4. Fit the model and squeeze residual the variances</h3>
<p>Fit the robust ridge models for each protein. The <code>fit.model</code> function will fit a model to each protein in the <code>protdata</code> object <code>proteinsFranc</code>. The default setting <code>reg=&quot;ridge&quot;</code> indicates that ridge regression should be used. The default setting <code>weights=&quot;Huber&quot;</code> indicates that an M estimation algorithm should be used that assigns Huber weights to each peptide observation based on their residuals.</p>
<p>Caution: fitting the models takes some time (less than 15 minutes for the <em>Francisella tularensis</em> dataset on our system). The <code>system.time</code> function was again only added to output the duration of the model fitting.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(protLMFranc &lt;-<span class="st"> </span><span class="kw">fit.model</span>(proteinsFranc, <span class="dt">response=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">fixed=</span><span class="kw">c</span>(<span class="st">&quot;genotype&quot;</span>),  <span class="dt">random=</span><span class="kw">c</span>(<span class="st">&quot;biorep&quot;</span>,<span class="st">&quot;run&quot;</span>,<span class="st">&quot;Sequence&quot;</span>), <span class="dt">add.intercept=</span><span class="ot">TRUE</span>, <span class="dt">weights=</span><span class="st">&quot;Huber&quot;</span>))</code></pre></div>
<pre><code>##     user   system  elapsed 
##  914.079   11.033 1005.321</code></pre>
</div>
<div id="inspect-the-model" class="section level3">
<h3>4.1. Inspect the model</h3>
<p><code>protLMFranc</code> is a <code>protLM</code> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(protLMFranc)</code></pre></div>
<pre><code>## [1] &quot;protLM&quot;
## attr(,&quot;package&quot;)
## [1] &quot;MSqRob&quot;</code></pre>
<p>Printing out a <code>protLM</code> object returns an overview of the first 6 accessions with their annotations and their corresponding models.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">protLMFranc</code></pre></div>
<pre><code>## @accession 
## 
## [1] &quot;WP_003038655&quot;
## 
##  @annotation 
## 
##                                       GI.number 
##                                  &quot;gi|118497196&quot; 
##                                   Protein.names 
## &quot;hypothetical protein [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 88.6243
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  Sequence     (Intercept) 0.6643  
##  biorep       (Intercept) 0.2302  
##  ridgeGroup.1 (Intercept) 1.1058  
##  Residual                 0.2979  
## Number of obs: 102, groups:  
## run, 18; Sequence, 7; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##             -260  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## @accession 
## 
## [1] &quot;WP_003041237&quot;
## 
##  @annotation 
## 
##                                            GI.number 
##                                       &quot;gi|118498194&quot; 
##                                        Protein.names 
## &quot;phosphoglucosamine mutase [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 139.3629
## Random effects:
##  Groups       Name        Std.Dev. 
##  run          (Intercept) 0.000e+00
##  Sequence     (Intercept) 1.287e+00
##  biorep       (Intercept) 9.394e-02
##  ridgeGroup.1 (Intercept) 1.033e-05
##  Residual                 2.781e-01
## Number of obs: 193, groups:  
## run, 18; Sequence, 12; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -370.9  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## @accession 
## 
## [1] &quot;WP_003038915&quot;
## 
##  @annotation 
## 
##                                                    GI.number 
##                                               &quot;gi|118497331&quot; 
##                                                Protein.names 
## &quot;glycine--tRNA ligase subunit beta [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 379.1122
## Random effects:
##  Groups       Name        Std.Dev.
##  Sequence     (Intercept) 1.07824 
##  run          (Intercept) 0.00000 
##  biorep       (Intercept) 0.07184 
##  ridgeGroup.1 (Intercept) 2.09142 
##  Residual                 0.46136 
## Number of obs: 221, groups:  
## Sequence, 18; run, 18; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -365.7  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## @accession 
## 
## [1] &quot;WP_003038264&quot;
## 
##  @annotation 
## 
##                                           GI.number 
##                                      &quot;gi|118496879&quot; 
##                                       Protein.names 
## &quot;molecular chaperone HtpG [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 431.674
## Random effects:
##  Groups       Name        Std.Dev.
##  Sequence     (Intercept) 1.01989 
##  run          (Intercept) 0.09085 
##  biorep       (Intercept) 0.12137 
##  ridgeGroup.1 (Intercept) 0.40874 
##  Residual                 0.36278 
## Number of obs: 344, groups:  
## Sequence, 23; run, 18; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##             -490  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## @accession 
## 
## [1] &quot;WP_003035781&quot;
## 
##  @annotation 
## 
##                                                        GI.number 
##                                                   &quot;gi|118497152&quot; 
##                                                    Protein.names 
## &quot;delta-aminolevulinic acid dehydratase [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 57.4696
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  biorep       (Intercept) 0.2069  
##  Sequence     (Intercept) 1.9042  
##  ridgeGroup.1 (Intercept) 0.5633  
##  Residual                 0.3543  
## Number of obs: 47, groups:  
## run, 18; biorep, 6; Sequence, 3; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -173.3  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## @accession 
## 
## [1] &quot;WP_003039212&quot;
## 
##  @annotation 
## 
##                                GI.number 
##                           &quot;gi|118497492&quot; 
##                            Protein.names 
## &quot;phosphorylase [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 9.4736
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  Sequence     (Intercept) 0.6600  
##  biorep       (Intercept) 0.0000  
##  ridgeGroup.1 (Intercept) 0.3170  
##  Residual                 0.2034  
## Number of obs: 62, groups:  
## run, 18; Sequence, 7; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -192.6  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## 1182 more elements...</code></pre>
<p>The numbers in the <code>Std.Dev.</code> column give the estimated standard deviations of the random effects. Effects starting with <code>ridgeGroup</code> indicate shrunken fixed effects (we make use of the random effects framework of the <code>lm4</code> package in order to estimate Ridge penalties). However, as the fixed effects in the ridgeGroups are orthogonalized using the Gramm-Schmidt procedure, these numbers are not very informative.</p>
<p>A <code>protLM</code> object has three slots:</p>
<ol style="list-style-type: decimal">
<li>an <code>accession</code> slot: a vector containing all the accessions</li>
<li>a <code>model</code> slot: a list of all fitted models (one for each accession)</li>
<li>an <code>annotation</code> slot: data frame of which each row (one for each accession) corresponds to the annotations of a different accession.</li>
</ol>
<p>These slots can be accessed using the <code>getAccessions</code>, <code>getModels</code> and <code>getAnnotations</code> functions respectively. Here we give the accessions, models and annotations for the first 5 models.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getAccessions</span>(protLMFranc)[<span class="dv">1</span>:<span class="dv">5</span>]</code></pre></div>
<pre><code>## [1] WP_003038655 WP_003041237 WP_003038915 WP_003038264 WP_003035781
## 1293 Levels:  ... WP_011733734</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getModels</span>(protLMFranc)[<span class="dv">1</span>:<span class="dv">5</span>]</code></pre></div>
<pre><code>## [[1]]
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 88.6243
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  Sequence     (Intercept) 0.6643  
##  biorep       (Intercept) 0.2302  
##  ridgeGroup.1 (Intercept) 1.1058  
##  Residual                 0.2979  
## Number of obs: 102, groups:  
## run, 18; Sequence, 7; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##             -260  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## [[2]]
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 139.3629
## Random effects:
##  Groups       Name        Std.Dev. 
##  run          (Intercept) 0.000e+00
##  Sequence     (Intercept) 1.287e+00
##  biorep       (Intercept) 9.394e-02
##  ridgeGroup.1 (Intercept) 1.033e-05
##  Residual                 2.781e-01
## Number of obs: 193, groups:  
## run, 18; Sequence, 12; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -370.9  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## [[3]]
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 379.1122
## Random effects:
##  Groups       Name        Std.Dev.
##  Sequence     (Intercept) 1.07824 
##  run          (Intercept) 0.00000 
##  biorep       (Intercept) 0.07184 
##  ridgeGroup.1 (Intercept) 2.09142 
##  Residual                 0.46136 
## Number of obs: 221, groups:  
## Sequence, 18; run, 18; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -365.7  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## [[4]]
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 431.674
## Random effects:
##  Groups       Name        Std.Dev.
##  Sequence     (Intercept) 1.01989 
##  run          (Intercept) 0.09085 
##  biorep       (Intercept) 0.12137 
##  ridgeGroup.1 (Intercept) 0.40874 
##  Residual                 0.36278 
## Number of obs: 344, groups:  
## Sequence, 23; run, 18; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##             -490  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## [[5]]
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 57.4696
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  biorep       (Intercept) 0.2069  
##  Sequence     (Intercept) 1.9042  
##  ridgeGroup.1 (Intercept) 0.5633  
##  Residual                 0.3543  
## Number of obs: 47, groups:  
## run, 18; biorep, 6; Sequence, 3; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -173.3  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getAnnotations</span>(protLMFranc)[<span class="dv">1</span>:<span class="dv">5</span>,]</code></pre></div>
<pre><code>##      GI.number     
## [1,] &quot;gi|118497196&quot;
## [2,] &quot;gi|118498194&quot;
## [3,] &quot;gi|118497331&quot;
## [4,] &quot;gi|118496879&quot;
## [5,] &quot;gi|118497152&quot;
##      Protein.names                                                   
## [1,] &quot;hypothetical protein [Francisella tularensis]&quot;                 
## [2,] &quot;phosphoglucosamine mutase [Francisella tularensis]&quot;            
## [3,] &quot;glycine--tRNA ligase subunit beta [Francisella tularensis]&quot;    
## [4,] &quot;molecular chaperone HtpG [Francisella tularensis]&quot;             
## [5,] &quot;delta-aminolevulinic acid dehydratase [Francisella tularensis]&quot;</code></pre>
<p><code>protLM</code> objects can be subsetted using either their numeric index or their corresponding accession.</p>
<p>Retain only the part corresponding to model 24 and model 50.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">protLMFranc[<span class="kw">c</span>(<span class="dv">24</span>,<span class="dv">50</span>)]</code></pre></div>
<pre><code>## @accession 
## 
## [1] &quot;WP_003035135&quot;
## 
##  @annotation 
## 
##                                       GI.number 
##                                  &quot;gi|118498204&quot; 
##                                   Protein.names 
## &quot;superoxide dismutase [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 141.1656
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.00000 
##  Sequence     (Intercept) 1.47998 
##  biorep       (Intercept) 0.05131 
##  ridgeGroup.1 (Intercept) 1.32076 
##  Residual                 0.31142 
## Number of obs: 148, groups:  
## run, 18; Sequence, 10; biorep, 6; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -344.8  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling
## 
## @accession 
## 
## [1] &quot;WP_003041024&quot;
## 
##  @annotation 
## 
##                                 GI.number 
##                            &quot;gi|118496740&quot; 
##                             Protein.names 
## &quot;acetate kinase [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 26.3867
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  biorep       (Intercept) 0.2991  
##  Sequence     (Intercept) 0.3212  
##  ridgeGroup.1 (Intercept) 0.0000  
##  Residual                 0.2702  
## Number of obs: 36, groups:  
## run, 16; biorep, 6; Sequence, 4; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -147.4  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling</code></pre>
<p>Inspect the <code>protLM</code> object corresponding to accession <code>&quot;WP_003033643&quot;</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">protLMFranc[<span class="st">&quot;WP_003033643&quot;</span>]</code></pre></div>
<pre><code>## @accession 
## 
## [1] &quot;WP_003033643&quot;
## 
##  @annotation 
## 
##                                  GI.number 
##                             &quot;gi|118497379&quot; 
##                              Protein.names 
## &quot;isochorismatase [Francisella tularensis]&quot; 
## 
##  @model 
## 
## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 5.9967
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.7034  
##  biorep       (Intercept) 0.2896  
##  Sequence     (Intercept) 0.9152  
##  ridgeGroup.1 (Intercept) 0.3192  
##  Residual                 0.7598  
## Number of obs: 3, groups:  run, 3; biorep, 2; Sequence, 1; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -36.79  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling</code></pre>
<p>Say we want to inspect cell division protein <code>&quot;WP_003040227&quot;</code>. We can then select protein <code>&quot;WP_003040227&quot;</code> from the <code>protLM</code> object. We can extract the model via the <code>getModels</code> function. <code>simplify=TRUE</code> is the default setting and indicates that if you select one model, you only want this model and not a list with as only element this model. Note that if you select multiple models, you automatically receive a list of models.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modelWP_003040227 &lt;-<span class="st"> </span><span class="kw">getModels</span>(protLMFranc[<span class="st">&quot;WP_003040227&quot;</span>], <span class="dt">simplify=</span><span class="ot">TRUE</span>)
modelWP_003040227</code></pre></div>
<pre><code>## Linear mixed model fit by REML ['lmerMod']
## REML criterion at convergence: 35.7398
## Random effects:
##  Groups       Name        Std.Dev.
##  run          (Intercept) 0.0000  
##  biorep       (Intercept) 0.5352  
##  Sequence     (Intercept) 0.0000  
##  ridgeGroup.1 (Intercept) 0.8295  
##  Residual                 0.5013  
## Number of obs: 20, groups:  
## run, 15; biorep, 5; Sequence, 2; ridgeGroup.1, 1
## Fixed Effects:
## Intercept_MSqRob  
##           -111.7  
## fit warnings:
## Some predictor variables are on very different scales: consider rescaling</code></pre>
<p>The <code>betaBVcovDf</code> function extracts a list from a model containing the following four elements:</p>
<ol style="list-style-type: decimal">
<li>beta: the estimated parameters: both fixed effect sizes and BLUPs (best linear unbiased predictors) for the random effects</li>
<li>vcov: the variance-covariance matrix</li>
<li>df: the residual degrees of freedom</li>
<li>sigma: the residual standard deviation</li>
<li>df_exp: if the number of experimental units would be specified: a more conservative estimate for the degrees of freedom based on the number of experimental units minus the degrees of freedom lost due to between-treatment effects. Otherwise: <code>NULL</code>.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf &lt;-<span class="st"> </span><span class="kw">getBetaVcovDf</span>(modelWP_003040227)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf$beta</code></pre></div>
<pre><code>##                              [,1]
## (Intercept)          2.504493e+01
## runX1WT_20_2h_n4_1  -3.489353e-19
## runX1WT_20_2h_n4_2   6.742506e-19
## runX1WT_20_2h_n4_3  -5.735848e-19
## runX1WT_20_2h_n5_1  -2.700470e-19
## runX1WT_20_2h_n5_2   1.254289e-19
## runX1WT_20_2h_n5_3   2.537552e-20
## runX3D8_20_2h_n3_1   6.742506e-19
## runX3D8_20_2h_n3_2   4.211313e-20
## runX3D8_20_2h_n3_3  -8.681736e-20
## runX3D8_20_2h_n4_1   3.680183e-19
## runX3D8_20_2h_n4_2   1.424543e-19
## runX3D8_20_2h_n4_3  -3.815842e-19
## runX3D8_20_2h_n5_1  -3.520650e-19
## runX3D8_20_2h_n5_2  -1.057841e-19
## runX3D8_20_2h_n5_3   6.692662e-20
## biorepb_2           -2.829943e-01
## biorepb_3           -1.359206e-01
## biorepb_4            7.175990e-01
## biorepb_5            1.469156e-01
## biorepb_6           -4.455996e-01
## SequenceMVFTFYNNLK   6.567722e-19
## SequenceYTYIYQIASFR -6.567722e-19
## genotypeWT          -2.396076e-01</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf$vcov</code></pre></div>
<pre><code>##                       (Intercept) runX1WT_20_2h_n4_1 runX1WT_20_2h_n4_2
## (Intercept)          3.482721e-01      -4.512815e-20      -3.306087e-20
## runX1WT_20_2h_n4_1  -4.512815e-20       1.000000e-18       2.222670e-37
## runX1WT_20_2h_n4_2  -3.306087e-20       2.222670e-37       1.000000e-18
## runX1WT_20_2h_n4_3  -4.512815e-20       3.033949e-37       2.222670e-37
## runX1WT_20_2h_n5_1  -4.201585e-20       2.455932e-38       1.799215e-38
## runX1WT_20_2h_n5_2  -4.201585e-20       2.455932e-38       1.799215e-38
## runX1WT_20_2h_n5_3  -4.201585e-20       2.455932e-38       1.799215e-38
## runX3D8_20_2h_n3_1  -3.858522e-20       4.999768e-39       3.662828e-39
## runX3D8_20_2h_n3_2  -9.277889e-20       1.202204e-38       8.807340e-39
## runX3D8_20_2h_n3_3  -9.277889e-20       1.202204e-38       8.807340e-39
## runX3D8_20_2h_n4_1  -5.198607e-20       6.736213e-39       4.934948e-39
## runX3D8_20_2h_n4_2  -1.039721e-19       1.347243e-38       9.869896e-39
## runX3D8_20_2h_n4_3  -1.039721e-19       1.347243e-38       9.869896e-39
## runX3D8_20_2h_n5_1  -8.885397e-20       1.151345e-38       8.434754e-39
## runX3D8_20_2h_n5_2  -8.885397e-20       1.151345e-38       8.434754e-39
## runX3D8_20_2h_n5_3  -8.885397e-20       1.151345e-38       8.434754e-39
## biorepb_2           -1.405651e-01      -1.948525e-19      -1.427489e-19
## biorepb_3           -1.436774e-01       8.398305e-20       6.152596e-20
## biorepb_4           -2.554932e-01       3.310611e-20       2.425353e-20
## biorepb_5           -2.962860e-01       3.839193e-20       2.812592e-20
## biorepb_6           -3.038451e-01       3.937142e-20       2.884349e-20
## SequenceMVFTFYNNLK  -6.366993e-19      -1.044206e-37      -7.649847e-38
## SequenceYTYIYQIASFR -3.633007e-19       1.044206e-37       7.649847e-38
## genotypeWT          -1.625788e-01      -6.341422e-20      -4.645724e-20
##                     runX1WT_20_2h_n4_3 runX1WT_20_2h_n5_1
## (Intercept)              -4.512815e-20      -4.201585e-20
## runX1WT_20_2h_n4_1        3.033949e-37       2.455932e-38
## runX1WT_20_2h_n4_2        2.222670e-37       1.799215e-38
## runX1WT_20_2h_n4_3        1.000000e-18       2.455932e-38
## runX1WT_20_2h_n5_1        2.455932e-38       1.000000e-18
## runX1WT_20_2h_n5_2        2.455932e-38       2.807773e-37
## runX1WT_20_2h_n5_3        2.455932e-38       2.807773e-37
## runX3D8_20_2h_n3_1        4.999768e-39       4.654955e-39
## runX3D8_20_2h_n3_2        1.202204e-38       1.119293e-38
## runX3D8_20_2h_n3_3        1.202204e-38       1.119293e-38
## runX3D8_20_2h_n4_1        6.736213e-39       6.271646e-39
## runX3D8_20_2h_n4_2        1.347243e-38       1.254329e-38
## runX3D8_20_2h_n4_3        1.347243e-38       1.254329e-38
## runX3D8_20_2h_n5_1        1.151345e-38       1.071942e-38
## runX3D8_20_2h_n5_2        1.151345e-38       1.071942e-38
## runX3D8_20_2h_n5_3        1.151345e-38       1.071942e-38
## biorepb_2                -1.948525e-19       7.649736e-20
## biorepb_3                 8.398305e-20      -1.797206e-19
## biorepb_4                 3.310611e-20       3.082293e-20
## biorepb_5                 3.839193e-20       3.574421e-20
## biorepb_6                 3.937142e-20       3.665614e-20
## SequenceMVFTFYNNLK       -1.044206e-37       1.290457e-37
## SequenceYTYIYQIASFR       1.044206e-37      -1.290457e-37
## genotypeWT               -6.341422e-20      -5.904082e-20
##                     runX1WT_20_2h_n5_2 runX1WT_20_2h_n5_3
## (Intercept)              -4.201585e-20      -4.201585e-20
## runX1WT_20_2h_n4_1        2.455932e-38       2.455932e-38
## runX1WT_20_2h_n4_2        1.799215e-38       1.799215e-38
## runX1WT_20_2h_n4_3        2.455932e-38       2.455932e-38
## runX1WT_20_2h_n5_1        2.807773e-37       2.807773e-37
## runX1WT_20_2h_n5_2        1.000000e-18       2.807773e-37
## runX1WT_20_2h_n5_3        2.807773e-37       1.000000e-18
## runX3D8_20_2h_n3_1        4.654955e-39       4.654955e-39
## runX3D8_20_2h_n3_2        1.119293e-38       1.119293e-38
## runX3D8_20_2h_n3_3        1.119293e-38       1.119293e-38
## runX3D8_20_2h_n4_1        6.271646e-39       6.271646e-39
## runX3D8_20_2h_n4_2        1.254329e-38       1.254329e-38
## runX3D8_20_2h_n4_3        1.254329e-38       1.254329e-38
## runX3D8_20_2h_n5_1        1.071942e-38       1.071942e-38
## runX3D8_20_2h_n5_2        1.071942e-38       1.071942e-38
## runX3D8_20_2h_n5_3        1.071942e-38       1.071942e-38
## biorepb_2                 7.649736e-20       7.649736e-20
## biorepb_3                -1.797206e-19      -1.797206e-19
## biorepb_4                 3.082293e-20       3.082293e-20
## biorepb_5                 3.574421e-20       3.574421e-20
## biorepb_6                 3.665614e-20       3.665614e-20
## SequenceMVFTFYNNLK        1.290457e-37       1.290457e-37
## SequenceYTYIYQIASFR      -1.290457e-37      -1.290457e-37
## genotypeWT               -5.904082e-20      -5.904082e-20
##                     runX3D8_20_2h_n3_1 runX3D8_20_2h_n3_2
## (Intercept)              -3.858522e-20      -9.277889e-20
## runX1WT_20_2h_n4_1        4.999768e-39       1.202204e-38
## runX1WT_20_2h_n4_2        3.662828e-39       8.807340e-39
## runX1WT_20_2h_n4_3        4.999768e-39       1.202204e-38
## runX1WT_20_2h_n5_1        4.654955e-39       1.119293e-38
## runX1WT_20_2h_n5_2        4.654955e-39       1.119293e-38
## runX1WT_20_2h_n5_3        4.654955e-39       1.119293e-38
## runX3D8_20_2h_n3_1        1.000000e-18       1.365654e-37
## runX3D8_20_2h_n3_2        1.365654e-37       1.000000e-18
## runX3D8_20_2h_n3_3        1.365654e-37       3.283740e-37
## runX3D8_20_2h_n4_1        5.759560e-39       1.384897e-38
## runX3D8_20_2h_n4_2        1.151912e-38       2.769794e-38
## runX3D8_20_2h_n4_3        1.151912e-38       2.769794e-38
## runX3D8_20_2h_n5_1        9.844172e-39       2.367050e-38
## runX3D8_20_2h_n5_2        9.844172e-39       2.367050e-38
## runX3D8_20_2h_n5_3        9.844172e-39       2.367050e-38
## biorepb_2                 1.557328e-20       3.744624e-20
## biorepb_3                 1.591809e-20       3.827534e-20
## biorepb_4                -9.798016e-20      -2.355951e-19
## biorepb_5                 3.282566e-20       7.892992e-20
## biorepb_6                 3.366314e-20       8.094364e-20
## SequenceMVFTFYNNLK       -4.025024e-38      -9.678247e-38
## SequenceYTYIYQIASFR       4.025024e-38       9.678247e-38
## genotypeWT                1.801218e-20       4.331062e-20
##                     runX3D8_20_2h_n3_3 runX3D8_20_2h_n4_1
## (Intercept)              -9.277889e-20      -5.198607e-20
## runX1WT_20_2h_n4_1        1.202204e-38       6.736213e-39
## runX1WT_20_2h_n4_2        8.807340e-39       4.934948e-39
## runX1WT_20_2h_n4_3        1.202204e-38       6.736213e-39
## runX1WT_20_2h_n5_1        1.119293e-38       6.271646e-39
## runX1WT_20_2h_n5_2        1.119293e-38       6.271646e-39
## runX1WT_20_2h_n5_3        1.119293e-38       6.271646e-39
## runX3D8_20_2h_n3_1        1.365654e-37       5.759560e-39
## runX3D8_20_2h_n3_2        3.283740e-37       1.384897e-38
## runX3D8_20_2h_n3_3        1.000000e-18       1.384897e-38
## runX3D8_20_2h_n4_1        1.384897e-38       1.000000e-18
## runX3D8_20_2h_n4_2        2.769794e-38       3.558123e-37
## runX3D8_20_2h_n4_3        2.769794e-38       3.558123e-37
## runX3D8_20_2h_n5_1        2.367050e-38       1.326310e-38
## runX3D8_20_2h_n5_2        2.367050e-38       1.326310e-38
## runX3D8_20_2h_n5_3        2.367050e-38       1.326310e-38
## biorepb_2                 3.744624e-20       2.098196e-20
## biorepb_3                 3.827534e-20       2.144652e-20
## biorepb_4                -2.355951e-19       3.813710e-20
## biorepb_5                 7.892992e-20      -1.259201e-19
## biorepb_6                 8.094364e-20       4.535452e-20
## SequenceMVFTFYNNLK       -9.678247e-38      -3.945219e-37
## SequenceYTYIYQIASFR       9.678247e-38       3.945219e-37
## genotypeWT                4.331062e-20       2.426790e-20
##                     runX3D8_20_2h_n4_2 runX3D8_20_2h_n4_3
## (Intercept)              -1.039721e-19      -1.039721e-19
## runX1WT_20_2h_n4_1        1.347243e-38       1.347243e-38
## runX1WT_20_2h_n4_2        9.869896e-39       9.869896e-39
## runX1WT_20_2h_n4_3        1.347243e-38       1.347243e-38
## runX1WT_20_2h_n5_1        1.254329e-38       1.254329e-38
## runX1WT_20_2h_n5_2        1.254329e-38       1.254329e-38
## runX1WT_20_2h_n5_3        1.254329e-38       1.254329e-38
## runX3D8_20_2h_n3_1        1.151912e-38       1.151912e-38
## runX3D8_20_2h_n3_2        2.769794e-38       2.769794e-38
## runX3D8_20_2h_n3_3        2.769794e-38       2.769794e-38
## runX3D8_20_2h_n4_1        3.558123e-37       3.558123e-37
## runX3D8_20_2h_n4_2        1.000000e-18       7.116247e-37
## runX3D8_20_2h_n4_3        7.116247e-37       1.000000e-18
## runX3D8_20_2h_n5_1        2.652621e-38       2.652621e-38
## runX3D8_20_2h_n5_2        2.652621e-38       2.652621e-38
## runX3D8_20_2h_n5_3        2.652621e-38       2.652621e-38
## biorepb_2                 4.196391e-20       4.196391e-20
## biorepb_3                 4.289305e-20       4.289305e-20
## biorepb_4                 7.627420e-20       7.627420e-20
## biorepb_5                -2.518402e-19      -2.518402e-19
## biorepb_6                 9.070904e-20       9.070904e-20
## SequenceMVFTFYNNLK        2.109561e-37       2.109561e-37
## SequenceYTYIYQIASFR      -2.109561e-37      -2.109561e-37
## genotypeWT                4.853580e-20       4.853580e-20
##                     runX3D8_20_2h_n5_1 runX3D8_20_2h_n5_2
## (Intercept)              -8.885397e-20      -8.885397e-20
## runX1WT_20_2h_n4_1        1.151345e-38       1.151345e-38
## runX1WT_20_2h_n4_2        8.434754e-39       8.434754e-39
## runX1WT_20_2h_n4_3        1.151345e-38       1.151345e-38
## runX1WT_20_2h_n5_1        1.071942e-38       1.071942e-38
## runX1WT_20_2h_n5_2        1.071942e-38       1.071942e-38
## runX1WT_20_2h_n5_3        1.071942e-38       1.071942e-38
## runX3D8_20_2h_n3_1        9.844172e-39       9.844172e-39
## runX3D8_20_2h_n3_2        2.367050e-38       2.367050e-38
## runX3D8_20_2h_n3_3        2.367050e-38       2.367050e-38
## runX3D8_20_2h_n4_1        1.326310e-38       1.326310e-38
## runX3D8_20_2h_n4_2        2.652621e-38       2.652621e-38
## runX3D8_20_2h_n4_3        2.652621e-38       2.652621e-38
## runX3D8_20_2h_n5_1        1.000000e-18       6.042931e-37
## runX3D8_20_2h_n5_2        6.042931e-37       1.000000e-18
## runX3D8_20_2h_n5_3        6.042931e-37       6.042931e-37
## biorepb_2                 3.586211e-20       3.586211e-20
## biorepb_3                 3.665614e-20       3.665614e-20
## biorepb_4                 6.518348e-20       6.518348e-20
## biorepb_5                 7.559087e-20       7.559087e-20
## biorepb_6                -2.132926e-19      -2.132926e-19
## SequenceMVFTFYNNLK        3.487583e-38       3.487583e-38
## SequenceYTYIYQIASFR      -3.487583e-38      -3.487583e-38
## genotypeWT                4.147841e-20       4.147841e-20
##                     runX3D8_20_2h_n5_3     biorepb_2     biorepb_3
## (Intercept)              -8.885397e-20 -1.405651e-01 -1.436774e-01
## runX1WT_20_2h_n4_1        1.151345e-38 -1.948525e-19  8.398305e-20
## runX1WT_20_2h_n4_2        8.434754e-39 -1.427489e-19  6.152596e-20
## runX1WT_20_2h_n4_3        1.151345e-38 -1.948525e-19  8.398305e-20
## runX1WT_20_2h_n5_1        1.071942e-38  7.649736e-20 -1.797206e-19
## runX1WT_20_2h_n5_2        1.071942e-38  7.649736e-20 -1.797206e-19
## runX1WT_20_2h_n5_3        1.071942e-38  7.649736e-20 -1.797206e-19
## runX3D8_20_2h_n3_1        9.844172e-39  1.557328e-20  1.591809e-20
## runX3D8_20_2h_n3_2        2.367050e-38  3.744624e-20  3.827534e-20
## runX3D8_20_2h_n3_3        2.367050e-38  3.744624e-20  3.827534e-20
## runX3D8_20_2h_n4_1        1.326310e-38  2.098196e-20  2.144652e-20
## runX3D8_20_2h_n4_2        2.652621e-38  4.196391e-20  4.289305e-20
## runX3D8_20_2h_n4_3        2.652621e-38  4.196391e-20  4.289305e-20
## runX3D8_20_2h_n5_1        6.042931e-37  3.586211e-20  3.665614e-20
## runX3D8_20_2h_n5_2        6.042931e-37  3.586211e-20  3.665614e-20
## runX3D8_20_2h_n5_3        1.000000e-18  3.586211e-20  3.665614e-20
## biorepb_2                 3.586211e-20  5.329403e-01  2.615904e-01
## biorepb_3                 3.665614e-20  2.615904e-01  5.252941e-01
## biorepb_4                 6.518348e-20  1.031189e-01  1.054021e-01
## biorepb_5                 7.559087e-20  1.195832e-01  1.222309e-01
## biorepb_6                -2.132926e-19  1.226341e-01  1.253494e-01
## SequenceMVFTFYNNLK        3.487583e-38 -3.252491e-19  4.412846e-19
## SequenceYTYIYQIASFR      -3.487583e-38  3.252491e-19 -4.412846e-19
## genotypeWT                4.147841e-20 -1.975226e-01 -2.018960e-01
##                         biorepb_4     biorepb_5     biorepb_6
## (Intercept)         -2.554932e-01 -2.962860e-01 -3.038451e-01
## runX1WT_20_2h_n4_1   3.310611e-20  3.839193e-20  3.937142e-20
## runX1WT_20_2h_n4_2   2.425353e-20  2.812592e-20  2.884349e-20
## runX1WT_20_2h_n4_3   3.310611e-20  3.839193e-20  3.937142e-20
## runX1WT_20_2h_n5_1   3.082293e-20  3.574421e-20  3.665614e-20
## runX1WT_20_2h_n5_2   3.082293e-20  3.574421e-20  3.665614e-20
## runX1WT_20_2h_n5_3   3.082293e-20  3.574421e-20  3.665614e-20
## runX3D8_20_2h_n3_1  -9.798016e-20  3.282566e-20  3.366314e-20
## runX3D8_20_2h_n3_2  -2.355951e-19  7.892992e-20  8.094364e-20
## runX3D8_20_2h_n3_3  -2.355951e-19  7.892992e-20  8.094364e-20
## runX3D8_20_2h_n4_1   3.813710e-20 -1.259201e-19  4.535452e-20
## runX3D8_20_2h_n4_2   7.627420e-20 -2.518402e-19  9.070904e-20
## runX3D8_20_2h_n4_3   7.627420e-20 -2.518402e-19  9.070904e-20
## runX3D8_20_2h_n5_1   6.518348e-20  7.559087e-20 -2.132926e-19
## runX3D8_20_2h_n5_2   6.518348e-20  7.559087e-20 -2.132926e-19
## runX3D8_20_2h_n5_3   6.518348e-20  7.559087e-20 -2.132926e-19
## biorepb_2            1.031189e-01  1.195832e-01  1.226341e-01
## biorepb_3            1.054021e-01  1.222309e-01  1.253494e-01
## biorepb_4            4.910883e-01  2.173561e-01  2.229015e-01
## biorepb_5            2.173561e-01  4.222061e-01  2.584906e-01
## biorepb_6            2.229015e-01  2.584906e-01  4.104914e-01
## SequenceMVFTFYNNLK  -2.665182e-19  3.122128e-20  1.192614e-19
## SequenceYTYIYQIASFR  2.665182e-19 -3.122128e-20 -1.192614e-19
## genotypeWT           1.192682e-01  1.383109e-01  1.418396e-01
##                     SequenceMVFTFYNNLK SequenceYTYIYQIASFR    genotypeWT
## (Intercept)              -6.366993e-19       -3.633007e-19 -1.625788e-01
## runX1WT_20_2h_n4_1       -1.044206e-37        1.044206e-37 -6.341422e-20
## runX1WT_20_2h_n4_2       -7.649847e-38        7.649847e-38 -4.645724e-20
## runX1WT_20_2h_n4_3       -1.044206e-37        1.044206e-37 -6.341422e-20
## runX1WT_20_2h_n5_1        1.290457e-37       -1.290457e-37 -5.904082e-20
## runX1WT_20_2h_n5_2        1.290457e-37       -1.290457e-37 -5.904082e-20
## runX1WT_20_2h_n5_3        1.290457e-37       -1.290457e-37 -5.904082e-20
## runX3D8_20_2h_n3_1       -4.025024e-38        4.025024e-38  1.801218e-20
## runX3D8_20_2h_n3_2       -9.678247e-38        9.678247e-38  4.331062e-20
## runX3D8_20_2h_n3_3       -9.678247e-38        9.678247e-38  4.331062e-20
## runX3D8_20_2h_n4_1       -3.945219e-37        3.945219e-37  2.426790e-20
## runX3D8_20_2h_n4_2        2.109561e-37       -2.109561e-37  4.853580e-20
## runX3D8_20_2h_n4_3        2.109561e-37       -2.109561e-37  4.853580e-20
## runX3D8_20_2h_n5_1        3.487583e-38       -3.487583e-38  4.147841e-20
## runX3D8_20_2h_n5_2        3.487583e-38       -3.487583e-38  4.147841e-20
## runX3D8_20_2h_n5_3        3.487583e-38       -3.487583e-38  4.147841e-20
## biorepb_2                -3.252491e-19        3.252491e-19 -1.975226e-01
## biorepb_3                 4.412846e-19       -4.412846e-19 -2.018960e-01
## biorepb_4                -2.665182e-19        2.665182e-19  1.192682e-01
## biorepb_5                 3.122128e-20       -3.122128e-20  1.383109e-01
## biorepb_6                 1.192614e-19       -1.192614e-19  1.418396e-01
## SequenceMVFTFYNNLK        1.000000e-18        3.150407e-36  6.636905e-20
## SequenceYTYIYQIASFR       3.150407e-36        1.000000e-18 -6.636905e-20
## genotypeWT                6.636905e-20       -6.636905e-20  4.235156e-01</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf$df</code></pre></div>
<pre><code>## [1] 18.96145</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf$df_exp</code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf$sigma</code></pre></div>
<pre><code>## [1] 0.4691506</code></pre>
<p>The <code>getBetaVcovDfList</code> function can be applied on (a subset of) the <code>protLM</code> object <code>protLMFranc</code>. This returns a list containing one or more lists similar to the <code>betaBVcovDf</code> list.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaVcovDfList &lt;-<span class="st"> </span><span class="kw">getBetaVcovDfList</span>(protLMFranc[<span class="dv">1</span>:<span class="dv">2</span>])
<span class="co">#Inspect the structure of the list without printing it out compeletely:</span>
<span class="kw">str</span>(betaVcovDfList)</code></pre></div>
<pre><code>## List of 2
##  $ WP_003038655:List of 5
##   ..$ beta  : num [1:33, 1] 2.58e+01 -1.09e-19 2.34e-19 -6.71e-19 -1.34e-19 ...
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:33] &quot;(Intercept)&quot; &quot;runX1WT_20_2h_n3_1&quot; &quot;runX1WT_20_2h_n3_2&quot; &quot;runX1WT_20_2h_n3_3&quot; ...
##   .. .. ..$ : NULL
##   ..$ vcov  : num [1:33, 1:33] 8.85e-01 -1.97e-20 -2.90e-20 -2.39e-20 -3.38e-20 ...
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:33] &quot;(Intercept)&quot; &quot;runX1WT_20_2h_n3_1&quot; &quot;runX1WT_20_2h_n3_2&quot; &quot;runX1WT_20_2h_n3_3&quot; ...
##   .. .. ..$ : chr [1:33] &quot;(Intercept)&quot; &quot;runX1WT_20_2h_n3_1&quot; &quot;runX1WT_20_2h_n3_2&quot; &quot;runX1WT_20_2h_n3_3&quot; ...
##   ..$ df    : num 93.8
##   ..$ sigma : num 0.297
##   ..$ df_exp: NULL
##  $ WP_003041237:List of 5
##   ..$ beta  : num [1:38, 1] 2.67e+01 -6.17e-19 -1.44e-20 3.24e-19 -1.85e-18 ...
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:38] &quot;(Intercept)&quot; &quot;runX1WT_20_2h_n3_1&quot; &quot;runX1WT_20_2h_n3_2&quot; &quot;runX1WT_20_2h_n3_3&quot; ...
##   .. .. ..$ : NULL
##   ..$ vcov  : num [1:38, 1:38] 1.81 -5.42e-20 -6.07e-20 -5.08e-20 -3.70e-20 ...
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:38] &quot;(Intercept)&quot; &quot;runX1WT_20_2h_n3_1&quot; &quot;runX1WT_20_2h_n3_2&quot; &quot;runX1WT_20_2h_n3_3&quot; ...
##   .. .. ..$ : chr [1:38] &quot;(Intercept)&quot; &quot;runX1WT_20_2h_n3_1&quot; &quot;runX1WT_20_2h_n3_2&quot; &quot;runX1WT_20_2h_n3_3&quot; ...
##   ..$ df    : num 180
##   ..$ sigma : num 0.278
##   ..$ df_exp: NULL</code></pre>
</div>
<div id="save-your-results" class="section level3">
<h3>4.2 Save your results</h3>
<p>You can save your <code>proteinsFranc</code> and <code>protLMFranc</code> objects to a .RDatas object that can be loaded in R or in the Shiny graphical user interface. If you want your .RDatas object to be loaded into the Shiny App, you should also specify the <code>levelOptions</code>, <code>random</code> and <code>plot2DependentVars</code> objects. These objects are settings that are saved automatically when using the MSqRob Shiny App. In this example, I save the .RDatas object to my desktop.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>

result_files_new &lt;-<span class="st"> </span><span class="kw">list</span>()
result_files_new$proteins &lt;-<span class="st"> </span>proteinsFranc
result_files_new$models &lt;-<span class="st"> </span>protLMFranc

###This part is only needed if you want to load your .RDatas file in the MSqRob Shiny App###

<span class="co">#Should contain all contrast options for the fixed effects (see also 5. Statistical inference)</span>
result_files_new$levelOptions &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;genotypeWT&quot;</span>,<span class="st">&quot;genotypeKO&quot;</span>)
<span class="co">#Should contain all random effects</span>
result_files_new$random &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;biorep&quot;</span>,<span class="st">&quot;run&quot;</span>)
<span class="co">#Should contain a list with all fixed and random effects</span>
result_files_new$plot2DependentVars &lt;-<span class="st"> </span><span class="kw">as.list</span>(<span class="kw">c</span>(<span class="st">&quot;genotype&quot;</span>,<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;biorep&quot;</span>,<span class="st">&quot;run&quot;</span>))

#############################################################################################

<span class="kw">saves_MSqRob</span>(result_files, <span class="dt">file=</span><span class="st">&quot;/Users/lgoeminn/Desktop/project_Francisella.RDatas&quot;</span>) <span class="co">#Change file path to your desired save location!</span></code></pre></div>
<p>If you want to load your saved file, first specify the folder where it is saved.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
result_path &lt;-<span class="st"> &quot;/Users/lgoeminn/Desktop/project_Francisella.RDatas&quot;</span></code></pre></div>
<p>To inspect the different objects in the file, you can use the <code>inspect_loads_MSqRob</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
<span class="kw">inspect_loads_MSqRob</span>(result_path)</code></pre></div>
<p>To load this file and assign new <code>protdata</code> and <code>protLM</code> object to it, execute the following piece of code.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>

<span class="co">#To load the &quot;project_Francisella.RDatas&quot; file</span>
result_files &lt;-<span class="st"> </span><span class="kw">loads_MSqRob</span>(result_path)

<span class="co">#assign a new protdata object to the saved proteins</span>
proteinsFranc_result &lt;-<span class="st"> </span>result_files$proteins
<span class="co">#assign a new protLM object to the saved models</span>
protLMFranc_result &lt;-<span class="st"> </span>result_files$models</code></pre></div>
</div>
<div id="statistical-inference" class="section level3">
<h3>5. Statistical inference</h3>
<p>We are interested in which proteins differ significantly in abundance between ArgP mutant and wild type <em>Francisella tularensis</em>. Via the <code>&quot;MSqRob_levels&quot;</code> attribute, you can check which factor levels are present in each model. Pick a model for which there are no missing levels for the factors for which you want to test contrasts and inspect the names. Note that these names are always the combination of the parameter name and the factor level.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">attr</span>(modelWP_003040227,<span class="st">&quot;MSqRob_levels&quot;</span>)</code></pre></div>
<pre><code>##  [1] &quot;(Intercept)&quot;         &quot;genotypeKO&quot;          &quot;genotypeWT&quot;         
##  [4] &quot;runX1WT_20_2h_n4_1&quot;  &quot;runX1WT_20_2h_n4_2&quot;  &quot;runX1WT_20_2h_n4_3&quot; 
##  [7] &quot;runX1WT_20_2h_n5_1&quot;  &quot;runX1WT_20_2h_n5_2&quot;  &quot;runX1WT_20_2h_n5_3&quot; 
## [10] &quot;runX3D8_20_2h_n3_1&quot;  &quot;runX3D8_20_2h_n3_2&quot;  &quot;runX3D8_20_2h_n3_3&quot; 
## [13] &quot;runX3D8_20_2h_n4_1&quot;  &quot;runX3D8_20_2h_n4_2&quot;  &quot;runX3D8_20_2h_n4_3&quot; 
## [16] &quot;runX3D8_20_2h_n5_1&quot;  &quot;runX3D8_20_2h_n5_2&quot;  &quot;runX3D8_20_2h_n5_3&quot; 
## [19] &quot;biorepb_2&quot;           &quot;biorepb_3&quot;           &quot;biorepb_4&quot;          
## [22] &quot;biorepb_5&quot;           &quot;biorepb_6&quot;           &quot;SequenceMVFTFYNNLK&quot; 
## [25] &quot;SequenceYTYIYQIASFR&quot;</code></pre>
<p>Here, we notice that the mutant is encoded as <code>&quot;genotypeKO&quot;</code> and the wild type as <code>&quot;genotypeWT&quot;</code>. We construct a contrast matrix <code>L</code> for inferring on the fold change of interest. We are interested in the log2 fold change between mutant and wild type. As all intensities are log2-transformed, we should take the difference between <code>&quot;genotypeKO&quot;</code> and <code>&quot;genotypeWT&quot;</code> (because of the following property of logarithms: log2(a/b)=log2(a)-log2(b)).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">L &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="dt">contrasts=</span><span class="kw">c</span>(<span class="st">&quot;genotypeKO-genotypeWT&quot;</span>),
                  <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;genotypeWT&quot;</span>,<span class="st">&quot;genotypeKO&quot;</span>))
L</code></pre></div>
<pre><code>##             Contrasts
## Levels       genotypeKO-genotypeWT
##   genotypeWT                    -1
##   genotypeKO                     1</code></pre>
<p>Imagine we would want to test the average log2 protein intensity between mutant and wild type, then <code>L</code> would be constructed as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">L2 &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="dt">contrasts=</span><span class="kw">c</span>(<span class="st">&quot;genotypeKO/2+genotypeWT/2&quot;</span>),
                  <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;genotypeWT&quot;</span>,<span class="st">&quot;genotypeKO&quot;</span>))
L2</code></pre></div>
<pre><code>##             Contrasts
## Levels       genotypeKO/2+genotypeWT/2
##   genotypeWT                       0.5
##   genotypeKO                       0.5</code></pre>
<p><strong>Important notification:</strong> do not use the estimated parameters beta to determine the factor levels! You will notice for example that the level <code>&quot;genotypeWT&quot;</code> is missing in the column names of the betas. This is because <code>&quot;genotypeWT&quot;</code> is the reference class, giving the parameter corresponding to <code>&quot;genotypeKO&quot;</code> the interpretation of the effect of <code>&quot;genotypeKO&quot;</code> relative to <code>&quot;genotypeWT&quot;</code>. It is important however to include <code>&quot;genotypeWT&quot;</code> in the contrast matrix because for proteins where factor levels are missing, these parameters can get different interpretations. The <code>makeContrast</code> function takes this into account and will return <code>NA</code> when a contrast cannot be properly estimated.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">betaBVcovDf$beta</code></pre></div>
<pre><code>##                              [,1]
## (Intercept)          2.504493e+01
## runX1WT_20_2h_n4_1  -3.489353e-19
## runX1WT_20_2h_n4_2   6.742506e-19
## runX1WT_20_2h_n4_3  -5.735848e-19
## runX1WT_20_2h_n5_1  -2.700470e-19
## runX1WT_20_2h_n5_2   1.254289e-19
## runX1WT_20_2h_n5_3   2.537552e-20
## runX3D8_20_2h_n3_1   6.742506e-19
## runX3D8_20_2h_n3_2   4.211313e-20
## runX3D8_20_2h_n3_3  -8.681736e-20
## runX3D8_20_2h_n4_1   3.680183e-19
## runX3D8_20_2h_n4_2   1.424543e-19
## runX3D8_20_2h_n4_3  -3.815842e-19
## runX3D8_20_2h_n5_1  -3.520650e-19
## runX3D8_20_2h_n5_2  -1.057841e-19
## runX3D8_20_2h_n5_3   6.692662e-20
## biorepb_2           -2.829943e-01
## biorepb_3           -1.359206e-01
## biorepb_4            7.175990e-01
## biorepb_5            1.469156e-01
## biorepb_6           -4.455996e-01
## SequenceMVFTFYNNLK   6.567722e-19
## SequenceYTYIYQIASFR -6.567722e-19
## genotypeWT          -2.396076e-01</code></pre>
<p>Build a data frame containing estimate, standard error, degrees of freedom, T-value and p-value for each protein in the <code>protLM</code> object protLMFranc, by using the <code>test.protLMcontrast</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resultFranc &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(protLMFranc, L)</code></pre></div>
<p>The <code>prot.p.adjust</code> function adds a new column <code>&quot;qval&quot;</code> to the data frame based on an existing <code>&quot;pval&quot;</code> column. The new column contains adjusted p-values using one of the methods in the <code>p.adjust.methods</code> vector.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resultFranc &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(resultFranc, <span class="dt">method=</span><span class="st">&quot;fdr&quot;</span>)</code></pre></div>
<p>The <code>prot.signif</code> function needs a matrix (or a list of matrices) containing a column named “pval” and a column named <code>&quot;qval&quot;</code>. The <code>&quot;pval&quot;</code> column will be used to sort the matrix. The function generates a new column <code>&quot;signif&quot;</code> containing 1 for all values of <code>&quot;qval&quot;</code> with a value lower than the specified FDR level (default: 0.05) and 0 otherwise.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resultFranc &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(resultFranc, <span class="dt">level =</span> <span class="fl">0.05</span>)</code></pre></div>
<p>Inspect the first 6 significant proteins:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(resultFranc,<span class="dv">6</span>)</code></pre></div>
<pre><code>##                 GI.number
## WP_003040894 gi|118496676
## WP_003026016 gi|118497407
## WP_003020026 gi|118496882
## WP_003023392 gi|118497654
## WP_003038350 gi|118496950
## WP_003040849 gi|118496659
##                                                                                           Protein.names
## WP_003040894                       3-isopropylmalate dehydratase large subunit [Francisella tularensis]
## WP_003026016                                                   biotin synthase [Francisella tularensis]
## WP_003020026                                                    CTP synthetase [Francisella tularensis]
## WP_003023392 tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB [Francisella tularensis]
## WP_003038350                                              hypothetical protein [Francisella tularensis]
## WP_003040849                                              hypothetical protein [Francisella tularensis]
##                estimate         se        df      Tval         pval
## WP_003040894 -0.9057853 0.06028469 121.37912 -15.02513 1.797567e-29
## WP_003026016 -1.0157807 0.08991071 133.33958 -11.29766 3.523002e-21
## WP_003020026 -0.5111857 0.04763902 151.34723 -10.73040 2.491610e-20
## WP_003023392 -1.4006399 0.09257445  44.61587 -15.12988 3.497995e-19
## WP_003038350  0.8113071 0.06754524  72.83773  12.01131 6.037520e-19
## WP_003040849 -1.1209131 0.10669850 101.73682 -10.50543 6.381582e-18
##                      qval signif
## WP_003040894 2.070798e-26   TRUE
## WP_003026016 2.029249e-18   TRUE
## WP_003020026 9.567781e-18   TRUE
## WP_003023392 1.007422e-16   TRUE
## WP_003038350 1.391045e-16   TRUE
## WP_003040849 1.225264e-15   TRUE</code></pre>
</div>
<div id="advanced-importing-as-a-data-frame" class="section level3">
<h3>Advanced: importing as a data frame</h3>
<p>It is also possible to import your data as a simple data frame and make use of some custom preprocessing pipeline. Here, we will show the same preprocessing pipeline for the <em>Francisella</em> dataset as outlined above, but note that adding, removing or modifying steps in the preprocessing pipeline is now very simple as the data is in data frame format.</p>
<p>First, we import the <em>Francisella</em> dataset as a data frame via the base R function <code>read.table</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pepFrancDf &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file =</span> file_peptides_txt, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">&quot;&quot;</span>, <span class="dt">comment.char =</span> <span class="st">&quot;&quot;</span>)</code></pre></div>
<p>Of course, other filetypes can also be imported. Say for example your input file is a comma separated file stored in My Documents on a Windows computer, you could use for example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Change to eval=TRUE to execute this piece of code.</span>
pepFrancDf &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;C:/Users/Ludger/Documents/peptides.csv&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="dt">dec=</span><span class="st">&quot;.&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">&quot;&quot;</span>, <span class="dt">comment.char =</span> <span class="st">&quot;&quot;</span>)</code></pre></div>
<p>The data frame <code>pepFrancD</code> is in wide format.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(pepFrancDf)</code></pre></div>
<pre><code>##      GI.number                Sequence Amino.acid.before First.amino.acid
## 1 gi|118497196               AAAEELDTR                 K                A
## 2 gi|118498194           AAAGFVITASHNK                 R                A
## 3 gi|118497331 AAANEYELALAYSIEEVAPDLHK                 K                A
## 4 gi|118496879            AAANNPQLEAFK                 K                A
## 5 gi|118497152             AAASAGLVDEK                 K                A
## 6 gi|118497492            AAASLDLYSYPK                 K                A
##   Second.amino.acid Second.last.amino.acid Last.amino.acid
## 1                 A                      T               R
## 2                 A                      N               K
## 3                 A                      H               K
## 4                 A                      F               K
## 5                 A                      E               K
## 6                 A                      P               K
##   Amino.acid.after A.Count R.Count N.Count D.Count C.Count Q.Count E.Count
## 1                K       3       1       0       1       0       0       2
## 2                F       4       0       1       0       0       0       0
## 3                Y       6       0       1       1       0       0       4
## 4                K       4       0       2       0       0       1       1
## 5                A       4       0       0       1       0       0       1
## 6                V       3       0       0       1       0       0       0
##   G.Count H.Count I.Count L.Count K.Count M.Count F.Count P.Count S.Count
## 1       0       0       0       1       0       0       0       0       0
## 2       1       1       1       0       1       0       1       0       1
## 3       0       1       1       3       1       0       0       1       1
## 4       0       0       0       1       1       0       1       1       0
## 5       1       0       0       1       1       0       0       0       1
## 6       0       0       0       2       1       0       0       1       2
##   T.Count W.Count Y.Count V.Count U.Count Length Missed.cleavages
## 1       1       0       0       0       0      9                0
## 2       1       0       0       1       0     13                0
## 3       0       0       2       1       0     23                0
## 4       0       0       0       0       0     12                0
## 5       0       0       0       1       0     11                0
## 6       0       0       2       0       0     12                0
##         Mass     Proteins
## 1 97.446.689 WP_003038655
## 2 12.856.779 WP_003041237
## 3 25.162.435 WP_003038915
## 4 12.726.463 WP_003038264
## 5 10.305.295 WP_003035781
## 6 12.976.554 WP_003039212
##                                                    Protein.names
## 1                  hypothetical protein [Francisella tularensis]
## 2             phosphoglucosamine mutase [Francisella tularensis]
## 3     glycine--tRNA ligase subunit beta [Francisella tularensis]
## 4              molecular chaperone HtpG [Francisella tularensis]
## 5 delta-aminolevulinic acid dehydratase [Francisella tularensis]
## 6                         phosphorylase [Francisella tularensis]
##   Leading.razor.protein Unique..Groups. Unique..Proteins. Charges
## 1          gi|118497196             yes               yes     1,2
## 2          gi|118498194             yes               yes       2
## 3          gi|118497331             yes               yes       3
## 4          gi|118496879             yes               yes       2
## 5          gi|118497152             yes               yes       2
## 6          gi|118497492             yes               yes       2
##          PEP  Score Experiment.1WT_20_2h_n3_1 Experiment.1WT_20_2h_n3_2
## 1 0.00012531  59.35                        NA                        NA
## 2   5,30E-02 45.825                        NA                         1
## 3   1,05E-41 84.327                         1                         1
## 4   8,75E-20 108.56                         1                         1
## 5   9,51E-01 67.385                         1                        NA
## 6   0.012563 31.675                        NA                         1
##   Experiment.1WT_20_2h_n3_3 Experiment.1WT_20_2h_n4_1
## 1                         1                         1
## 2                        NA                        NA
## 3                         2                         1
## 4                         1                         1
## 5                         1                         1
## 6                         1                        NA
##   Experiment.1WT_20_2h_n4_2 Experiment.1WT_20_2h_n4_3
## 1                        NA                         1
## 2                        NA                        NA
## 3                         1                         1
## 4                         1                         1
## 5                         1                        NA
## 6                         1                         1
##   Experiment.1WT_20_2h_n5_1 Experiment.1WT_20_2h_n5_2
## 1                        NA                         1
## 2                        NA                         1
## 3                         2                         2
## 4                         1                         1
## 5                         1                         1
## 6                        NA                        NA
##   Experiment.1WT_20_2h_n5_3 Experiment.3D8_20_2h_n3_1
## 1                        NA                        NA
## 2                         1                         1
## 3                         1                         1
## 4                         1                         1
## 5                         1                         1
## 6                        NA                        NA
##   Experiment.3D8_20_2h_n3_2 Experiment.3D8_20_2h_n3_3
## 1                         1                        NA
## 2                         1                        NA
## 3                         2                         1
## 4                         1                         1
## 5                         1                         1
## 6                        NA                        NA
##   Experiment.3D8_20_2h_n4_1 Experiment.3D8_20_2h_n4_2
## 1                        NA                        NA
## 2                         1                         1
## 3                         1                         1
## 4                         1                         1
## 5                         1                         1
## 6                        NA                         1
##   Experiment.3D8_20_2h_n4_3 Experiment.3D8_20_2h_n5_1
## 1                        NA                        NA
## 2                         1                         1
## 3                         1                        NA
## 4                         1                         1
## 5                        NA                         1
## 6                        NA                        NA
##   Experiment.3D8_20_2h_n5_2 Experiment.3D8_20_2h_n5_3   Intensity
## 1                        NA                        NA  1329500000
## 2                        NA                        NA   167190000
## 3                         1                         1   556750000
## 4                         1                         1 14830000000
## 5                         1                         1  4820900000
## 6                        NA                        NA   137150000
##   Intensity.1WT_20_2h_n3_1 Intensity.1WT_20_2h_n3_2
## 1                        0                        0
## 2                        0                 20679000
## 3                 28853000                 28091000
## 4                339830000                420390000
## 5                240280000                        0
## 6                        0                 31069000
##   Intensity.1WT_20_2h_n3_3 Intensity.1WT_20_2h_n4_1
## 1                  9290400                 92996000
## 2                        0                        0
## 3                 30436000                 19476000
## 4                393930000                235060000
## 5                241890000                222250000
## 6                 31448000                        0
##   Intensity.1WT_20_2h_n4_2 Intensity.1WT_20_2h_n4_3
## 1                        0                 98059000
## 2                        0                        0
## 3                 22050000                 20687000
## 4                381090000                360270000
## 5                189930000                        0
## 6                 27721000                 28657000
##   Intensity.1WT_20_2h_n5_1 Intensity.1WT_20_2h_n5_2
## 1                        0                 80803000
## 2                        0                 17358000
## 3                 36588000                 28654000
## 4                255710000                311370000
## 5                160440000                147230000
## 6                        0                        0
##   Intensity.1WT_20_2h_n5_3 Intensity.3D8_20_2h_n3_1
## 1                        0                        0
## 2                 13841000                 12278000
## 3                  7009100                 14174000
## 4                271440000                225140000
## 5                143340000                135910000
## 6                        0                        0
##   Intensity.3D8_20_2h_n3_2 Intensity.3D8_20_2h_n3_3
## 1                 95433000                        0
## 2                 10712000                        0
## 3                  8548000                  7859200
## 4                289880000                282500000
## 5                185740000                168620000
## 6                        0                        0
##   Intensity.3D8_20_2h_n4_1 Intensity.3D8_20_2h_n4_2
## 1                        0                        0
## 2                  9500700                 11111000
## 3                 14871000                 11690000
## 4                209490000                223610000
## 5                160240000                158000000
## 6                        0                 18253000
##   Intensity.3D8_20_2h_n4_3 Intensity.3D8_20_2h_n5_1
## 1                        0                        0
## 2                  8723500                  8125200
## 3                  4470500                        0
## 4                219700000                169120000
## 5                        0                 92542000
## 6                        0                        0
##   Intensity.3D8_20_2h_n5_2 Intensity.3D8_20_2h_n5_3 Reverse Contaminant id
## 1                        0                        0                      0
## 2                        0                        0                      1
## 3                  4048100                  3422300                      2
## 4                178150000                170400000                      3
## 5                102350000                100090000                      4
## 6                        0                        0                      5
##   Protein.group.IDs Mod..peptide.IDs
## 1               419                0
## 2              1150                1
## 3               513                2
## 4               199                3
## 5               388                4
## 6               624                5
##                                                                                                                                                                 Evidence.IDs
## 1                                                                                                                           &quot;0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18&quot;
## 2                                                                                                                             &quot;19;20;21;22;23;24;25;26;27;28;29;30;31;32;33&quot;
## 3                                         &quot;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76&quot;
## 4 &quot;77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124&quot;
## 5                                              &quot;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155&quot;
## 6                                                                                                                                                      &quot;156;157;158;159;160&quot;
##                                                                                                                                                                                                                                                                                                                                                           MS.MS.IDs
## 1                                                                                                                                                                                                                                                                                                                  &quot;0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18&quot;
## 2                                                                                                                                                                                                                                                                                                                    &quot;19;20;21;22;23;24;25;26;27;28;29;30;31;32;33&quot;
## 3                                                                                                                                                                                                                       &quot;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79&quot;
## 4 &quot;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172&quot;
## 5                                                                                                                                                                                                                                     &quot;173;174;175;176;177;178;179;180;181;182;183;184;185;186;187;188;189;190;191;192;193;194;195;196;197;198;199;200;201;202;203&quot;
## 6                                                                                                                                                                                                                                                                                                                                             &quot;204;205;206;207;208&quot;
##   Best.MS.MS Oxidation..M..site.IDs
## 1         15                       
## 2         26                       
## 3         39                       
## 4        159                       
## 5        191                       
## 6        205</code></pre>
<p>If we want to mimick the MaxQuant preprocessing pipeline, we need to add a factor that indicates whether a protein is only identified by modified peptides. This requires some basic R programming. The information about which proteins are only identified by modified peptides can be found in the proteinGroups.txt file. We again make use of the proteinGroups.txt file that is delivered with the package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Import the proteinGroups.txt file delivered with the MSqRobData package </span>
file_proteinGroups &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/Francisella&quot;</span>, <span class="st">&quot;proteinGroups.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MSqRobData&quot;</span>)

  proteinGroups &lt;-<span class="st"> </span><span class="kw">read.table</span>(file_proteinGroups, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">&quot;&quot;</span>, <span class="dt">comment.char =</span> <span class="st">&quot;&quot;</span>)
  
  <span class="co">#Extract the column that indicates which proteins are only identified by a modification site</span>
  only_site &lt;-<span class="st"> </span>proteinGroups$Only.identified.by.site
  
  <span class="co">#If there are no such proteins (as is the case here), the column will be completely empty and R will import this by default as &quot;NA&quot;. However, we want the an empty value instead of &quot;NA&quot; to keep this column consistent with the &quot;Contaminant&quot; and &quot;Reverse&quot; columns.</span>
  only_site[<span class="kw">is.na</span>(only_site)] &lt;-<span class="st"> &quot;&quot;</span>
  <span class="co">#Select the protein accessions that are only identified by one or more modified peptides</span>
  filter_symbol &lt;-<span class="st"> &quot;+&quot;</span>
  removed_proteins &lt;-<span class="st"> </span>proteinGroups$Protein.IDs[only_site==filter_symbol]

  <span class="co">#Create a logical variable &quot;rem&quot; that holds &quot;TRUE&quot; if a row in the pepFrancDf data frame should be removed and FALSE otherwise.</span>
  accession &lt;-<span class="st"> &quot;Proteins&quot;</span>
  rem &lt;-<span class="st"> </span><span class="kw">as.character</span>(pepFrancDf[,accession]) %in%<span class="st"> </span><span class="kw">as.character</span>(removed_proteins)
  
  <span class="co">#Create a new column &quot;only_site&quot; in &quot;pepFrancDf&quot; that indicates with a &quot;+&quot; which rows should be removed.</span>
  pepFrancDf$only_site &lt;-<span class="st"> &quot;&quot;</span>
  pepFrancDf$only_site[rem] &lt;-<span class="st"> &quot;+&quot;</span></code></pre></div>
<p>Note that in this particular case, the previous chunck of code was superfluous, as there are no proteins in the dataset that are only identified by modified peptides.</p>
<p>We can easily preprocess the data frame using the <code>preprocess_wide</code> function (for data in “long” format, use the <code>preprocess_long</code> function). This function uses our standard preprocessing pipeline on data frames in “wide” format (i.e. the data frame has one observation row per subject with each measurement present as a different variable).</p>
<p>We already made the <code>exp_annotation</code> object before (see “2.1. Create an experimental annotation data frame.”), so we will not repeat this code again here. <strong>Importantly, however, contrary to the <code>exprs</code> slot in the <code>peptidesFranc2</code> object created before, the names of all intensity columns in data frame <code>pepFrancDf</code> start with “Intensity.”.</strong></p>
<p><strong>This is because the default setting <code>remove_pattern=TRUE</code> in the <code>read_MaxQuant</code> function removes the prefix “Intensity ” from the columns containing the intensities as this allows users that are unfamiliar with R to just copy and paste the names they provided in MaxQuant’s “Experiment” column as an input. Here, we just imported the peptides.txt file as it is.</strong> We thus create a new experimental annotation data frame called <code>exp_annotation2</code> with run names starting with “Intensity.” (a space is not syntactically valid in R, so it is automatically converted to a dot).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">runs2 &lt;-<span class="st"> </span><span class="kw">colnames</span>(pepFrancDf)[<span class="kw">grepl</span>(<span class="st">&quot;Intensity.&quot;</span>,<span class="kw">colnames</span>(pepFrancDf))]
runs2</code></pre></div>
<pre><code>##  [1] &quot;Intensity.1WT_20_2h_n3_1&quot; &quot;Intensity.1WT_20_2h_n3_2&quot;
##  [3] &quot;Intensity.1WT_20_2h_n3_3&quot; &quot;Intensity.1WT_20_2h_n4_1&quot;
##  [5] &quot;Intensity.1WT_20_2h_n4_2&quot; &quot;Intensity.1WT_20_2h_n4_3&quot;
##  [7] &quot;Intensity.1WT_20_2h_n5_1&quot; &quot;Intensity.1WT_20_2h_n5_2&quot;
##  [9] &quot;Intensity.1WT_20_2h_n5_3&quot; &quot;Intensity.3D8_20_2h_n3_1&quot;
## [11] &quot;Intensity.3D8_20_2h_n3_2&quot; &quot;Intensity.3D8_20_2h_n3_3&quot;
## [13] &quot;Intensity.3D8_20_2h_n4_1&quot; &quot;Intensity.3D8_20_2h_n4_2&quot;
## [15] &quot;Intensity.3D8_20_2h_n4_3&quot; &quot;Intensity.3D8_20_2h_n5_1&quot;
## [17] &quot;Intensity.3D8_20_2h_n5_2&quot; &quot;Intensity.3D8_20_2h_n5_3&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Extract the relevant part of the mass spec run names that points to the genotype.</span>
genotype2 &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">substr</span>(runs2, <span class="dv">11</span>,<span class="dv">13</span>))
genotype2</code></pre></div>
<pre><code>##  [1] 1WT 1WT 1WT 1WT 1WT 1WT 1WT 1WT 1WT 3D8 3D8 3D8 3D8 3D8 3D8 3D8 3D8
## [18] 3D8
## Levels: 1WT 3D8</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">n &lt;-<span class="st"> </span><span class="kw">nchar</span>(runs2)
biorep2 &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">paste0</span>(<span class="st">&quot;b_&quot;</span>,<span class="kw">factor</span>(<span class="kw">as.numeric</span>(<span class="kw">factor</span>(<span class="kw">paste0</span>(<span class="kw">substr</span>(runs2, <span class="dv">10</span>,<span class="dv">10</span>),<span class="kw">substr</span>(runs2, n<span class="dv">-2</span>,n<span class="dv">-2</span>))))+<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>),<span class="kw">rep</span>(<span class="dv">3</span>,<span class="dv">9</span>)))))
biorep</code></pre></div>
<pre><code>##  [1] b_1 b_1 b_1 b_2 b_2 b_2 b_3 b_3 b_3 b_4 b_4 b_4 b_5 b_5 b_5 b_6 b_6
## [18] b_6
## Levels: b_1 b_2 b_3 b_4 b_5 b_6</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">exp_annotation2 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">run=</span>runs2, <span class="dt">genotype=</span>genotype2, <span class="dt">biorep=</span>biorep2)
exp_annotation2</code></pre></div>
<pre><code>##                         run genotype biorep
## 1  Intensity.1WT_20_2h_n3_1      1WT    b_1
## 2  Intensity.1WT_20_2h_n3_2      1WT    b_1
## 3  Intensity.1WT_20_2h_n3_3      1WT    b_1
## 4  Intensity.1WT_20_2h_n4_1      1WT    b_2
## 5  Intensity.1WT_20_2h_n4_2      1WT    b_2
## 6  Intensity.1WT_20_2h_n4_3      1WT    b_2
## 7  Intensity.1WT_20_2h_n5_1      1WT    b_3
## 8  Intensity.1WT_20_2h_n5_2      1WT    b_3
## 9  Intensity.1WT_20_2h_n5_3      1WT    b_3
## 10 Intensity.3D8_20_2h_n3_1      3D8    b_4
## 11 Intensity.3D8_20_2h_n3_2      3D8    b_4
## 12 Intensity.3D8_20_2h_n3_3      3D8    b_4
## 13 Intensity.3D8_20_2h_n4_1      3D8    b_5
## 14 Intensity.3D8_20_2h_n4_2      3D8    b_5
## 15 Intensity.3D8_20_2h_n4_3      3D8    b_5
## 16 Intensity.3D8_20_2h_n5_1      3D8    b_6
## 17 Intensity.3D8_20_2h_n5_2      3D8    b_6
## 18 Intensity.3D8_20_2h_n5_3      3D8    b_6</code></pre>
<p>The <code>preprocess_wide</code> function allows for the same preprocessing as the <code>preprocess_MaxQuant</code> function on a regular data frame. If you inspect this function, you will notice that the same 8 preprocessing steps are executed. Remember that you can do any preprocessing you like using basic R data frame manipulations. The <code>split</code> argument indicates which character string is used to separate the accession groups. The <code>exp_annotation</code> argument, like before, contains the experimental annotation data frame or a character string indicating the filepath where the experimental annotation is saved as an Excel file or a tab-delimited file. The <code>quant_cols</code> argument contains a vector of names or a vector of numeric indices pointing to the columns containing the quantitative values of interest (mostly intensities or areas under the curve). If <code>quant_cols</code> contains only one character string, each column containing this pattern will be regarded as a column containing the quantitative values of interest. <code>aggr_by</code> indicates the column by which the data should be aggregated. Here, the data is already aggregated to the peptide level (i.e. over different charge states and modification statuses). We need no further aggregation, thus we provide the “Sequence” column (the level to which aggregation has already been done). <code>aggr_function</code> will be superfluous here, but if further aggregation would have been necessary, raw intensities would be summed up (<code>aggr_function=&quot;sum&quot;</code>). The other parameters are similar to the <code>preprocess_MaxQuant</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pepFrancDf2 &lt;-<span class="st"> </span><span class="kw">preprocess_wide</span>(pepFrancDf, <span class="dt">accession=</span><span class="st">&quot;Proteins&quot;</span>, <span class="dt">split=</span><span class="st">&quot;;&quot;</span>, <span class="dt">exp_annotation=</span>exp_annotation2, <span class="dt">quant_cols=</span><span class="st">&quot;Intensity.&quot;</span>, <span class="dt">aggr_by=</span><span class="st">&quot;Sequence&quot;</span>, <span class="dt">aggr_function=</span><span class="st">&quot;sum&quot;</span>, <span class="dt">logtransform=</span><span class="ot">TRUE</span>, <span class="dt">base=</span><span class="dv">2</span>, <span class="dt">normalisation=</span><span class="st">&quot;quantiles&quot;</span>, <span class="dt">smallestUniqueGroups=</span><span class="ot">TRUE</span>, <span class="dt">useful_properties=</span><span class="kw">c</span>(<span class="st">&quot;GI.number&quot;</span>,<span class="st">&quot;Protein.names&quot;</span>,<span class="st">&quot;Sequence&quot;</span>), <span class="dt">filter=</span><span class="kw">c</span>(<span class="st">&quot;Contaminant&quot;</span>,<span class="st">&quot;Reverse&quot;</span>,<span class="st">&quot;only_site&quot;</span>), <span class="dt">filter_symbol=</span><span class="st">&quot;+&quot;</span>, <span class="dt">minIdentified=</span><span class="dv">2</span>)</code></pre></div>
<p>Note that the experimental annotation is now present as an attribute of the <code>pepFrancDf2</code> data frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">attr</span>(pepFrancDf2,<span class="st">&quot;MSqRob_exp_annotation&quot;</span>)</code></pre></div>
<pre><code>##                         run genotype biorep
## 1  Intensity.1WT_20_2h_n3_1      1WT    b_1
## 2  Intensity.1WT_20_2h_n3_2      1WT    b_1
## 3  Intensity.1WT_20_2h_n3_3      1WT    b_1
## 4  Intensity.1WT_20_2h_n4_1      1WT    b_2
## 5  Intensity.1WT_20_2h_n4_2      1WT    b_2
## 6  Intensity.1WT_20_2h_n4_3      1WT    b_2
## 7  Intensity.1WT_20_2h_n5_1      1WT    b_3
## 8  Intensity.1WT_20_2h_n5_2      1WT    b_3
## 9  Intensity.1WT_20_2h_n5_3      1WT    b_3
## 10 Intensity.3D8_20_2h_n3_1      3D8    b_4
## 11 Intensity.3D8_20_2h_n3_2      3D8    b_4
## 12 Intensity.3D8_20_2h_n3_3      3D8    b_4
## 13 Intensity.3D8_20_2h_n4_1      3D8    b_5
## 14 Intensity.3D8_20_2h_n4_2      3D8    b_5
## 15 Intensity.3D8_20_2h_n4_3      3D8    b_5
## 16 Intensity.3D8_20_2h_n5_1      3D8    b_6
## 17 Intensity.3D8_20_2h_n5_2      3D8    b_6
## 18 Intensity.3D8_20_2h_n5_3      3D8    b_6</code></pre>
<p>After custom preprocessing, you can transform the data to a <code>protdata</code> object via the <code>df2protdata</code> function.</p>
<p>The <code>acc_col</code> argument in the <code>df2protdata</code> function indicates the column index or column name of the accessions (i.e. the protein (group) identifiers). The <code>quant_cols</code> argument is a vector of indices pointing to all columns containing peptide intensities. In the example below, the column named “Proteins” will be added to the <code>accession</code> slot while all columns with indices in the <code>quant_cols</code> vector will be added as a matrix to the <code>intensities</code> slot. <code>quant_name</code> indicates the name that will be given to the column containing the (preprocessed) intensities and <code>run_name</code> indicates the name that will be given to the column containing the mass spec run names in the new <code>protdata</code> object. Like before, we keep “GI.number” and “Protein.names” as annotations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Select columns with intensities.</span>
quant_cols &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">grepl</span>(<span class="st">&quot;Intensity.&quot;</span>, <span class="kw">colnames</span>(pepFrancDf2)))
quant_cols</code></pre></div>
<pre><code>##  [1]  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Call df2protdata function</span>
proteinsFrancDf &lt;-<span class="st"> </span><span class="kw">df2protdata</span>(pepFrancDf2, <span class="dt">acc_col=</span><span class="st">&quot;Proteins&quot;</span>, <span class="dt">quant_cols=</span>quant_cols, <span class="dt">quant_name=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">run_name=</span><span class="st">&quot;run&quot;</span>, <span class="dt">annotations=</span><span class="kw">c</span>(<span class="st">&quot;GI.number&quot;</span>,<span class="st">&quot;Protein.names&quot;</span>))
<span class="co">#Inspect proteinsFrancDf protdata object.</span>
proteinsFrancDf</code></pre></div>
<pre><code>## @accession 
## 
## [1] &quot;WP_003038655&quot;
## 
##  @annotation 
## 
##                                       GI.number 
##                                  &quot;gi|118497196&quot; 
##                                   Protein.names 
## &quot;hypothetical protein [Francisella tularensis]&quot; 
## 
##  @data 
## 
##   quant_value           Sequence genotype biorep                      run
## 2    25.11978       AVVVFFDYGCGK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 3    25.34586 DLLSDPATPTVGPQDANK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 4    24.11491   ETNGELTVQDVDNVVK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 5    26.50674          IIAIPDIVK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 6    24.69046     QTNVITDNYQQAAK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 9    24.75196       AVVVFFDYGCGK      1WT    b_1 Intensity.1WT_20_2h_n3_2
## 
##  96 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003041237&quot;
## 
##  @annotation 
## 
##                                            GI.number 
##                                       &quot;gi|118498194&quot; 
##                                        Protein.names 
## &quot;phosphoglucosamine mutase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##   quant_value         Sequence genotype biorep                      run
## 2    25.81732    ADLGISLDGDADR      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 3    27.36716        FVIVGQDTR      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 4    24.91571  GEVANSTITAEFTQK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 5    27.38539 ILANAIDEYIESIHSR      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 7    27.93398    SLATNEAEYLVEK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 8    26.59163       VASDVNDVEK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 
##  187 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003038915&quot;
## 
##  @annotation 
## 
##                                                    GI.number 
##                                               &quot;gi|118497331&quot; 
##                                                Protein.names 
## &quot;glycine--tRNA ligase subunit beta [Francisella tularensis]&quot; 
## 
##  @data 
## 
##   quant_value                Sequence genotype biorep
## 1    24.37301 AAANEYELALAYSIEEVAPDLHK      1WT    b_1
## 4    22.60222   EEGIAVDIFEAINNTNYDSIK      1WT    b_1
## 5    25.34002           EGEPTQVGLGFAK      1WT    b_1
## 6    25.13163          EYSYALELLTCLDK      1WT    b_1
## 7    24.09977       FHHPQAIVIDNISDYIK      1WT    b_1
## 9    25.75712              IIDITLESYK      1WT    b_1
##                        run
## 1 Intensity.1WT_20_2h_n3_1
## 4 Intensity.1WT_20_2h_n3_1
## 5 Intensity.1WT_20_2h_n3_1
## 6 Intensity.1WT_20_2h_n3_1
## 7 Intensity.1WT_20_2h_n3_1
## 9 Intensity.1WT_20_2h_n3_1
## 
##  215 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003038264&quot;
## 
##  @annotation 
## 
##                                           GI.number 
##                                      &quot;gi|118496879&quot; 
##                                       Protein.names 
## &quot;molecular chaperone HtpG [Francisella tularensis]&quot; 
## 
##  @data 
## 
##   quant_value        Sequence genotype biorep                      run
## 1    27.92076    AAANNPQLEAFK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 3    26.48741 EGQDTIYYITSDSYK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 5    27.33299    EHLDLLEYHVLK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 6    26.37656         EILQHNK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 7    28.11488        ELLPPYLR      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 8    28.77850    ELVSNSSDAIEK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 
##  338 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003035781&quot;
## 
##  @annotation 
## 
##                                                        GI.number 
##                                                   &quot;gi|118497152&quot; 
##                                                    Protein.names 
## &quot;delta-aminolevulinic acid dehydratase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##   quant_value        Sequence genotype biorep                      run
## 1    27.47327     AAASAGLVDEK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 2    24.05872 VILTYTALDVANWLK      1WT    b_1 Intensity.1WT_20_2h_n3_1
## 5    23.88890 VILTYTALDVANWLK      1WT    b_1 Intensity.1WT_20_2h_n3_2
## 6    25.37749      YASAYYGPFR      1WT    b_1 Intensity.1WT_20_2h_n3_2
## 7    27.47269     AAASAGLVDEK      1WT    b_1 Intensity.1WT_20_2h_n3_3
## 8    24.93174 VILTYTALDVANWLK      1WT    b_1 Intensity.1WT_20_2h_n3_3
## 
##  41 more rows... 
## 
## @accession 
## 
## [1] &quot;WP_003039212&quot;
## 
##  @annotation 
## 
##                                GI.number 
##                           &quot;gi|118497492&quot; 
##                            Protein.names 
## &quot;phosphorylase [Francisella tularensis]&quot; 
## 
##  @data 
## 
##    quant_value             Sequence genotype biorep
## 3     22.96510 FPAGLNDVEFVAEHIFQHSK      1WT    b_1
## 4     25.10257           LQDLLTFIGK      1WT    b_1
## 7     24.55598             VVFSVDYR      1WT    b_1
## 8     24.53369         AAASLDLYSYPK      1WT    b_1
## 9     24.68029              EVLFLNR      1WT    b_1
## 10    23.16038 FPAGLNDVEFVAEHIFQHSK      1WT    b_1
##                         run
## 3  Intensity.1WT_20_2h_n3_1
## 4  Intensity.1WT_20_2h_n3_1
## 7  Intensity.1WT_20_2h_n3_1
## 8  Intensity.1WT_20_2h_n3_2
## 9  Intensity.1WT_20_2h_n3_2
## 10 Intensity.1WT_20_2h_n3_2
## 
##  56 more rows... 
## 
## 1183 more elements...</code></pre>
<p>Note that the <code>proteinsFrancDf</code> object is the same as the <code>proteinsFranc</code> object, only with a different order of columns in their <code>data</code> slot.</p>
</div>
</div>
<div id="cptac-dataset-randomized-complete-block-design" class="section level2">
<h2>CPTAC dataset (randomized complete block design)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(MSqRob)
<span class="kw">library</span>(Biobase)</code></pre></div>
<p>This example involves the CPTAC Study 6. In this dataset Sigma’s UPS1 standard containing 48 human proteins was spiked into a reference yeast proteome and analyzed on 7 instruments in 5 different laboratories (Paulovich et al., 2010). Raw data files can be downloaded at: <a href="https://cptac-data-portal.georgetown.edu/cptac/dataPublic/list?currentPath=%2FPhase_I_Data%2FStudy6&amp;nonav=true">https://cptac-data-portal.georgetown.edu/cptac/dataPublic/list?currentPath=%2FPhase_I_Data%2FStudy6&amp;nonav=true</a></p>
<p>Here, we limited ourselves to the data of LTQ-Orbitrap at site 86, LTQ-Orbitrap O at site 65 and LTQ-Orbitrap W at site 56. The data was searched with MaxQuant version 1.5.2.8. Detailed search settings can be found in Goeminne et al. (2016).</p>
<div id="import-maxquants-peptides.txt-file" class="section level3">
<h3>1. Import MaxQuant’s peptides.txt file</h3>
<p>First, specify the location of the peptides.txt MaxQuant output file. By default, MaxQuant creates a “combined” folder in the same folder as the raw files are (or if the raw files are in different directories, in the directory of the first raw file). The peptides.txt file can then be found in “path_to_raw_files/combined/txt/peptides.txt”. Here, we make use of the peptides.txt file that is delivered with the package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file_peptides_txt &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/CPTAC&quot;</span>, <span class="st">&quot;peptides.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MSqRobData&quot;</span>)</code></pre></div>
<p>Import MaxQuant’s peptides.txt file and convert it to an <code>MSnSet</code> object (see package <code>MSnbase</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peptidesCPTAC &lt;-<span class="st"> </span><span class="kw">read_MaxQuant</span>(file_peptides_txt, <span class="dt">pattern=</span><span class="st">&quot;Intensity &quot;</span>, <span class="dt">remove_pattern=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>The “pattern” argument indicates which columns in the peptides.txt file contain peptide intensities. Currently, all columns containing “Intensity ” in their column names will be treated as peptide intensities (default for MaxQuant). <code>remove_pattern=TRUE</code> (default) indicates that the name of the pattern is removed from the column names.</p>
</div>
<div id="preprocessing-1" class="section level3">
<h3>2. Preprocessing</h3>
</div>
<div id="add-information-on-the-design-create-an-experimental-annotation-data-frame." class="section level3">
<h3>2.1. Add information on the design: create an experimental annotation data frame.</h3>
<p>Extract the names of the mass spec runs via the <code>sampleNames</code> function from the <code>Biobase</code> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">runs &lt;-<span class="st"> </span><span class="kw">sampleNames</span>(peptidesCPTAC)</code></pre></div>
<p>Note that as the names again start with a number, an “X” is prepended to make the names syntactically valid. Add a grouping factor for the spike-in condition (i.e. 6A, 6B, 6C, 6D and 6E).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">condition &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">substr</span>(runs, <span class="dv">2</span>,<span class="dv">3</span>))</code></pre></div>
<!--
Add a grouping factor for each of the 9 technical repeats (i.e. instrument x run).

```r
rep <- factor(substr(runs, 5,5))
```
-->
<p>Add a grouping factor for the lab effect (i.e. LTQ-Orbitrap at site 86, LTQ-Orbitrap O at site 65 and LTQ-Orbitrap W at site 56).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lab &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">rep</span>(<span class="kw">rep</span>(<span class="dv">1</span>:<span class="dv">3</span>,<span class="dt">each=</span><span class="dv">3</span>),<span class="dv">5</span>))
<span class="co">#Alternative:</span>
<span class="co">#lab &lt;- factor((as.numeric(substr(runs, 5,5))-1)%/%3+1)</span></code></pre></div>
<p>Create an experimental annotation data frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">exp_annotation &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">run=</span>runs, <span class="dt">condition=</span>condition, <span class="dt">lab=</span>lab)
exp_annotation</code></pre></div>
<pre><code>##      run condition lab
## 1  X6A_1        6A   1
## 2  X6A_2        6A   1
## 3  X6A_3        6A   1
## 4  X6A_4        6A   2
## 5  X6A_5        6A   2
## 6  X6A_6        6A   2
## 7  X6A_7        6A   3
## 8  X6A_8        6A   3
## 9  X6A_9        6A   3
## 10 X6B_1        6B   1
## 11 X6B_2        6B   1
## 12 X6B_3        6B   1
## 13 X6B_4        6B   2
## 14 X6B_5        6B   2
## 15 X6B_6        6B   2
## 16 X6B_7        6B   3
## 17 X6B_8        6B   3
## 18 X6B_9        6B   3
## 19 X6C_1        6C   1
## 20 X6C_2        6C   1
## 21 X6C_3        6C   1
## 22 X6C_4        6C   2
## 23 X6C_5        6C   2
## 24 X6C_6        6C   2
## 25 X6C_7        6C   3
## 26 X6C_8        6C   3
## 27 X6C_9        6C   3
## 28 X6D_1        6D   1
## 29 X6D_2        6D   1
## 30 X6D_3        6D   1
## 31 X6D_4        6D   2
## 32 X6D_5        6D   2
## 33 X6D_6        6D   2
## 34 X6D_7        6D   3
## 35 X6D_8        6D   3
## 36 X6D_9        6D   3
## 37 X6E_1        6E   1
## 38 X6E_2        6E   1
## 39 X6E_3        6E   1
## 40 X6E_4        6E   2
## 41 X6E_5        6E   2
## 42 X6E_6        6E   2
## 43 X6E_7        6E   3
## 44 X6E_8        6E   3
## 45 X6E_9        6E   3</code></pre>
<p>Remember that the <code>exp_annotation</code> can also be given as an Excel file or a tab delimited file!</p>
</div>
<div id="preprocess-the-data-using-the-preprocess_maxquant-function.-1" class="section level3">
<h3>2.2. Preprocess the data using the preprocess_MaxQuant function.</h3>
<p>If we want to remove proteins that are only identified by modified peptides from the dataset, we also need the proteinGroups.txt file. This file can be found in the same location as the peptides.txt file (“path_to_raw_files/combined/txt/proteinGroups.txt”). Here, we make use of the proteinGroups.txt file that is delivered with the package.</p>
<p>Note: if you don’t want to remove proteins that are only identified by modified peptides, set <code>remove_only_site=FALSE</code> and leave the <code>file_proteinGroups</code> argument at its default value (<code>NULL</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file_proteinGroups &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/CPTAC&quot;</span>, <span class="st">&quot;proteinGroups.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MSqRobData&quot;</span>)</code></pre></div>
<p><strong>Important:</strong> in the CPTAC dataset, some human UPS proteins are NOT contaminants as these proteins were spiked in on purpose! We only want to remove those contaminants that are not human UPS proteins. We thus need to unmark these proteins as contaminant before preprocessing.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fData</span>(peptidesCPTAC)[<span class="kw">grepl</span>(<span class="st">&quot;_HUMAN_UPS&quot;</span>, <span class="kw">fData</span>(peptidesCPTAC)$Proteins),]$Potential.contaminant &lt;-<span class="st"> &quot;&quot;</span></code></pre></div>
<p>Add to <code>useful_properties</code> all column names of the <code>fData</code> slot of the peptidesCPTAC <code>MSnSet</code> object that you would like to keep (such as gene names, protein names, ontologies, etc.). Other columns will be removed for improved efficiency and memory usage. We certainly want to keep the “Proteins” slot, as this will be our protein (group) identifier. You also want to keep the “Sequence” slot, as peptide-specific effects must be included in our model.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peptidesCPTAC2 &lt;-<span class="st"> </span><span class="kw">preprocess_MaxQuant</span>(peptidesCPTAC, <span class="dt">accession=</span><span class="st">&quot;Proteins&quot;</span>, <span class="dt">exp_annotation=</span>exp_annotation, <span class="dt">logtransform=</span><span class="ot">TRUE</span>, <span class="dt">base=</span><span class="dv">2</span>, <span class="dt">normalisation=</span><span class="st">&quot;quantiles&quot;</span>, <span class="dt">smallestUniqueGroups=</span><span class="ot">TRUE</span>, <span class="dt">useful_properties=</span><span class="kw">c</span>(<span class="st">&quot;Proteins&quot;</span>,<span class="st">&quot;Sequence&quot;</span>), <span class="dt">filter=</span><span class="kw">c</span>(<span class="st">&quot;Potential.contaminant&quot;</span>,<span class="st">&quot;Reverse&quot;</span>), <span class="dt">remove_only_site=</span><span class="ot">TRUE</span>, <span class="dt">file_proteinGroups=</span>file_proteinGroups, <span class="dt">filter_symbol=</span><span class="st">&quot;+&quot;</span>, <span class="dt">minIdentified=</span><span class="dv">2</span>)</code></pre></div>
<p>The following preprocessing steps are executed in this order:</p>
<ol style="list-style-type: decimal">
<li><p>Peptide intensities are log2-transformed (<code>logtransform=TRUE, base=2</code>).</p></li>
<li><p>As a normalization approach, we choose to quantile normalize the peptide intensities (<code>normalisation=&quot;quantiles&quot;</code>). Other options are <code>&quot;sum&quot;</code>, <code>&quot;max&quot;</code>, <code>&quot;center.mean&quot;</code>, <code>&quot;center.median&quot;</code>, <code>&quot;quantiles.robust&quot;</code>, <code>&quot;vsn&quot;</code> or <code>&quot;none&quot;</code> (see <code>MSnbase::normalise</code> for more information).</p></li>
<li><p>Handling overlapping protein groups (<code>smallestUniqueGroups=TRUE</code>): in our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup. Peptides that map to protein groups for wich also subgroups are present in the dataset are removed from the dataset.</p></li>
<li><p>Remove reverse sequences and contaminants (<code>filter=c(&quot;Potential.contaminant&quot;,&quot;Reverse&quot;)</code>).</p></li>
<li><p>Remove all proteins that are only identified by peptides carrying a modification site (<code>remove_only_site=TRUE</code>, <code>file_proteinGroups=file_proteinGroups</code>).</p></li>
<li><p>Remove all superfluous columns in the fData slot, except the “Proteins” and “Sequence” columns (<code>useful_properties=c(&quot;Proteins&quot;,&quot;Sequence&quot;)</code>).</p></li>
<li><p>Remove all peptides that are only identified once in the dataset (<code>minIdentified=2</code>).</p></li>
<li><p>Add experimental annotation.</p></li>
</ol>
</div>
<div id="advanced-alternative-do-the-preprocessing-yourself." class="section level3">
<h3>2.3. Advanced alternative: do the preprocessing yourself.</h3>
<p>It is possible to do all the preprocessing steps yourself. That way, you can do more advanced preprocessing or change the order of certain preprocessing steps if you like to.</p>
<ol style="list-style-type: decimal">
<li>Log2-transform the peptide intensities.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peptidesCPTAC3 &lt;-<span class="st"> </span><span class="kw">log</span>(peptidesCPTAC, <span class="dt">base=</span><span class="dv">2</span>)
<span class="co">#Alternative: </span>
<span class="co">#peptidesCPTAC3 &lt;- log2(peptidesCPTAC3)</span></code></pre></div>
<p>After log2-transformation, we want to replace <code>-Inf</code> values in the peptide intensities by <code>NA</code> values as these values originate from zeros in the untransformed <code>intensities</code> matrix, which means these intensities were actually missing.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">exprs &lt;-<span class="st"> </span><span class="kw">exprs</span>(peptidesCPTAC3)
exprs[<span class="kw">is.infinite</span>(exprs)] &lt;-<span class="st"> </span><span class="ot">NA</span>
<span class="kw">exprs</span>(peptidesCPTAC3) &lt;-<span class="st"> </span>exprs</code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>As a normalization approach, we choose to quantile normalize the peptide intensities.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peptidesCPTAC3 &lt;-<span class="st"> </span><span class="kw">normalise</span>(peptidesCPTAC3, <span class="st">&quot;quantiles&quot;</span>)</code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Handling overlapping protein groups: in our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">groups2 &lt;-<span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">fData</span>(peptidesCPTAC3)$Proteins)
sel &lt;-<span class="st"> </span><span class="kw">fData</span>(peptidesCPTAC3)$Proteins %in%<span class="st"> </span>groups2
peptidesCPTAC3 &lt;-<span class="st"> </span>peptidesCPTAC3[sel]
<span class="kw">head</span>(<span class="kw">fData</span>(peptidesCPTAC3))</code></pre></div>
<pre><code>##             Sequence N.term.cleavage.window C.term.cleavage.window
## 1 AAAAGAGGAGDSGDAVTK       EHQHDEQKAAAAGAGG       DSGDAVTKIGSEDVKL
## 2          AAAALAGGK       QQLSKAAKAAAALAGG       AAALAGGKKSKKKWSK
## 3         AAAALAGGKK       QQLSKAAKAAAALAGG       AALAGGKKSKKKWSKK
## 4       AAADALSDLEIK       MPKETPSKAAADALSD       ALSDLEIKDSKSNLNK
## 5    AAADALSDLEIKDSK       MPKETPSKAAADALSD       DLEIKDSKSNLNKELE
## 6           AAAEEFQR       RERKEREKAAAEEFQR       AAAEEFQRQQELLRQQ
##   Amino.acid.before First.amino.acid Second.amino.acid
## 1                 K                A                 A
## 2                 K                A                 A
## 3                 K                A                 A
## 4                 K                A                 A
## 5                 K                A                 A
## 6                 K                A                 A
##   Second.last.amino.acid Last.amino.acid Amino.acid.after A.Count R.Count
## 1                      T               K                I       7       0
## 2                      G               K                K       5       0
## 3                      K               K                S       5       0
## 4                      I               K                D       4       0
## 5                      S               K                S       4       0
## 6                      Q               R                Q       3       1
##   N.Count D.Count C.Count Q.Count E.Count G.Count H.Count I.Count L.Count
## 1       0       2       0       0       0       5       0       0       0
## 2       0       0       0       0       0       2       0       0       1
## 3       0       0       0       0       0       2       0       0       1
## 4       0       2       0       0       1       0       0       1       2
## 5       0       3       0       0       1       0       0       1       2
## 6       0       0       0       1       2       0       0       0       0
##   K.Count M.Count F.Count P.Count S.Count T.Count W.Count Y.Count V.Count
## 1       1       0       0       0       1       1       0       0       1
## 2       1       0       0       0       0       0       0       0       0
## 3       2       0       0       0       0       0       0       0       0
## 4       1       0       0       0       1       0       0       0       0
## 5       2       0       0       0       2       0       0       0       0
## 6       0       0       1       0       0       0       0       0       0
##   U.Count Length Missed.cleavages      Mass
## 1       0     18                0 1445.6746
## 2       0      9                0  728.4181
## 3       0     10                1  856.5131
## 4       0     12                0 1215.6347
## 5       0     15                1 1545.7886
## 6       0      8                0  920.4352
##                                      Proteins Leading.razor.protein
## 1                        sp|P38915|SPT8_YEAST  sp|P38915|SPT8_YEAST
## 2 sp|Q3E792|RS25A_YEAST;sp|P0C0T4|RS25B_YEAST sp|Q3E792|RS25A_YEAST
## 3 sp|Q3E792|RS25A_YEAST;sp|P0C0T4|RS25B_YEAST sp|Q3E792|RS25A_YEAST
## 4                        sp|P09938|RIR2_YEAST  sp|P09938|RIR2_YEAST
## 5                        sp|P09938|RIR2_YEAST  sp|P09938|RIR2_YEAST
## 6                       sp|P53075|SHE10_YEAST sp|P53075|SHE10_YEAST
##   Start.position End.position Unique..Groups. Unique..Proteins. Charges
## 1             97          114             yes               yes       2
## 2             13           21             yes                no       2
## 3             13           22             yes                no       2
## 4              9           20             yes               yes       2
## 5              9           23             yes               yes       3
## 6            538          545             yes               yes       2
##          PEP   Score Identification.type.6A_1 Identification.type.6A_2
## 1 1.1843e-05  82.942              By matching                 By MS/MS
## 2 7.4562e-06 134.810              By matching              By matching
## 3 3.3094e-09 143.730              By matching              By matching
## 4 9.1593e-23 182.230                 By MS/MS                 By MS/MS
## 5 1.5319e-04  73.927              By matching              By matching
## 6 1.7588e-02  79.474              By matching              By matching
##   Identification.type.6A_3 Identification.type.6A_4
## 1              By matching                 By MS/MS
## 2              By matching                 By MS/MS
## 3              By matching                 By MS/MS
## 4              By matching                 By MS/MS
## 5              By matching                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6A_5 Identification.type.6A_6
## 1              By matching              By matching
## 2              By matching              By matching
## 3              By matching              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6A_7 Identification.type.6A_8
## 1                 By MS/MS                 By MS/MS
## 2                 By MS/MS                 By MS/MS
## 3                 By MS/MS                 By MS/MS
## 4                 By MS/MS              By matching
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6A_9 Identification.type.6B_1
## 1                 By MS/MS              By matching
## 2                 By MS/MS                 By MS/MS
## 3                 By MS/MS              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS              By matching
## 6              By matching                 By MS/MS
##   Identification.type.6B_2 Identification.type.6B_3
## 1              By matching              By matching
## 2              By matching              By matching
## 3                 By MS/MS                 By MS/MS
## 4                 By MS/MS              By matching
## 5              By matching              By matching
## 6              By matching              By matching
##   Identification.type.6B_4 Identification.type.6B_5
## 1              By matching              By matching
## 2              By matching              By matching
## 3              By matching              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6                 By MS/MS              By matching
##   Identification.type.6B_6 Identification.type.6B_7
## 1              By matching              By matching
## 2              By matching                 By MS/MS
## 3              By matching                 By MS/MS
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6B_8 Identification.type.6B_9
## 1                 By MS/MS                 By MS/MS
## 2                 By MS/MS                 By MS/MS
## 3                 By MS/MS                 By MS/MS
## 4              By matching              By matching
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6C_1 Identification.type.6C_2
## 1              By matching              By matching
## 2              By matching                 By MS/MS
## 3              By matching                 By MS/MS
## 4                 By MS/MS              By matching
## 5              By matching              By matching
## 6              By matching              By matching
##   Identification.type.6C_3 Identification.type.6C_4
## 1              By matching              By matching
## 2              By matching                 By MS/MS
## 3              By matching                 By MS/MS
## 4                 By MS/MS                 By MS/MS
## 5              By matching                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6C_5 Identification.type.6C_6
## 1                 By MS/MS              By matching
## 2              By matching              By matching
## 3              By matching              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6C_7 Identification.type.6C_8
## 1                 By MS/MS              By matching
## 2                 By MS/MS                 By MS/MS
## 3                 By MS/MS                 By MS/MS
## 4              By matching                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6C_9 Identification.type.6D_1
## 1              By matching              By matching
## 2                 By MS/MS              By matching
## 3                 By MS/MS              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6D_2 Identification.type.6D_3
## 1              By matching              By matching
## 2              By matching              By matching
## 3              By matching              By matching
## 4              By matching              By matching
## 5                 By MS/MS              By matching
## 6              By matching              By matching
##   Identification.type.6D_4 Identification.type.6D_5
## 1              By matching              By matching
## 2              By matching              By matching
## 3                 By MS/MS              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6D_6 Identification.type.6D_7
## 1                 By MS/MS              By matching
## 2              By matching                 By MS/MS
## 3              By matching                 By MS/MS
## 4                 By MS/MS              By matching
## 5              By matching                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6D_8 Identification.type.6D_9
## 1              By matching              By matching
## 2                 By MS/MS                 By MS/MS
## 3                 By MS/MS                 By MS/MS
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6E_1 Identification.type.6E_2
## 1              By matching              By matching
## 2              By matching              By matching
## 3              By matching              By matching
## 4                 By MS/MS                 By MS/MS
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6E_3 Identification.type.6E_4
## 1              By matching              By matching
## 2              By matching                 By MS/MS
## 3              By matching              By matching
## 4              By matching                 By MS/MS
## 5                 By MS/MS              By matching
## 6              By matching              By matching
##   Identification.type.6E_5 Identification.type.6E_6
## 1              By matching              By matching
## 2              By matching              By matching
## 3              By matching              By matching
## 4                 By MS/MS              By matching
## 5                 By MS/MS                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6E_7 Identification.type.6E_8
## 1              By matching              By matching
## 2                 By MS/MS                 By MS/MS
## 3                 By MS/MS                 By MS/MS
## 4                 By MS/MS                 By MS/MS
## 5              By matching                 By MS/MS
## 6              By matching              By matching
##   Identification.type.6E_9 Experiment.6A_1 Experiment.6A_2 Experiment.6A_3
## 1              By matching              NA               1              NA
## 2                 By MS/MS              NA               1               2
## 3                 By MS/MS              NA               1              NA
## 4                 By MS/MS               1               1               1
## 5                 By MS/MS               1               1              NA
## 6              By matching              NA              NA              NA
##   Experiment.6A_4 Experiment.6A_5 Experiment.6A_6 Experiment.6A_7
## 1               1               1               1               1
## 2               1               1               1               2
## 3               1              NA               1               1
## 4               1               1               1               1
## 5               1               1               1               1
## 6              NA               1              NA              NA
##   Experiment.6A_8 Experiment.6A_9 Experiment.6B_1 Experiment.6B_2
## 1               1               1              NA              NA
## 2               1               1               1               1
## 3               1               1              NA               1
## 4               1               1               1               1
## 5               1               1              NA               1
## 6              NA               1               1              NA
##   Experiment.6B_3 Experiment.6B_4 Experiment.6B_5 Experiment.6B_6
## 1              NA              NA               1               1
## 2               1               2               1              NA
## 3               1               1              NA              NA
## 4               1               1               1               1
## 5              NA               1               1               1
## 6              NA               1              NA              NA
##   Experiment.6B_7 Experiment.6B_8 Experiment.6B_9 Experiment.6C_1
## 1              NA               1               1              NA
## 2               2               1               2              NA
## 3               1               1               1              NA
## 4               1               1               1               1
## 5               1               1               1               1
## 6              NA              NA               1              NA
##   Experiment.6C_2 Experiment.6C_3 Experiment.6C_4 Experiment.6C_5
## 1              NA              NA               1               1
## 2               1               2               2              NA
## 3               1              NA               1              NA
## 4               1               1               1               1
## 5               1               1               1               1
## 6              NA              NA               1              NA
##   Experiment.6C_6 Experiment.6C_7 Experiment.6C_8 Experiment.6C_9
## 1               1               1               1               1
## 2              NA               2               1               1
## 3              NA               1               1               1
## 4               1               1               1               1
## 5               1               1               1               1
## 6               1              NA              NA              NA
##   Experiment.6D_1 Experiment.6D_2 Experiment.6D_3 Experiment.6D_4
## 1              NA              NA              NA               1
## 2              NA               1               1               1
## 3              NA              NA              NA               1
## 4               1               1               1               1
## 5               1               1               1               1
## 6              NA              NA              NA               1
##   Experiment.6D_5 Experiment.6D_6 Experiment.6D_7 Experiment.6D_8
## 1               1               1               1              NA
## 2               1              NA               2               1
## 3              NA              NA               1               1
## 4               1               1               1               1
## 5               1               1               1               1
## 6              NA               1              NA              NA
##   Experiment.6D_9 Experiment.6E_1 Experiment.6E_2 Experiment.6E_3
## 1              NA              NA               1              NA
## 2               2              NA               1               2
## 3               1              NA              NA              NA
## 4               1               1               1               1
## 5               1               1               1               1
## 6               1              NA              NA              NA
##   Experiment.6E_4 Experiment.6E_5 Experiment.6E_6 Experiment.6E_7
## 1              NA               1               1              NA
## 2               2               1              NA               2
## 3               1              NA              NA               1
## 4               1               1               1               1
## 5               1               1               1              NA
## 6              NA              NA              NA              NA
##   Experiment.6E_8 Experiment.6E_9 Intensity Reverse Potential.contaminant
## 1              NA              NA   1190800                              
## 2               2               1 280990000                              
## 3               1               1  33360000                              
## 4               1               1  54622000                              
## 5               1               1  18910000                              
## 6              NA              NA   1158600                              
##   id Protein.group.IDs Mod..peptide.IDs
## 1  0               859                0
## 2  1               230                1
## 3  2               230                2
## 4  3               229                3
## 5  4               229                4
## 6  5              1143                5
##                                                                                                                                                                         Evidence.IDs
## 1                                                                                                                      0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20;21;22;23
## 2                              24;25;26;27;28;29;30;31;32;33;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73
## 3                                                                                                         74;75;76;77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98
## 4 99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143
## 5                144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172;173;174;175;176;177;178;179;180;181;182;183;184
## 6                                                                                                                                            185;186;187;188;189;190;191;192;193;194
##                                                                                                          MS.MS.IDs
## 1                                                                                              0;1;2;3;4;5;6;7;8;9
## 2                                                      10;11;12;13;14;15;16;17;18;19;20;21;22;23;24;25;26;27;28;29
## 3                                                   30;31;32;33;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50
## 4            51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79;80;81;82;83;84
## 5 85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116
## 6                                                                                                          117;118
##   Best.MS.MS Oxidation..M..site.IDs MS.MS.Count
## 1          0                                 10
## 2         21                                 18
## 3         31                                 21
## 4         72                                 29
## 5         94                                 32
## 6        117                                  2</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Remove reverse sequences and contaminants.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Remove reverse sequences</span>
sel &lt;-<span class="st"> </span><span class="kw">fData</span>(peptidesCPTAC3)$Reverse !=<span class="st"> &quot;+&quot;</span>
peptidesCPTAC3 &lt;-<span class="st"> </span>peptidesCPTAC3[sel]
<span class="kw">head</span>(<span class="kw">exprs</span>(peptidesCPTAC3))</code></pre></div>
<pre><code>##      X6A_1    X6A_2    X6A_3    X6A_4    X6A_5    X6A_6    X6A_7    X6A_8
## 1       NA 15.89625       NA 15.74427 16.11690 16.26360       NA       NA
## 2       NA 23.33130 23.66245 23.91300 15.60841 17.66008 21.89950 20.94838
## 3       NA 21.62554       NA 22.05381       NA 15.02926 20.68958 20.00483
## 4 19.64640 20.09650 20.04574 20.32064 20.75817 20.45584 19.63882 20.01568
## 5 15.85773 15.59100       NA 19.33941 19.55917 19.62396 18.94303 19.31408
## 6       NA       NA       NA       NA 17.42813       NA       NA       NA
##      X6A_9    X6B_1    X6B_2    X6B_3    X6B_4    X6B_5    X6B_6    X6B_7
## 1 16.40301       NA       NA       NA       NA 15.60504 16.04583       NA
## 2 20.96813 23.43180 23.26411 23.64645 24.20048 14.91464       NA 22.20164
## 3 19.81333       NA 21.71777 19.37231 16.85073       NA       NA 20.36880
## 4 19.98197 19.62953 19.95218 20.13390 20.04605 20.30895 20.39173 19.42857
## 5 18.96642       NA 16.09153       NA 19.20895 19.09520 19.49627 19.01121
## 6 16.00602       NA       NA       NA 17.54346       NA       NA       NA
##      X6B_8    X6B_9    X6C_1    X6C_2    X6C_3    X6C_4    X6C_5    X6C_6
## 1 15.53808       NA       NA       NA       NA 16.16971 16.09331 15.76673
## 2 20.62686 21.25133       NA 23.52247 24.08292 23.82946       NA       NA
## 3 20.02875 19.62006       NA 21.60703       NA 20.82764       NA       NA
## 4 19.97507 20.02002 20.13560 19.58226 20.02870 19.84447 20.53751 20.65843
## 5 19.23466 19.15420 16.65312 16.28174 15.45126 18.94057 19.12564 19.46605
## 6       NA 16.52722       NA       NA       NA 17.69237       NA 17.58594
##      X6C_7    X6C_8    X6C_9    X6D_1    X6D_2    X6D_3    X6D_4    X6D_5
## 1       NA 15.68773 15.35170       NA       NA       NA 15.96516 16.06759
## 2 22.30765 20.49290 21.00310       NA 23.52274 23.29155 24.05100 16.88207
## 3 20.33127 19.95157 19.64324       NA       NA       NA 22.07140       NA
## 4 19.49678 20.41416 20.01655 20.46055 20.63712 20.54149 20.12157 20.54021
## 5 18.68434 19.24157 19.08266 18.35102 18.22513 18.08876 19.04524 19.98076
## 6       NA       NA       NA       NA       NA       NA 18.17092       NA
##      X6D_6    X6D_7    X6D_8    X6D_9    X6E_1    X6E_2    X6E_3    X6E_4
## 1 16.38325 16.60706       NA       NA       NA 15.48811       NA       NA
## 2       NA 22.09701 20.51658 21.57285       NA 22.82601 22.99603 23.72723
## 3       NA 20.36570 19.82115 19.79300       NA       NA       NA 15.44656
## 4 20.53321 19.90789 19.95061 19.64517 20.26774 19.74733 19.81515 19.80937
## 5 19.65537 18.40360 19.38517 18.73883 18.10882 17.61519 18.09828 18.94790
## 6 15.59841       NA       NA 16.25820       NA       NA       NA       NA
##      X6E_5    X6E_6    X6E_7    X6E_8    X6E_9
## 1 15.48788 15.44796       NA       NA       NA
## 2 13.78541       NA 21.64825 19.69567 20.88645
## 3       NA       NA 20.17699 18.78196 19.33404
## 4 19.62211 19.86977 19.80026 19.96206 19.76630
## 5 18.87684 19.06988       NA 18.53583 18.69385
## 6       NA       NA       NA       NA       NA</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Remove contaminants</span>
sel &lt;-<span class="st"> </span><span class="kw">fData</span>(peptidesCPTAC3)$Potential.contaminant !=<span class="st"> &quot;+&quot;</span>
peptidesCPTAC3 &lt;-<span class="st"> </span>peptidesCPTAC3[sel]
<span class="kw">head</span>(<span class="kw">exprs</span>(peptidesCPTAC3))</code></pre></div>
<pre><code>##      X6A_1    X6A_2    X6A_3    X6A_4    X6A_5    X6A_6    X6A_7    X6A_8
## 1       NA 15.89625       NA 15.74427 16.11690 16.26360       NA       NA
## 2       NA 23.33130 23.66245 23.91300 15.60841 17.66008 21.89950 20.94838
## 3       NA 21.62554       NA 22.05381       NA 15.02926 20.68958 20.00483
## 4 19.64640 20.09650 20.04574 20.32064 20.75817 20.45584 19.63882 20.01568
## 5 15.85773 15.59100       NA 19.33941 19.55917 19.62396 18.94303 19.31408
## 6       NA       NA       NA       NA 17.42813       NA       NA       NA
##      X6A_9    X6B_1    X6B_2    X6B_3    X6B_4    X6B_5    X6B_6    X6B_7
## 1 16.40301       NA       NA       NA       NA 15.60504 16.04583       NA
## 2 20.96813 23.43180 23.26411 23.64645 24.20048 14.91464       NA 22.20164
## 3 19.81333       NA 21.71777 19.37231 16.85073       NA       NA 20.36880
## 4 19.98197 19.62953 19.95218 20.13390 20.04605 20.30895 20.39173 19.42857
## 5 18.96642       NA 16.09153       NA 19.20895 19.09520 19.49627 19.01121
## 6 16.00602       NA       NA       NA 17.54346       NA       NA       NA
##      X6B_8    X6B_9    X6C_1    X6C_2    X6C_3    X6C_4    X6C_5    X6C_6
## 1 15.53808       NA       NA       NA       NA 16.16971 16.09331 15.76673
## 2 20.62686 21.25133       NA 23.52247 24.08292 23.82946       NA       NA
## 3 20.02875 19.62006       NA 21.60703       NA 20.82764       NA       NA
## 4 19.97507 20.02002 20.13560 19.58226 20.02870 19.84447 20.53751 20.65843
## 5 19.23466 19.15420 16.65312 16.28174 15.45126 18.94057 19.12564 19.46605
## 6       NA 16.52722       NA       NA       NA 17.69237       NA 17.58594
##      X6C_7    X6C_8    X6C_9    X6D_1    X6D_2    X6D_3    X6D_4    X6D_5
## 1       NA 15.68773 15.35170       NA       NA       NA 15.96516 16.06759
## 2 22.30765 20.49290 21.00310       NA 23.52274 23.29155 24.05100 16.88207
## 3 20.33127 19.95157 19.64324       NA       NA       NA 22.07140       NA
## 4 19.49678 20.41416 20.01655 20.46055 20.63712 20.54149 20.12157 20.54021
## 5 18.68434 19.24157 19.08266 18.35102 18.22513 18.08876 19.04524 19.98076
## 6       NA       NA       NA       NA       NA       NA 18.17092       NA
##      X6D_6    X6D_7    X6D_8    X6D_9    X6E_1    X6E_2    X6E_3    X6E_4
## 1 16.38325 16.60706       NA       NA       NA 15.48811       NA       NA
## 2       NA 22.09701 20.51658 21.57285       NA 22.82601 22.99603 23.72723
## 3       NA 20.36570 19.82115 19.79300       NA       NA       NA 15.44656
## 4 20.53321 19.90789 19.95061 19.64517 20.26774 19.74733 19.81515 19.80937
## 5 19.65537 18.40360 19.38517 18.73883 18.10882 17.61519 18.09828 18.94790
## 6 15.59841       NA       NA 16.25820       NA       NA       NA       NA
##      X6E_5    X6E_6    X6E_7    X6E_8    X6E_9
## 1 15.48788 15.44796       NA       NA       NA
## 2 13.78541       NA 21.64825 19.69567 20.88645
## 3       NA       NA 20.17699 18.78196 19.33404
## 4 19.62211 19.86977 19.80026 19.96206 19.76630
## 5 18.87684 19.06988       NA 18.53583 18.69385
## 6       NA       NA       NA       NA       NA</code></pre>
<!-- More advanced, mostly not necessary in a standard analysis: add a UPS-indicator to the properties slot of the peptides object. -->
<!-- ```{r eval=TRUE} -->
<!-- fData(peptidesCPTAC3)$ups <- as.double(grepl("_HUMAN_UPS", fData(peptidesCPTAC3)$Proteins)) -->
<!-- ``` -->
<p>Small check:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">fData</span>(peptidesCPTAC3)$Proteins[<span class="kw">grepl</span>(<span class="st">'_HUMAN_UPS'</span>,<span class="kw">fData</span>(peptidesCPTAC3)$Proteins)]))</code></pre></div>
<pre><code>## [1] 46</code></pre>
<p>46 out of a total 48 spiked-in human UPS proteins are present in this dataset with at least one peptide sequence identified.</p>
<ol start="5" style="list-style-type: decimal">
<li>Remove proteins that are only identified by modified peptides.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  proteinGroups &lt;-<span class="st"> </span><span class="kw">read.table</span>(file_proteinGroups, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">&quot;&quot;</span>, <span class="dt">comment.char =</span> <span class="st">&quot;&quot;</span>)
  only_site &lt;-<span class="st"> </span>proteinGroups$Only.identified.by.site
  only_site[<span class="kw">is.na</span>(only_site)] &lt;-<span class="st"> &quot;&quot;</span>
  removed_proteins &lt;-<span class="st"> </span>proteinGroups$Protein.IDs[only_site==<span class="st">&quot;+&quot;</span>]

  sel &lt;-<span class="st"> </span>!(<span class="kw">as.character</span>(Biobase::<span class="kw">fData</span>(peptidesCPTAC3)[,<span class="st">&quot;Proteins&quot;</span>]) %in%<span class="st"> </span><span class="kw">as.character</span>(removed_proteins))
  peptidesCPTAC3 &lt;-<span class="st"> </span>peptidesCPTAC3[sel]</code></pre></div>
<ol start="6" style="list-style-type: decimal">
<li>Retain only those properties in the properties slot that are useful for our further analysis.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fData</span>(peptidesCPTAC3) &lt;-<span class="st"> </span><span class="kw">fData</span>(peptidesCPTAC3)[,<span class="kw">c</span>(<span class="st">&quot;Proteins&quot;</span>,<span class="st">&quot;Sequence&quot;</span>)] <span class="co">#, &quot;ups&quot;</span></code></pre></div>
<ol start="7" style="list-style-type: decimal">
<li>How many times shoud a peptide be identified in order to keep it in the dataset? We require at least 2 identifications of a peptide sequence, as with 1 identification, the models will be perfectly confounded.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minIdentified &lt;-<span class="st"> </span><span class="dv">2</span>
keepers &lt;-<span class="st"> </span><span class="kw">rowSums</span>(!<span class="kw">is.na</span>(<span class="kw">exprs</span>(peptidesCPTAC3)))&gt;=minIdentified
peptidesCPTAC3 &lt;-<span class="st"> </span>peptidesCPTAC3[keepers]
<span class="kw">head</span>(<span class="kw">exprs</span>(peptidesCPTAC3))</code></pre></div>
<pre><code>##      X6A_1    X6A_2    X6A_3    X6A_4    X6A_5    X6A_6    X6A_7    X6A_8
## 1       NA 15.89625       NA 15.74427 16.11690 16.26360       NA       NA
## 2       NA 23.33130 23.66245 23.91300 15.60841 17.66008 21.89950 20.94838
## 3       NA 21.62554       NA 22.05381       NA 15.02926 20.68958 20.00483
## 4 19.64640 20.09650 20.04574 20.32064 20.75817 20.45584 19.63882 20.01568
## 5 15.85773 15.59100       NA 19.33941 19.55917 19.62396 18.94303 19.31408
## 6       NA       NA       NA       NA 17.42813       NA       NA       NA
##      X6A_9    X6B_1    X6B_2    X6B_3    X6B_4    X6B_5    X6B_6    X6B_7
## 1 16.40301       NA       NA       NA       NA 15.60504 16.04583       NA
## 2 20.96813 23.43180 23.26411 23.64645 24.20048 14.91464       NA 22.20164
## 3 19.81333       NA 21.71777 19.37231 16.85073       NA       NA 20.36880
## 4 19.98197 19.62953 19.95218 20.13390 20.04605 20.30895 20.39173 19.42857
## 5 18.96642       NA 16.09153       NA 19.20895 19.09520 19.49627 19.01121
## 6 16.00602       NA       NA       NA 17.54346       NA       NA       NA
##      X6B_8    X6B_9    X6C_1    X6C_2    X6C_3    X6C_4    X6C_5    X6C_6
## 1 15.53808       NA       NA       NA       NA 16.16971 16.09331 15.76673
## 2 20.62686 21.25133       NA 23.52247 24.08292 23.82946       NA       NA
## 3 20.02875 19.62006       NA 21.60703       NA 20.82764       NA       NA
## 4 19.97507 20.02002 20.13560 19.58226 20.02870 19.84447 20.53751 20.65843
## 5 19.23466 19.15420 16.65312 16.28174 15.45126 18.94057 19.12564 19.46605
## 6       NA 16.52722       NA       NA       NA 17.69237       NA 17.58594
##      X6C_7    X6C_8    X6C_9    X6D_1    X6D_2    X6D_3    X6D_4    X6D_5
## 1       NA 15.68773 15.35170       NA       NA       NA 15.96516 16.06759
## 2 22.30765 20.49290 21.00310       NA 23.52274 23.29155 24.05100 16.88207
## 3 20.33127 19.95157 19.64324       NA       NA       NA 22.07140       NA
## 4 19.49678 20.41416 20.01655 20.46055 20.63712 20.54149 20.12157 20.54021
## 5 18.68434 19.24157 19.08266 18.35102 18.22513 18.08876 19.04524 19.98076
## 6       NA       NA       NA       NA       NA       NA 18.17092       NA
##      X6D_6    X6D_7    X6D_8    X6D_9    X6E_1    X6E_2    X6E_3    X6E_4
## 1 16.38325 16.60706       NA       NA       NA 15.48811       NA       NA
## 2       NA 22.09701 20.51658 21.57285       NA 22.82601 22.99603 23.72723
## 3       NA 20.36570 19.82115 19.79300       NA       NA       NA 15.44656
## 4 20.53321 19.90789 19.95061 19.64517 20.26774 19.74733 19.81515 19.80937
## 5 19.65537 18.40360 19.38517 18.73883 18.10882 17.61519 18.09828 18.94790
## 6 15.59841       NA       NA 16.25820       NA       NA       NA       NA
##      X6E_5    X6E_6    X6E_7    X6E_8    X6E_9
## 1 15.48788 15.44796       NA       NA       NA
## 2 13.78541       NA 21.64825 19.69567 20.88645
## 3       NA       NA 20.17699 18.78196 19.33404
## 4 19.62211 19.86977 19.80026 19.96206 19.76630
## 5 18.87684 19.06988       NA 18.53583 18.69385
## 6       NA       NA       NA       NA       NA</code></pre>
<ol start="8" style="list-style-type: decimal">
<li>Add the experimental design to the MSnSet object.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pData</span>(peptidesCPTAC3) &lt;-<span class="st"> </span>exp_annotation</code></pre></div>
<p>Check whether our manually preprocessed <code>peptidesCPTAC3</code> is equal to <code>peptidesCPTAC2</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">all</span>(<span class="kw">exprs</span>(peptidesCPTAC2)==<span class="kw">exprs</span>(peptidesCPTAC3), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">all</span>(<span class="kw">fData</span>(peptidesCPTAC2)==<span class="kw">fData</span>(peptidesCPTAC3), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">all</span>(<span class="kw">pData</span>(peptidesCPTAC2)==<span class="kw">pData</span>(peptidesCPTAC3), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="convert-the-msnset-object-peptidescptac2-into-a-protdata-object-proteinscptac." class="section level3">
<h3>3. Convert the MSnSet object peptidesCPTAC2 into a protdata object proteinsCPTAC.</h3>
<p><strong>Caution:</strong> this step might take some time! (5 - 10 minutes on our system). The accession element denotes by which column the individual peptide observations should be grouped.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(proteinsCPTAC &lt;-<span class="st"> </span><span class="kw">MSnSet2protdata</span>(peptidesCPTAC2, <span class="dt">accession=</span><span class="st">&quot;Proteins&quot;</span>))</code></pre></div>
<pre><code>##    user  system elapsed 
## 323.422  11.840 336.164</code></pre>
</div>
<div id="main-analysis" class="section level3">
<h3>4. Main Analysis</h3>
</div>
<div id="fit-the-model" class="section level3">
<h3>4.1. Fit the model</h3>
<p>When your goal is to analyze the CPTAC data in the recommended way (i.e. a robust ridge model with variance shrinkage and M estimation), your model should be constructed look as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(modelCPTAC_RR &lt;-<span class="st"> </span><span class="kw">fit.model</span>(<span class="dt">protdata=</span>proteinsCPTAC, <span class="dt">response=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">fixed=</span><span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;lab&quot;</span>), <span class="dt">random=</span><span class="kw">c</span>(<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;run&quot;</span>), <span class="dt">shrinkage.fixed=</span><span class="ot">NULL</span>, <span class="dt">weights=</span><span class="st">&quot;Huber&quot;</span>, <span class="dt">squeezeVar=</span><span class="ot">TRUE</span>))</code></pre></div>
<pre><code>##     user   system  elapsed 
## 1122.451    4.687 1563.430</code></pre>
<p>Caution: model fitting takes some time with large datasets! On our computer, fitting the robust ridge model took about 13 minutes.</p>
</div>
<div id="save-your-results-1" class="section level3">
<h3>4.2. Save your results</h3>
<p>Saving results is completely analogous to the <em>Francisella</em> example.</p>
</div>
<div id="comparing-different-types-of-models" class="section level3">
<h3>4.3. Comparing different types of models</h3>
<p>In this section, we fit different types of models to the CPTAC data, analogous to what has been done in Goeminne et al. (2016) in order to compare their performances. We compare an ordinary least squares (OLS) model with and without M estimation (Huber weights) to a ridge model with and without M estimation. As we also want to assess the impact of the empirical Bayes variance estimation, we set <code>squeezeVar=FALSE</code> so that no variance squeezing is performed yet.</p>
<p>In order to do an objective comparison between different approaches, we chose here not to include the “run” effect, because this leads to massive overfitting in the ordinary least squares approaches. This effect will mostly be equal to zero anyways because each repeat in the CPTAC dataset is in fact a technical repeat and therefore the variability in peptide intensities between samples will be very small. When doing a standard robust ridge analysis on true biological data, we advise however to include the “run” effect as a random effect. As the simultanous shrinkage of fixed effects was not yet implemented at the time, we provide “lab” as a random effect in order to get results that are as close as possible to those of Goeminne et al. (2016).</p>
<p>Fit an OLS model (<code>shrinkage.fixed=c(0,0,0,0)</code>) without M estimation (<code>weights=NULL</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(modelCPTAC_Lm &lt;-<span class="st"> </span><span class="kw">fit.model</span>(<span class="dt">protdata=</span>proteinsCPTAC, <span class="dt">response=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">fixed=</span><span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;lab&quot;</span>), <span class="dt">random=</span><span class="ot">NULL</span>, <span class="dt">shrinkage.fixed=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="dt">weights=</span><span class="ot">NULL</span>, <span class="dt">squeezeVar=</span><span class="ot">FALSE</span>))</code></pre></div>
<pre><code>##    user  system elapsed 
##   4.404   0.090   4.529</code></pre>
<p>Fit an OLS model (<code>shrinkage.fixed=c(0,0,0,0)</code>) with M estimation (<code>weights=Huber</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(modelCPTAC_Lm_M &lt;-<span class="st"> </span><span class="kw">fit.model</span>(<span class="dt">protdata=</span>proteinsCPTAC, <span class="dt">response=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">fixed=</span><span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;lab&quot;</span>), <span class="dt">random=</span><span class="ot">NULL</span>, <span class="dt">shrinkage.fixed=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="dt">weights=</span><span class="st">&quot;Huber&quot;</span>, <span class="dt">squeezeVar=</span><span class="ot">FALSE</span>))</code></pre></div>
<pre><code>##    user  system elapsed 
##  13.234   0.224  13.681</code></pre>
<p>Fit a ridge model (<code>shrinkage.fixed=c(0,1)</code>) without M estimation (<code>weights=NULL</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(modelCPTAC_Ridge &lt;-<span class="st"> </span><span class="kw">fit.model</span>(<span class="dt">protdata=</span>proteinsCPTAC, <span class="dt">response=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">fixed=</span><span class="st">&quot;condition&quot;</span>, <span class="dt">random=</span><span class="kw">c</span>(<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;lab&quot;</span>), <span class="dt">shrinkage.fixed=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">weights=</span><span class="ot">NULL</span>, <span class="dt">squeezeVar=</span><span class="ot">FALSE</span>))</code></pre></div>
<pre><code>##    user  system elapsed 
## 101.706   0.834 104.035</code></pre>
<p>Fit a ridge model (<code>shrinkage.fixed=c(0,1)</code>) with M estimation (<code>weights=Huber</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system.time</span>(modelCPTAC_Ridge_M &lt;-<span class="st"> </span><span class="kw">fit.model</span>(<span class="dt">protdata=</span>proteinsCPTAC, <span class="dt">response=</span><span class="st">&quot;quant_value&quot;</span>, <span class="dt">fixed=</span><span class="st">&quot;condition&quot;</span>, <span class="dt">random=</span><span class="kw">c</span>(<span class="st">&quot;Sequence&quot;</span>,<span class="st">&quot;lab&quot;</span>), <span class="dt">shrinkage.fixed=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">weights=</span><span class="st">&quot;Huber&quot;</span>, <span class="dt">squeezeVar=</span><span class="ot">FALSE</span>))</code></pre></div>
<pre><code>## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q

## Warning in optwrap(optimizer, devfun, getStart(start, rho$lower, rho$pp), :
## convergence code 3 from bobyqa: bobyqa -- a trust region step failed to
## reduce q</code></pre>
<pre><code>##     user   system  elapsed 
## 1927.854    8.751 1954.455</code></pre>
<p>As you noticed, an OLS model can be fit by setting <code>shrinkage.fixed=c(0,0,0,0)</code>. This indicates that neither the intercept and none of our three fixed effects should be shrunken. The ridge model is here specified by setting <code>shrinkage.fixed=c(0,1)</code>. This indicates that the intercept should not be shrunken, but the fixed effect <code>&quot;condition&quot;</code> should be shrunken. Alternatively, one could also set <code>shrinkage.fixed=NULL</code> as the shrinkage of all fixed effects except for the intercept is the default behaviour of MSqRob.</p>
</div>
<div id="inspect-the-models" class="section level3">
<h3>4.4. Inspect the models</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">attr</span>(<span class="kw">getModels</span>(modelCPTAC_RR[<span class="dv">1</span>]),<span class="st">&quot;MSqRob_levels&quot;</span>)</code></pre></div>
<pre><code>##  [1] &quot;(Intercept)&quot;                &quot;condition6A&quot;               
##  [3] &quot;condition6B&quot;                &quot;condition6C&quot;               
##  [5] &quot;condition6D&quot;                &quot;condition6E&quot;               
##  [7] &quot;lab1&quot;                       &quot;lab2&quot;                      
##  [9] &quot;lab3&quot;                       &quot;runX6A_2&quot;                  
## [11] &quot;runX6A_4&quot;                   &quot;runX6A_5&quot;                  
## [13] &quot;runX6A_6&quot;                   &quot;runX6A_9&quot;                  
## [15] &quot;runX6B_5&quot;                   &quot;runX6B_6&quot;                  
## [17] &quot;runX6B_8&quot;                   &quot;runX6C_4&quot;                  
## [19] &quot;runX6C_5&quot;                   &quot;runX6C_6&quot;                  
## [21] &quot;runX6C_8&quot;                   &quot;runX6C_9&quot;                  
## [23] &quot;runX6D_4&quot;                   &quot;runX6D_5&quot;                  
## [25] &quot;runX6D_6&quot;                   &quot;runX6D_7&quot;                  
## [27] &quot;runX6E_2&quot;                   &quot;runX6E_5&quot;                  
## [29] &quot;runX6E_6&quot;                   &quot;SequenceAAAAGAGGAGDSGDAVTK&quot;</code></pre>
<p>MSqRob always tries to fit a model, but some models are overparameterized because too many parameters are fit on too few observations. These models have convergence problems and can be removed from the data prior to estimating p-values. This is only relevant for models that perform shrinkage or use any kind of random effect.</p>
<p>Extract the data as a list from the protdata object <code>proteinsCPTAC</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw">getData</span>(proteinsCPTAC)</code></pre></div>
</div>
<div id="avanced-and-optional-check-which-models-would-fail-to-converge-by-building-a-linear-mixed-effects-model." class="section level3">
<h3>4.5. Avanced and optional: check which models would fail to converge by building a linear mixed effects model.</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">convergence &lt;-<span class="st"> </span><span class="kw">lapply</span>(data, function(x){<span class="kw">return</span>(<span class="kw">tryCatch</span>(lme4::<span class="kw">lFormula</span>(<span class="kw">formula</span>(<span class="st">&quot;quant_value~1+(1|condition)+(1|lab)+(1|Sequence)&quot;</span>),x,<span class="dt">control =</span> lme4::<span class="kw">lmerControl</span>()), <span class="dt">error=</span>function(e){
  <span class="kw">return</span>(<span class="ot">NA</span>)
}))})</code></pre></div>
<p>Get the indices of these overparameterized models.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">na_indices &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">is.na</span>(convergence))</code></pre></div>
<p>See which proteins you removed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getAccessions</span>(proteinsCPTAC)[na_indices]</code></pre></div>
<pre><code>##   [1] sp|P38915|SPT8_YEAST                                                                                                                                                                                                                                                   
##   [2] sp|P10622|RLA3_YEAST                                                                                                                                                                                                                                                   
##   [3] sp|P33421|SDH3_YEAST                                                                                                                                                                                                                                                   
##   [4] sp|P33328|SNC2_YEAST                                                                                                                                                                                                                                                   
##   [5] sp|P38555|YPT31_YEAST                                                                                                                                                                                                                                                  
##   [6] sp|P36139|PET10_YEAST                                                                                                                                                                                                                                                  
##   [7] sp|Q12517|DCP1_YEAST                                                                                                                                                                                                                                                   
##   [8] sp|P39516|RS14B_YEAST                                                                                                                                                                                                                                                  
##   [9] sp|Q12078|SMF3_YEAST                                                                                                                                                                                                                                                   
##  [10] sp|P38885|AIM46_YEAST                                                                                                                                                                                                                                                  
##  [11] sp|P40548|SNL1_YEAST                                                                                                                                                                                                                                                   
##  [12] sp|P48363|PFD3_YEAST                                                                                                                                                                                                                                                   
##  [13] sp|P14065|GCY1_YEAST                                                                                                                                                                                                                                                   
##  [14] sp|P46970|NMD5_YEAST                                                                                                                                                                                                                                                   
##  [15] sp|P23201|SPA2_YEAST                                                                                                                                                                                                                                                   
##  [16] sp|P05743|RL26A_YEAST                                                                                                                                                                                                                                                  
##  [17] sp|P0CH09|RL402_YEAST;sp|P0CH08|RL401_YEAST                                                                                                                                                                                                                            
##  [18] sp|P38256|YBU6_YEAST                                                                                                                                                                                                                                                   
##  [19] sp|Q03880|PKR1_YEAST                                                                                                                                                                                                                                                   
##  [20] sp|P38747|OTU2_YEAST                                                                                                                                                                                                                                                   
##  [21] sp|O43137|YB085_YEAST                                                                                                                                                                                                                                                  
##  [22] sp|P40212|RL13B_YEAST                                                                                                                                                                                                                                                  
##  [23] sp|Q04177|UTP5_YEAST                                                                                                                                                                                                                                                   
##  [24] sp|P24031|PPA3_YEAST;sp|P00635|PPA5_YEAST                                                                                                                                                                                                                              
##  [25] sp|P32504|CBF3A_YEAST                                                                                                                                                                                                                                                  
##  [26] sp|P38145|RISA_YEAST                                                                                                                                                                                                                                                   
##  [27] sp|P32803|EMP24_YEAST                                                                                                                                                                                                                                                  
##  [28] sp|Q12040|YO283_YEAST                                                                                                                                                                                                                                                  
##  [29] sp|Q12099|FAL1_YEAST                                                                                                                                                                                                                                                   
##  [30] sp|Q04175|SXM1_YEAST                                                                                                                                                                                                                                                   
##  [31] sp|Q07914|TIM14_YEAST                                                                                                                                                                                                                                                  
##  [32] sp|P30665|MCM4_YEAST                                                                                                                                                                                                                                                   
##  [33] sp|P06704|CDC31_YEAST                                                                                                                                                                                                                                                  
##  [34] sp|P40956|GTS1_YEAST                                                                                                                                                                                                                                                   
##  [35] sp|P53304|SLI1_YEAST                                                                                                                                                                                                                                                   
##  [36] sp|Q04409|EMI2_YEAST                                                                                                                                                                                                                                                   
##  [37] sp|P54858|DOS2_YEAST                                                                                                                                                                                                                                                   
##  [38] sp|P50106|RPA14_YEAST                                                                                                                                                                                                                                                  
##  [39] sp|P53064|RTF1_YEAST                                                                                                                                                                                                                                                   
##  [40] sp|P11746|MCM1_YEAST                                                                                                                                                                                                                                                   
##  [41] sp|P08456|PSS_YEAST                                                                                                                                                                                                                                                    
##  [42] sp|Q03525|TMA23_YEAST                                                                                                                                                                                                                                                  
##  [43] sp|Q12156|AIM7_YEAST                                                                                                                                                                                                                                                   
##  [44] sp|P47771|ALDH2_YEAST                                                                                                                                                                                                                                                  
##  [45] sp|P17898|CPT1_YEAST                                                                                                                                                                                                                                                   
##  [46] sp|P32500|NDC1_YEAST                                                                                                                                                                                                                                                   
##  [47] sp|P32910|RPC6_YEAST                                                                                                                                                                                                                                                   
##  [48] sp|P39998|EDC3_YEAST                                                                                                                                                                                                                                                   
##  [49] sp|Q03529|SCS7_YEAST                                                                                                                                                                                                                                                   
##  [50] sp|Q06449|PIN3_YEAST                                                                                                                                                                                                                                                   
##  [51] sp|P25618|CWH43_YEAST                                                                                                                                                                                                                                                  
##  [52] sp|P53093|YIP4_YEAST                                                                                                                                                                                                                                                   
##  [53] sp|P52893|ALAM_YEAST                                                                                                                                                                                                                                                   
##  [54] sp|Q02753|RL21A_YEAST                                                                                                                                                                                                                                                  
##  [55] sp|Q12672|RL21B_YEAST                                                                                                                                                                                                                                                  
##  [56] sp|P0CT04|IPB2_YEAST                                                                                                                                                                                                                                                   
##  [57] sp|P20436|RPAB3_YEAST                                                                                                                                                                                                                                                  
##  [58] sp|P89105|CTR9_YEAST                                                                                                                                                                                                                                                   
##  [59] sp|P05744|RL33A_YEAST                                                                                                                                                                                                                                                  
##  [60] sp|P41056|RL33B_YEAST                                                                                                                                                                                                                                                  
##  [61] sp|Q07897|CMS1_YEAST                                                                                                                                                                                                                                                   
##  [62] sp|P48234|NOL10_YEAST                                                                                                                                                                                                                                                  
##  [63] sp|P17442|PHO81_YEAST                                                                                                                                                                                                                                                  
##  [64] sp|Q06132|SGD1_YEAST                                                                                                                                                                                                                                                   
##  [65] sp|Q07451|YET3_YEAST                                                                                                                                                                                                                                                   
##  [66] sp|Q08925|MRN1_YEAST                                                                                                                                                                                                                                                   
##  [67] sp|O13516|RS9A_YEAST                                                                                                                                                                                                                                                   
##  [68] sp|P38063|KPR4_YEAST                                                                                                                                                                                                                                                   
##  [69] sp|P47019|ALB1_YEAST                                                                                                                                                                                                                                                   
##  [70] sp|P46654|RSSA2_YEAST                                                                                                                                                                                                                                                  
##  [71] sp|Q06580|MLC2_YEAST                                                                                                                                                                                                                                                   
##  [72] sp|P38066|RIB1_YEAST                                                                                                                                                                                                                                                   
##  [73] sp|P41939|IDHC_YEAST;sp|P53982|IDHH_YEAST                                                                                                                                                                                                                              
##  [74] sp|P47172|YJ9I_YEAST                                                                                                                                                                                                                                                   
##  [75] sp|P40541|SCC3_YEAST                                                                                                                                                                                                                                                   
##  [76] sp|P18852|GBG_YEAST                                                                                                                                                                                                                                                    
##  [77] sp|Q07540|FRDA_YEAST                                                                                                                                                                                                                                                   
##  [78] sp|A5Z2X5|YP010_YEAST                                                                                                                                                                                                                                                  
##  [79] sp|Q07915|RLP24_YEAST                                                                                                                                                                                                                                                  
##  [80] sp|P56628|RL22B_YEAST                                                                                                                                                                                                                                                  
##  [81] sp|P38697|IMDH2_YEAST                                                                                                                                                                                                                                                  
##  [82] sp|P52490|UBC13_YEAST                                                                                                                                                                                                                                                  
##  [83] sp|P53250|TWF1_YEAST                                                                                                                                                                                                                                                   
##  [84] sp|Q99385|VCX1_YEAST                                                                                                                                                                                                                                                   
##  [85] sp|P33448|TOM6_YEAST                                                                                                                                                                                                                                                   
##  [86] sp|Q04773|YMW4_YEAST                                                                                                                                                                                                                                                   
##  [87] sp|Q12273|YO11B_YEAST                                                                                                                                                                                                                                                  
##  [88] sp|Q03434|YM12B_YEAST;sp|Q12141|YG11B_YEAST;sp|P47100|YJ12B_YEAST;sp|P47099|YJ12A_YEAST                                                                                                                                                                                
##  [89] sp|P80967|TOM5_YEAST                                                                                                                                                                                                                                                   
##  [90] sp|P40030|ERG28_YEAST                                                                                                                                                                                                                                                  
##  [91] sp|Q08746|RRS1_YEAST                                                                                                                                                                                                                                                   
##  [92] sp|P11076|ARF1_YEAST                                                                                                                                                                                                                                                   
##  [93] sp|P25642|IMG2_YEAST                                                                                                                                                                                                                                                   
##  [94] sp|P38759|VPS29_YEAST                                                                                                                                                                                                                                                  
##  [95] sp|Q04119|PPN1_YEAST                                                                                                                                                                                                                                                   
##  [96] sp|Q12690|RL13A_YEAST                                                                                                                                                                                                                                                  
##  [97] sp|P53859|CSL4_YEAST                                                                                                                                                                                                                                                   
##  [98] sp|P39726|GCSH_YEAST                                                                                                                                                                                                                                                   
##  [99] sp|Q01590|SED5_YEAST                                                                                                                                                                                                                                                   
## [100] sp|P51862|ROM2_YEAST                                                                                                                                                                                                                                                   
## [101] sp|P38760|MIP6_YEAST                                                                                                                                                                                                                                                   
## [102] sp|P40005|PFD2_YEAST                                                                                                                                                                                                                                                   
## [103] sp|Q06497|ANT1_YEAST                                                                                                                                                                                                                                                   
## [104] sp|Q3E7Y3|RS22B_YEAST                                                                                                                                                                                                                                                  
## [105] sp|P32835|GSP1_YEAST                                                                                                                                                                                                                                                   
## [106] sp|P32905|RSSA1_YEAST                                                                                                                                                                                                                                                  
## [107] sp|P11632|NHP6A_YEAST                                                                                                                                                                                                                                                  
## [108] sp|P11633|NHP6B_YEAST                                                                                                                                                                                                                                                  
## [109] sp|P39002|LCF3_YEAST                                                                                                                                                                                                                                                   
## [110] sp|P53507|TOM7_YEAST                                                                                                                                                                                                                                                   
## [111] sp|Q12284|ERV2_YEAST                                                                                                                                                                                                                                                   
## [112] sp|P32770|NRP1_YEAST                                                                                                                                                                                                                                                   
## [113] sp|P32857|PTM1_YEAST                                                                                                                                                                                                                                                   
## [114] sp|P32477|GSH1_YEAST                                                                                                                                                                                                                                                   
## [115] sp|P47138|DPH4_YEAST                                                                                                                                                                                                                                                   
## [116] sp|Q3E834|YOS1_YEAST                                                                                                                                                                                                                                                   
## [117] sp|Q05123|ARP9_YEAST                                                                                                                                                                                                                                                   
## [118] sp|P07275|PUT2_YEAST                                                                                                                                                                                                                                                   
## [119] sp|P0C2I9|YP12B_YEAST;sp|Q99337|YN12B_YEAST;sp|Q12501|YO22B_YEAST;sp|Q12337|YG21B_YEAST;sp|Q12113|YO21B_YEAST;sp|Q03494|YD22B_YEAST;sp|P25384|YC21B_YEAST;sp|P0CX64|YG22B_YEAST;sp|P0CX63|YF21B_YEAST;sp|P0C2J3|YL21B_YEAST;sp|Q12491|YB21B_YEAST;sp|Q12472|YD21B_YEAST
## [120] sp|P25627|BUD23_YEAST                                                                                                                                                                                                                                                  
## [121] sp|Q06991|PUN1_YEAST                                                                                                                                                                                                                                                   
## [122] sp|P38616|YGP1_YEAST                                                                                                                                                                                                                                                   
## [123] sp|Q03178|PIR1_YEAST                                                                                                                                                                                                                                                   
## [124] sp|Q07746|YD242_YEAST                                                                                                                                                                                                                                                  
## [125] sp|Q3E747|YL363_YEAST                                                                                                                                                                                                                                                  
## [126] sp|Q12030|TAF10_YEAST                                                                                                                                                                                                                                                  
## [127] sp|P40087|DDI1_YEAST                                                                                                                                                                                                                                                   
## [128] sp|O94742|SEM1_YEAST                                                                                                                                                                                                                                                   
## [129] sp|P28496|LAC1_YEAST                                                                                                                                                                                                                                                   
## [130] sp|P36516|RM03_YEAST                                                                                                                                                                                                                                                   
## [131] sp|P53121|FLC3_YEAST                                                                                                                                                                                                                                                   
## [132] sp|Q12213|RL7B_YEAST                                                                                                                                                                                                                                                   
## [133] sp|P05737|RL7A_YEAST                                                                                                                                                                                                                                                   
## [134] sp|Q06508|LOA1_YEAST                                                                                                                                                                                                                                                   
## [135] sp|P33416|HSP78_YEAST                                                                                                                                                                                                                                                  
## [136] sp|P25582|SPB1_YEAST                                                                                                                                                                                                                                                   
## [137] sp|P53847|DSL1_YEAST                                                                                                                                                                                                                                                   
## [138] sp|Q07505|DLHH_YEAST                                                                                                                                                                                                                                                   
## [139] sp|Q12160|YP063_YEAST                                                                                                                                                                                                                                                  
## [140] sp|P32349|RPC3_YEAST                                                                                                                                                                                                                                                   
## [141] sp|P36135|UTH1_YEAST                                                                                                                                                                                                                                                   
## [142] sp|P32478|HS150_YEAST                                                                                                                                                                                                                                                  
## [143] sp|Q12255|NYV1_YEAST                                                                                                                                                                                                                                                   
## [144] sp|Q03305|RMT2_YEAST                                                                                                                                                                                                                                                   
## [145] sp|P05747|RL29_YEAST                                                                                                                                                                                                                                                   
## [146] sp|P18851|GBB_YEAST                                                                                                                                                                                                                                                    
## [147] sp|P53885|MDG1_YEAST                                                                                                                                                                                                                                                   
## [148] sp|P46956|PHO86_YEAST                                                                                                                                                                                                                                                  
## [149] P69905ups|HBA_HUMAN_UPS                                                                                                                                                                                                                                                
## [150] sp|P20049|TYR1_YEAST                                                                                                                                                                                                                                                   
## [151] sp|Q06543|GPN3_YEAST                                                                                                                                                                                                                                                   
## [152] sp|Q3E833|PCC1_YEAST                                                                                                                                                                                                                                                   
## [153] sp|P25039|EFGM_YEAST                                                                                                                                                                                                                                                   
## [154] sp|P81449|ATPJ_YEAST                                                                                                                                                                                                                                                   
## [155] sp|P53890|BNI5_YEAST                                                                                                                                                                                                                                                   
## [156] sp|P06100|NOT2_YEAST                                                                                                                                                                                                                                                   
## [157] sp|Q03020|ISU1_YEAST                                                                                                                                                                                                                                                   
## [158] sp|P25358|ELO2_YEAST                                                                                                                                                                                                                                                   
## [159] sp|P17558|RTPT_YEAST                                                                                                                                                                                                                                                   
## [160] sp|P38985|SRP14_YEAST                                                                                                                                                                                                                                                  
## [161] sp|P34909|NOT4_YEAST                                                                                                                                                                                                                                                   
## [162] sp|P13574|STE12_YEAST                                                                                                                                                                                                                                                  
## [163] sp|P22214|SEC22_YEAST                                                                                                                                                                                                                                                  
## [164] sp|Q08004|BUD20_YEAST                                                                                                                                                                                                                                                  
## [165] sp|P53173|ERV14_YEAST                                                                                                                                                                                                                                                  
## [166] sp|P40043|RGI1_YEAST                                                                                                                                                                                                                                                   
## [167] sp|Q08951|AP3D_YEAST                                                                                                                                                                                                                                                   
## [168] sp|P39938|RS26A_YEAST                                                                                                                                                                                                                                                  
## [169] P41159ups|LEP_HUMAN_UPS                                                                                                                                                                                                                                                
## [170] sp|P34248|YKK0_YEAST                                                                                                                                                                                                                                                   
## [171] sp|P46964|OST2_YEAST                                                                                                                                                                                                                                                   
## [172] sp|P32529|RPA12_YEAST                                                                                                                                                                                                                                                  
## [173] sp|P39013|END3_YEAST                                                                                                                                                                                                                                                   
## [174] sp|P46990|RL17B_YEAST                                                                                                                                                                                                                                                  
## [175] sp|P05740|RL17A_YEAST                                                                                                                                                                                                                                                  
## [176] sp|P53868|ALG9_YEAST                                                                                                                                                                                                                                                   
## [177] sp|P12630|BAR1_YEAST                                                                                                                                                                                                                                                   
## [178] sp|P56508|SNA2_YEAST                                                                                                                                                                                                                                                   
## 1722 Levels:  CON__P00761 ... sp|Q99385|VCX1_YEAST</code></pre>
<p>Both UPS proteins in this list are only identified by one peptide.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getData</span>(proteinsCPTAC[<span class="st">&quot;P69905ups|HBA_HUMAN_UPS&quot;</span>])</code></pre></div>
<pre><code>##    quant_value        Sequence   run condition lab
## 13    16.34396 VGAHAGEYGAEALER X6B_4        6B   2
## 20    18.98716 VGAHAGEYGAEALER X6C_2        6C   1
## 22    16.89253 VGAHAGEYGAEALER X6C_4        6C   2
## 23    18.10633 VGAHAGEYGAEALER X6C_5        6C   2
## 24    17.52955 VGAHAGEYGAEALER X6C_6        6C   2
## 31    18.97273 VGAHAGEYGAEALER X6D_4        6D   2
## 32    16.95212 VGAHAGEYGAEALER X6D_5        6D   2
## 33    19.29081 VGAHAGEYGAEALER X6D_6        6D   2
## 35    16.09004 VGAHAGEYGAEALER X6D_8        6D   3
## 37    17.82644 VGAHAGEYGAEALER X6E_1        6E   1
## 40    20.41722 VGAHAGEYGAEALER X6E_4        6E   2
## 41    17.83335 VGAHAGEYGAEALER X6E_5        6E   2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getData</span>(proteinsCPTAC[<span class="st">&quot;P41159ups|LEP_HUMAN_UPS&quot;</span>])</code></pre></div>
<pre><code>##    quant_value           Sequence   run condition lab
## 31    15.32290 VTGLDFIPGLHPILTLSK X6D_4        6D   2
## 32    15.85747 VTGLDFIPGLHPILTLSK X6D_5        6D   2
## 40    19.09555 VTGLDFIPGLHPILTLSK X6E_4        6E   2
## 41    20.80418 VTGLDFIPGLHPILTLSK X6E_5        6E   2
## 42    20.65861 VTGLDFIPGLHPILTLSK X6E_6        6E   2
## 43    16.17743 VTGLDFIPGLHPILTLSK X6E_7        6E   3
## 44    17.42877 VTGLDFIPGLHPILTLSK X6E_8        6E   3
## 45    18.18318 VTGLDFIPGLHPILTLSK X6E_9        6E   3</code></pre>
<p>Remove the overparameterized models from the protLM object <code>modelCPTAC_RR</code> and from the data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modelCPTAC_RR_cleaned &lt;-<span class="st"> </span>modelCPTAC_RR[-na_indices]
data &lt;-<span class="st"> </span>data[-na_indices]</code></pre></div>
</div>
<div id="statistical-inference-1" class="section level3">
<h3>5. Statistical inference</h3>
<p>Construct the contrast matrix L for inferring on the fold changes of interest.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">L &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="kw">c</span>(<span class="st">&quot;condition6B-condition6A&quot;</span>,<span class="st">&quot;condition6C-condition6A&quot;</span>,<span class="st">&quot;condition6D-condition6A&quot;</span>,<span class="st">&quot;condition6E-condition6A&quot;</span>,<span class="st">&quot;condition6C-condition6B&quot;</span>,<span class="st">&quot;condition6D-condition6B&quot;</span>,<span class="st">&quot;condition6E-condition6B&quot;</span>,<span class="st">&quot;condition6D-condition6C&quot;</span>,<span class="st">&quot;condition6E-condition6C&quot;</span>,<span class="st">&quot;condition6E-condition6D&quot;</span>),
                  <span class="kw">c</span>(<span class="st">&quot;condition6A&quot;</span>,<span class="st">&quot;condition6B&quot;</span>,<span class="st">&quot;condition6C&quot;</span>,<span class="st">&quot;condition6D&quot;</span>,<span class="st">&quot;condition6E&quot;</span>))</code></pre></div>
<p>Make the DA rankings for the ordinary least squares model (no ridge, no squeezed variances, no M estimation).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#modelCPTAC_Lm &lt;- squeezePars(modelCPTAC_Lm, squeezeVar=FALSE)</span>
lm_normal &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_Lm, L)
lm_normal &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(lm_normal)
lm_normal &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(lm_normal)</code></pre></div>
<p>Make the DA rankings for the ordinary least squares model with squeezed variances (no ridge, no M estimation).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modelCPTAC_Lm_Sq &lt;-<span class="st"> </span><span class="kw">squeezePars</span>(modelCPTAC_Lm, <span class="dt">squeezeVar=</span><span class="ot">TRUE</span>)
lm_Sq &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_Lm_Sq, L)
lm_Sq &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(lm_Sq)
lm_Sq &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(lm_Sq)</code></pre></div>
<p>Make the DA rankings for the ordinary least squares model with squeezed variances and M estimation (no ridge).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modelCPTAC_Lm_M_Sq &lt;-<span class="st"> </span><span class="kw">squeezePars</span>(modelCPTAC_Lm_M, <span class="dt">squeezeVar=</span><span class="ot">TRUE</span>)
lm_SqM &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_Lm_M_Sq, L)
lm_SqM &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(lm_SqM)
lm_SqM &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(lm_SqM)</code></pre></div>
<p>Make the DA rankings for the ridge regression model (no squeezed variances, no M estimation).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#modelCPTAC_Ridge &lt;- squeezePars(modelCPTAC_Ridge, squeezeVar=FALSE)</span>
ridge &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_Ridge, L)
ridge &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(ridge)
ridge &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(ridge)</code></pre></div>
<p>Make the DA rankings for the ridge regression model with squeezed variances (no M estimation).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modelCPTAC_Ridge_Sq &lt;-<span class="st"> </span><span class="kw">squeezePars</span>(modelCPTAC_Ridge, <span class="dt">squeezeVar=</span><span class="ot">TRUE</span>)
ridgeSq &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_Ridge_Sq, L)
ridgeSq &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(ridgeSq)
ridgeSq &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(ridgeSq)</code></pre></div>
<p>Make the DA rankings for the ridge regression model with squeezed variances and M estimation. This is our robust ridge (RR) approach (the default MSqRob approach, although we included “lab” here as a random effect and we did not include a “run” effect).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modelCPTAC_Ridge_M_Sq &lt;-<span class="st"> </span><span class="kw">squeezePars</span>(modelCPTAC_Ridge_M, <span class="dt">squeezeVar=</span><span class="ot">TRUE</span>)
ridgeSqM &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_Ridge_M, L)
ridgeSqM &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(ridgeSqM)
ridgeSqM &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(ridgeSqM)</code></pre></div>
<p>Our standard approach is to calculate the degrees of freedom via the trace of the Hat matrix. One could as well use custom degrees of freedom, for example in order to work more conservatively. Indeed, if we consider the peptides as pseudo-replicates, our experimental units are the different spike-in conditions in each lab. The number of degrees of freedom is then the number of combinations lab x condition minus the number of parameters estimated for lab (i.e. the number of labs minus 1) minus the number of parameters estimated for condition (i.e. the number of conditions minus 1) minus 1 for the intercept. We show an example for the <code>modelCPTAC_RR</code> model.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">custom_dfs &lt;-<span class="st"> </span><span class="kw">sapply</span>(data, function(x){<span class="kw">return</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">paste0</span>(x$condition,x$lab)))-(<span class="kw">length</span>(<span class="kw">unique</span>(x$condition))-<span class="dv">1</span>)-(<span class="kw">length</span>(<span class="kw">unique</span>(x$lab))-<span class="dv">1</span>)-<span class="dv">1</span>)})
custom_dfs</code></pre></div>
<pre><code>##    [1] 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8
##   [35] 1 8 8 8 8 8 3 7 8 8 8 8 7 8 5 8 8 8 8 4 8 8 8 8 8 8 8 8 8 8 7 8 8 8
##   [69] 8 8 8 8 8 4 8 8 8 8 8 8 8 4 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8
##  [103] 8 8 8 8 8 5 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 7 8 8 8 8 7 8 8 4 8 8 8 8
##  [137] 8 8 8 8 8 8 8 8 8 8 8 8 8 7 7 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 5 8 8 8
##  [171] 8 8 8 8 8 6 8 8 8 8 7 7 8 8 8 8 8 7 8 8 8 4 8 8 8 7 8 8 8 8 8 8 8 8
##  [205] 8 8 8 8 8 8 8 2 8 8 8 8 8 8 8 8 8 8 8 8 6 8 8 8 8 4 8 8 8 8 8 8 8 8
##  [239] 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 4 8 8 8 8 8 7 8 8 8 8 8 8 8 5 8
##  [273] 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 5 8 8 8 8 8 8 8 8 8 8 1 8 8 8 8 7 8
##  [307] 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 1 8 8 8 6 4 7 8 8 6 8 8 6 8 7 8 5 5
##  [341] 8 8 8 8 8 8 8 8 1 8 8 8 7 2 8 8 8 8 8 8 5 8 6 8 8 8 7 8 8 8 7 5 8 8
##  [375] 8 8 8 8 8 8 8 8 8 8 8 8 8 8 7 5 8 8 8 8 8 8 8 8 8 8 2 8 8 8 8 3 8 8
##  [409] 3 8 8 8 8 8 8 8 7 3 8 7 0 8 8 1 5 8 7 8 8 8 6 8 8 8 8 8 8 8 8 8 3 8
##  [443] 8 8 8 8 8 6 4 4 8 8 8 8 8 8 8 8 8 8 8 8 4 8 2 8 8 8 8 8 8 8 8 8 8 3
##  [477] 8 8 8 5 8 8 8 7 8 8 8 5 8 4 8 8 8 8 8 5 5 8 5 8 8 8 8 8 5 4 5 8 8 8
##  [511] 8 7 7 7 8 8 7 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 4 8 8 8 8 7 8 7 8 8
##  [545] 8 7 8 8 8 8 8 6 8 8 8 7 8 8 8 8 8 8 8 8 4 8 8 7 8 8 8 8 8 8 8 8 8 4
##  [579] 4 5 7 8 8 8 8 8 7 8 8 8 8 2 8 8 4 5 8 8 8 8 5 5 8 8 4 8 8 8 8 8 8 8
##  [613] 8 7 4 8 8 8 8 8 7 8 8 7 8 8 8 8 8 8 8 6 7 8 5 8 8 8 8 8 6 8 2 8 8 8
##  [647] 8 7 8 8 8 8 8 8 7 8 7 8 5 8 8 8 8 8 7 8 8 7 8 8 8 8 8 2 2 8 7 8 7 4
##  [681] 8 8 4 8 4 8 8 8 8 8 8 5 8 8 8 8 8 8 8 8 8 8 8 7 8 8 8 8 8 8 8 8 7 8
##  [715] 8 7 8 8 8 6 8 8 8 8 8 8 8 8 0 8 6 7 6 8 8 8 8 8 8 4 4 8 8 4 8 8 7 3
##  [749] 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 4 8 8 8 7 8 8 8 8 8 6 8
##  [783] 8 8 8 8 8 8 8 8 7 8 8 8 7 8 6 8 8 8 5 8 8 2 5 8 8 8 6 8 8 8 6 8 8 8
##  [817] 8 8 8 8 5 8 8 8 8 8 5 5 8 7 3 8 8 8 7 8 8 8 8 8 5 8 2 8 8 8 7 8 8 0
##  [851] 6 8 8 3 8 8 6 4 8 5 7 8 8 8 8 8 7 7 5 8 8 7 8 8 8 8 8 7 8 8 6 8 8 8
##  [885] 4 7 8 8 8 8 8 8 8 8 8 1 8 6 8 8 8 8 8 8 7 7 8 8 8 8 8 8 8 8 7 8 7 8
##  [919] 8 8 8 8 8 8 8 7 5 8 7 8 4 8 8 5 7 6 8 8 8 5 8 8 8 8 8 8 2 8 8 4 8 8
##  [953] 8 8 8 8 6 8 8 8 6 7 8 8 8 7 8 2 8 5 8 7 8 8 8 7 8 8 1 8 5 8 7 4 7 5
##  [987] 8 8 8 3 8 8 8 7 8 8 8 5 8 8 8 6 8 1 5 8 8 1 8 4 7 8 8 4 8 8 7 8 8 8
## [1021] 5 1 8 6 4 5 8 8 8 6 5 8 8 8 8 8 8 8 8 5 8 6 8 8 7 7 8 8 2 8 4 8 8 8
## [1055] 8 8 8 8 4 8 8 6 8 8 8 8 8 4 8 8 8 8 4 4 8 8 3 8 8 8 4 8 3 8 8 8 4 2
## [1089] 3 8 7 3 8 7 8 6 4 7 8 8 6 6 7 4 6 7 4 8 7 5 8 4 4 4 4 8 8 8 8 5 8 4
## [1123] 8 6 8 6 8 6 8 8 7 8 7 8 8 6 8 7 4 2 8 8 6 8 2 6 8 8 8 7 8 6 8 8 8 3
## [1157] 8 6 8 8 8 3 3 8 8 8 8 8 8 8 7 7 8 6 8 7 8 8 8 5 8 8 8 8 8 4 7 8 8 4
## [1191] 8 7 4 4 4 4 8 8 4 8 2 7 1 4 8 4 0 6 8 4 5 7 6 3 4 8 8 0 8 0 6 5 8 8
## [1225] 0 1 8 4 8 4 4 8 7 8 4 4 5 8 4 4 5 5 8 8 8 1 8 8 5 7 6 4 8 8 8 7 8 6
## [1259] 6 8 1 2 7 3 6 6 6 8 7 8 8 6 8 3 4 8 8 5 7 7 8 8 2 8 6 4 8 7 7 7 4 4
## [1293] 5 4 5 4 5 1 6 8 5 3 5 8 8 5 6 8 3 8 7 3 8 8 8 8 3 3 8 8 8 1 8 8 8 5
## [1327] 8 5 7 8 4 8 4 6 7 3 8</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ridgeSqM_custom &lt;-<span class="st"> </span><span class="kw">test.protLMcontrast</span>(modelCPTAC_RR, L, <span class="dt">custom_dfs =</span> custom_dfs)</code></pre></div>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.right, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<pre><code>## Warning in pt(tstat.left, df = df, lower.tail = FALSE): NaNs produced</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ridgeSqM_custom &lt;-<span class="st"> </span><span class="kw">prot.p.adjust</span>(ridgeSqM_custom)
ridgeSqM_custom &lt;-<span class="st"> </span><span class="kw">prot.signif</span>(ridgeSqM_custom)</code></pre></div>
<p>Extra: add a column containing 1 if the corresponding protein is a spiked-in UPS1 protein and 0 if it is a yeast protein.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lm_normal &lt;-<span class="st"> </span><span class="kw">lapply</span>(lm_normal, function(x){
  x &lt;-<span class="st"> </span><span class="kw">cbind</span>(x,<span class="st">&quot;ups&quot;</span>=<span class="kw">grepl</span>(<span class="st">&quot;UPS&quot;</span>,<span class="kw">rownames</span>(x)))
  <span class="kw">return</span>(x)
  })

lm_Sq &lt;-<span class="st"> </span><span class="kw">lapply</span>(lm_Sq, function(x){
  x &lt;-<span class="st"> </span><span class="kw">cbind</span>(x,<span class="st">&quot;ups&quot;</span>=<span class="kw">grepl</span>(<span class="st">&quot;UPS&quot;</span>,<span class="kw">rownames</span>(x)))
  <span class="kw">return</span>(x)
})

lm_SqM &lt;-<span class="st"> </span><span class="kw">lapply</span>(lm_SqM, function(x){
  x &lt;-<span class="st"> </span><span class="kw">cbind</span>(x,<span class="st">&quot;ups&quot;</span>=<span class="kw">grepl</span>(<span class="st">&quot;UPS&quot;</span>,<span class="kw">rownames</span>(x)))
  <span class="kw">return</span>(x)
})

ridge &lt;-<span class="st"> </span><span class="kw">lapply</span>(ridge, function(x){
  x &lt;-<span class="st"> </span><span class="kw">cbind</span>(x,<span class="st">&quot;ups&quot;</span>=<span class="kw">grepl</span>(<span class="st">&quot;UPS&quot;</span>,<span class="kw">rownames</span>(x)))
  <span class="kw">return</span>(x)
})

ridgeSq &lt;-<span class="st"> </span><span class="kw">lapply</span>(ridgeSq, function(x){
  x &lt;-<span class="st"> </span><span class="kw">cbind</span>(x,<span class="st">&quot;ups&quot;</span>=<span class="kw">grepl</span>(<span class="st">&quot;UPS&quot;</span>,<span class="kw">rownames</span>(x)))
  <span class="kw">return</span>(x)
})

ridgeSqM &lt;-<span class="st"> </span><span class="kw">lapply</span>(ridgeSqM, function(x){
  x &lt;-<span class="st"> </span><span class="kw">cbind</span>(x,<span class="st">&quot;ups&quot;</span>=<span class="kw">grepl</span>(<span class="st">&quot;UPS&quot;</span>,<span class="kw">rownames</span>(x)))
  <span class="kw">return</span>(x)
})</code></pre></div>
</div>
<div id="create-the-summary-tables-summary_lm-and-the-like" class="section level3">
<h3>6. Create the summary tables “summary_lm” and the like</h3>
<p>Custom function for evaluating the performance of all methods on the CPTAC dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">makeSummary &lt;-<span class="st"> </span>function(CPTACobject, upratio){

  summary_object &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">bias0=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">bias1=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">sd0=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">sd1=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">mad0=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">mad1=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">RMSE0=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">RMSE1=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">TP5proc=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">FP5proc=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">TN5proc=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>),<span class="dt">FN5proc=</span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">10</span>))

  for(i in <span class="dv">1</span>:<span class="dv">10</span>)
  {

    significante &lt;-<span class="st"> </span><span class="kw">subset</span>(CPTACobject[[i]], CPTACobject[[i]][,<span class="st">&quot;qval&quot;</span>]&lt;=<span class="fl">0.05</span>)
    nonsignificante &lt;-<span class="st"> </span><span class="kw">subset</span>(CPTACobject[[i]], CPTACobject[[i]][,<span class="st">&quot;qval&quot;</span>]&gt;<span class="fl">0.05</span>)

    waarden3 &lt;-<span class="st"> </span><span class="kw">na.omit</span>(CPTACobject[[i]])

    <span class="co">#RMSE</span>
    summary_object$RMSE0[i] &lt;-<span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">mean</span>(((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">0</span>)[,<span class="st">&quot;estimate&quot;</span>])-<span class="dv">0</span>)^<span class="dv">2</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))
    <span class="co">#bias:</span>
    summary_object$bias0[i] &lt;-<span class="st"> </span><span class="kw">mean</span>((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">0</span>)[,<span class="st">&quot;estimate&quot;</span>])-<span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)
    <span class="co">#standard deviation:</span>
    summary_object$sd0[i] &lt;-<span class="st"> </span><span class="kw">sd</span>((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">0</span>)[,<span class="st">&quot;estimate&quot;</span>]), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)
    <span class="co">#mad:</span>
    summary_object$mad0[i] &lt;-<span class="st"> </span><span class="kw">mad</span>((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">0</span>)[,<span class="st">&quot;estimate&quot;</span>]), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)

    <span class="co">#RMSE</span>
    summary_object$RMSE1[i] &lt;-<span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">mean</span>(((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">1</span>)[,<span class="st">&quot;estimate&quot;</span>])-upratio[i])^<span class="dv">2</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))
    <span class="co">#bias:</span>
    summary_object$bias1[i] &lt;-<span class="st"> </span><span class="kw">mean</span>((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">1</span>)[,<span class="st">&quot;estimate&quot;</span>])-upratio[i], <span class="dt">na.rm=</span><span class="ot">TRUE</span>)
    <span class="co">#standard deviation:</span>
    summary_object$sd1[i] &lt;-<span class="st"> </span><span class="kw">sd</span>((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">1</span>)[,<span class="st">&quot;estimate&quot;</span>]), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)
    <span class="co">#standard deviation:</span>
    summary_object$mad1[i] &lt;-<span class="st"> </span><span class="kw">mad</span>((<span class="kw">subset</span>(waarden3, waarden3[,<span class="st">&quot;ups&quot;</span>]==<span class="dv">1</span>)[,<span class="st">&quot;estimate&quot;</span>]), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)

    <span class="co">#TP:</span>
    summary_object$TP5proc[i] &lt;-<span class="st"> </span><span class="kw">sum</span>(significante[,<span class="st">&quot;ups&quot;</span>])
    <span class="co">#FP:</span>
    summary_object$FP5proc[i] &lt;-<span class="st"> </span><span class="kw">length</span>(significante[,<span class="st">&quot;ups&quot;</span>])-<span class="kw">sum</span>(significante[,<span class="st">&quot;ups&quot;</span>])
    <span class="co">#FN:</span>
    summary_object$FN5proc[i] &lt;-<span class="st"> </span><span class="kw">sum</span>(nonsignificante[,<span class="st">&quot;ups&quot;</span>])
    <span class="co">#TN:</span>
    summary_object$TN5proc[i] &lt;-<span class="st"> </span><span class="kw">length</span>(nonsignificante[,<span class="st">&quot;ups&quot;</span>])-<span class="kw">sum</span>(nonsignificante[,<span class="st">&quot;ups&quot;</span>])

  }

  <span class="kw">rownames</span>(summary_object) &lt;-<span class="st"> </span><span class="kw">names</span>(CPTACobject)
  
  <span class="kw">return</span>(summary_object)
}</code></pre></div>
<p>Create a vector that contains the theoretical log2 fold changes of the spiked-in UPS1 proteins.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">upratio &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">1.565597</span>,<span class="fl">3.137504</span>,<span class="fl">4.744161</span>,<span class="fl">6.321928</span>,<span class="fl">1.571906</span>,
             <span class="fl">3.178564</span>,<span class="fl">4.756331</span>,<span class="fl">1.60665</span>,<span class="fl">3.18442</span>,<span class="fl">1.577767</span>)</code></pre></div>
<p>Create the summary objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_lm &lt;-<span class="st"> </span><span class="kw">makeSummary</span>(lm_normal, upratio)
summary_lmSq &lt;-<span class="st"> </span><span class="kw">makeSummary</span>(lm_Sq, upratio)
summary_lmSqM &lt;-<span class="st"> </span><span class="kw">makeSummary</span>(lm_SqM, upratio)
summary_ridge &lt;-<span class="st"> </span><span class="kw">makeSummary</span>(ridge, upratio)
summary_ridgeSq &lt;-<span class="st"> </span><span class="kw">makeSummary</span>(ridgeSq, upratio)
summary_ridgeSqM &lt;-<span class="st"> </span><span class="kw">makeSummary</span>(ridgeSqM, upratio)</code></pre></div>
<p>Results</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_lm</code></pre></div>
<pre><code>##                                 bias0       bias1       sd0       sd1
## condition6B-condition6A -0.0490650006 -0.26374480 0.1889214 0.5458871
## condition6C-condition6A -0.0636028162 -0.35999656 0.1986675 0.6987931
## condition6D-condition6A -0.0504891063 -0.33193686 0.2635353 0.8405903
## condition6E-condition6A -0.1361429644 -0.46343147 0.3454989 0.8998965
## condition6C-condition6B -0.0120151063 -0.02433229 0.1617376 0.3461417
## condition6D-condition6B -0.0009803456  0.01369797 0.2927206 0.4690942
## condition6E-condition6B -0.0886586784 -0.09994053 0.3895888 0.6527204
## condition6D-condition6C  0.0133347292  0.03967844 0.2826181 0.2338392
## condition6E-condition6C -0.0719998523 -0.05900461 0.3895968 0.4054120
## condition6E-condition6D -0.0864572044 -0.09868005 0.2562558 0.2862106
##                              mad0      mad1     RMSE0     RMSE1 TP5proc
## condition6B-condition6A 0.1314349 0.3276234 0.1951180 0.5999277      28
## condition6C-condition6A 0.1440518 0.4102281 0.2085272 0.7782683      38
## condition6D-condition6A 0.1825928 0.5441178 0.2682280 0.8939292      39
## condition6E-condition6A 0.2320079 0.4309940 0.3712302 1.0021662      40
## condition6C-condition6B 0.1136509 0.2651505 0.1621208 0.3429573      39
## condition6D-condition6B 0.2182888 0.2758328 0.2926088 0.4638099      42
## condition6E-condition6B 0.2663954 0.3490388 0.3994020 0.6527817      43
## condition6D-condition6C 0.2104883 0.1883573 0.2828232 0.2345473      42
## condition6E-condition6C 0.2694255 0.2595494 0.3960455 0.4050987      44
## condition6E-condition6D 0.1647321 0.1827531 0.2703534 0.2996540      43
##                         FP5proc TN5proc FN5proc
## condition6B-condition6A       1    1290      11
## condition6C-condition6A       4    1289       2
## condition6D-condition6A      34    1258       1
## condition6E-condition6A     268    1022       0
## condition6C-condition6B       0    1291       4
## condition6D-condition6B      47    1243       1
## condition6E-condition6B     272    1016       0
## condition6D-condition6C      39    1253       2
## condition6E-condition6C     234    1056       0
## condition6E-condition6D      27    1263       1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_lmSq</code></pre></div>
<pre><code>##                                 bias0       bias1       sd0       sd1
## condition6B-condition6A -0.0490650006 -0.26374480 0.1889214 0.5458871
## condition6C-condition6A -0.0636028162 -0.35999656 0.1986675 0.6987931
## condition6D-condition6A -0.0504891063 -0.33193686 0.2635353 0.8405903
## condition6E-condition6A -0.1361429644 -0.46343147 0.3454989 0.8998965
## condition6C-condition6B -0.0120151063 -0.02433229 0.1617376 0.3461417
## condition6D-condition6B -0.0009803456  0.01369797 0.2927206 0.4690942
## condition6E-condition6B -0.0886586784 -0.09994053 0.3895888 0.6527204
## condition6D-condition6C  0.0133347292  0.03967844 0.2826181 0.2338392
## condition6E-condition6C -0.0719998523 -0.05900461 0.3895968 0.4054120
## condition6E-condition6D -0.0864572044 -0.09868005 0.2562558 0.2862106
##                              mad0      mad1     RMSE0     RMSE1 TP5proc
## condition6B-condition6A 0.1314349 0.3276234 0.1951180 0.5999277      28
## condition6C-condition6A 0.1440518 0.4102281 0.2085272 0.7782683      38
## condition6D-condition6A 0.1825928 0.5441178 0.2682280 0.8939292      39
## condition6E-condition6A 0.2320079 0.4309940 0.3712302 1.0021662      40
## condition6C-condition6B 0.1136509 0.2651505 0.1621208 0.3429573      39
## condition6D-condition6B 0.2182888 0.2758328 0.2926088 0.4638099      42
## condition6E-condition6B 0.2663954 0.3490388 0.3994020 0.6527817      43
## condition6D-condition6C 0.2104883 0.1883573 0.2828232 0.2345473      43
## condition6E-condition6C 0.2694255 0.2595494 0.3960455 0.4050987      44
## condition6E-condition6D 0.1647321 0.1827531 0.2703534 0.2996540      43
##                         FP5proc TN5proc FN5proc
## condition6B-condition6A       1    1290      11
## condition6C-condition6A       3    1290       2
## condition6D-condition6A      35    1257       1
## condition6E-condition6A     256    1034       0
## condition6C-condition6B       0    1291       4
## condition6D-condition6B      46    1244       1
## condition6E-condition6B     267    1021       0
## condition6D-condition6C      40    1252       1
## condition6E-condition6C     228    1062       0
## condition6E-condition6D      24    1266       1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_lmSqM</code></pre></div>
<pre><code>##                                bias0       bias1       sd0       sd1
## condition6B-condition6A -0.049298411 -0.23349295 0.1777832 0.5740949
## condition6C-condition6A -0.069454077 -0.27481569 0.1861274 0.7150222
## condition6D-condition6A -0.064727904 -0.22611893 0.2446658 0.8407585
## condition6E-condition6A -0.163773694 -0.34271420 0.3306083 0.9031051
## condition6C-condition6B -0.017537011  0.02945471 0.1475790 0.3112478
## condition6D-condition6B -0.014967156  0.08500111 0.2764503 0.4310143
## condition6E-condition6B -0.115997223 -0.01288364 0.3793133 0.6061190
## condition6D-condition6C  0.004958186  0.05725495 0.2690348 0.2177633
## condition6E-condition6C -0.093789526 -0.02674425 0.3811994 0.3826285
## condition6E-condition6D -0.099860230 -0.08399620 0.2460404 0.2694353
##                              mad0      mad1     RMSE0     RMSE1 TP5proc
## condition6B-condition6A 0.1218369 0.3082833 0.1844254 0.6129054      30
## condition6C-condition6A 0.1267718 0.4690643 0.1985963 0.7576272      38
## condition6D-condition6A 0.1587477 0.5907056 0.2529916 0.8604259      39
## condition6E-condition6A 0.2123974 0.4680543 0.3688346 0.9553334      40
## condition6C-condition6B 0.1027001 0.1830817 0.1485605 0.3090143      41
## condition6D-condition6B 0.2002622 0.2561993 0.2767482 0.4343710      43
## condition6E-condition6B 0.2482147 0.2973742 0.3965126 0.5991682      43
## condition6D-condition6C 0.1952049 0.2198974 0.2689764 0.2227582      43
## condition6E-condition6C 0.2490746 0.2174304 0.3924242 0.3791997      44
## condition6E-condition6D 0.1536106 0.1796506 0.2654450 0.2792864      43
##                         FP5proc TN5proc FN5proc
## condition6B-condition6A       2    1289       9
## condition6C-condition6A       7    1286       2
## condition6D-condition6A      52    1240       1
## condition6E-condition6A     398     892       0
## condition6C-condition6B       1    1290       2
## condition6D-condition6B      60    1230       0
## condition6E-condition6B     351     937       0
## condition6D-condition6C      42    1250       1
## condition6E-condition6C     324     966       0
## condition6E-condition6D      69    1221       1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_ridge</code></pre></div>
<pre><code>##                                bias0       bias1       sd0       sd1
## condition6B-condition6A -0.022821204 -0.28651011 0.1285001 0.5510196
## condition6C-condition6A -0.031517517 -0.40657348 0.1363416 0.7167196
## condition6D-condition6A -0.025406053 -0.40840787 0.1898219 0.8732607
## condition6E-condition6A -0.058529497 -0.56970162 0.2660547 0.9688576
## condition6C-condition6B -0.007309999 -0.08452214 0.1225707 0.4156455
## condition6D-condition6B -0.004071671 -0.11608481 0.2272722 0.6692130
## condition6E-condition6B -0.039083209 -0.29294584 0.3178031 0.9618865
## condition6D-condition6C  0.003774038 -0.03025730 0.2192206 0.3371939
## condition6E-condition6C -0.030789519 -0.19193586 0.3110181 0.6221030
## condition6E-condition6D -0.033137803 -0.09782654 0.1959141 0.5637166
##                               mad0      mad1     RMSE0     RMSE1 TP5proc
## condition6B-condition6A 0.02986928 0.2967893 0.1304674 0.6147564      28
## condition6C-condition6A 0.04068911 0.4511841 0.1398914 0.8161782      38
## condition6D-condition6A 0.05174847 0.5286015 0.1914499 0.9541051      39
## condition6E-condition6A 0.09720540 0.4315930 0.2723269 1.1134531      40
## condition6C-condition6B 0.02228200 0.3069350 0.1227465 0.4194982      37
## condition6D-condition6B 0.05551943 0.2833159 0.2272306 0.6716721      42
## condition6E-condition6B 0.09570841 0.3440744 0.3200883 0.9949950      43
## condition6D-condition6C 0.04909078 0.1920756 0.2191778 0.3347963      42
## condition6E-condition6C 0.09226521 0.2562150 0.3124314 0.6443999      43
## condition6E-condition6D 0.05357493 0.1897926 0.1986304 0.5660727      42
##                         FP5proc TN5proc FN5proc
## condition6B-condition6A       0    1456      11
## condition6C-condition6A       1    1454       2
## condition6D-condition6A      27    1429       1
## condition6E-condition6A     178    1271       0
## condition6C-condition6B       0    1456       7
## condition6D-condition6B      31    1425       2
## condition6E-condition6B     190    1257       1
## condition6D-condition6C      28    1428       3
## condition6E-condition6C     148    1299       2
## condition6E-condition6D      17    1437       4</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_ridgeSq</code></pre></div>
<pre><code>##                                bias0       bias1       sd0       sd1
## condition6B-condition6A -0.022821204 -0.28651011 0.1285001 0.5510196
## condition6C-condition6A -0.031517517 -0.40657348 0.1363416 0.7167196
## condition6D-condition6A -0.025406053 -0.40840787 0.1898219 0.8732607
## condition6E-condition6A -0.058529497 -0.56970162 0.2660547 0.9688576
## condition6C-condition6B -0.007309999 -0.08452214 0.1225707 0.4156455
## condition6D-condition6B -0.004071671 -0.11608481 0.2272722 0.6692130
## condition6E-condition6B -0.039083209 -0.29294584 0.3178031 0.9618865
## condition6D-condition6C  0.003774038 -0.03025730 0.2192206 0.3371939
## condition6E-condition6C -0.030789519 -0.19193586 0.3110181 0.6221030
## condition6E-condition6D -0.033137803 -0.09782654 0.1959141 0.5637166
##                               mad0      mad1     RMSE0     RMSE1 TP5proc
## condition6B-condition6A 0.02986928 0.2967893 0.1304674 0.6147564      28
## condition6C-condition6A 0.04068911 0.4511841 0.1398914 0.8161782      38
## condition6D-condition6A 0.05174847 0.5286015 0.1914499 0.9541051      39
## condition6E-condition6A 0.09720540 0.4315930 0.2723269 1.1134531      40
## condition6C-condition6B 0.02228200 0.3069350 0.1227465 0.4194982      38
## condition6D-condition6B 0.05551943 0.2833159 0.2272306 0.6716721      42
## condition6E-condition6B 0.09570841 0.3440744 0.3200883 0.9949950      43
## condition6D-condition6C 0.04909078 0.1920756 0.2191778 0.3347963      42
## condition6E-condition6C 0.09226521 0.2562150 0.3124314 0.6443999      43
## condition6E-condition6D 0.05357493 0.1897926 0.1986304 0.5660727      43
##                         FP5proc TN5proc FN5proc
## condition6B-condition6A       0    1456      11
## condition6C-condition6A       1    1454       2
## condition6D-condition6A      28    1428       1
## condition6E-condition6A     175    1274       0
## condition6C-condition6B       0    1456       6
## condition6D-condition6B      30    1426       2
## condition6E-condition6B     182    1265       1
## condition6D-condition6C      27    1429       3
## condition6E-condition6C     152    1295       2
## condition6E-condition6D      16    1438       3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary_ridgeSqM</code></pre></div>
<pre><code>##                                bias0         bias1       sd0       sd1
## condition6B-condition6A -0.020206485 -0.2246285174 0.1256596 0.5797967
## condition6C-condition6A -0.034153373 -0.2610975530 0.1322110 0.7537752
## condition6D-condition6A -0.038235593 -0.2252771856 0.1758739 0.8831463
## condition6E-condition6A -0.087143371 -0.3585885100 0.2564536 0.9837767
## condition6C-condition6B -0.012587203 -0.0013144994 0.1159392 0.3935278
## condition6D-condition6B -0.019522557 -0.0038337706 0.2152418 0.6593331
## condition6E-condition6B -0.070251938 -0.1505534392 0.3124419 0.9479481
## condition6D-condition6C -0.006402685 -0.0003491773 0.2091651 0.3250145
## condition6E-condition6C -0.056795099 -0.1338177618 0.3041673 0.6081468
## condition6E-condition6D -0.048902893 -0.0702298204 0.1930072 0.5508130
##                               mad0      mad1     RMSE0     RMSE1 TP5proc
## condition6B-condition6A 0.02646967 0.2649400 0.1272312 0.6148192      31
## condition6C-condition6A 0.03972730 0.4477229 0.1365071 0.7887614      38
## condition6D-condition6A 0.05294331 0.5038811 0.1799232 0.9006656      39
## condition6E-condition6A 0.10167133 0.3793773 0.2707712 1.0354742      39
## condition6C-condition6B 0.02066450 0.2048808 0.1165808 0.3890324      41
## condition6D-condition6B 0.05313198 0.2414776 0.2160517 0.6518089      42
## condition6E-condition6B 0.10050353 0.2836452 0.3201372 0.9491307      43
## condition6D-condition6C 0.04896884 0.2042296 0.2091913 0.3213832      43
## condition6E-condition6C 0.09947162 0.1527536 0.3093210 0.6160609      44
## condition6E-condition6D 0.05419340 0.1771095 0.1990419 0.5493011      44
##                         FP5proc TN5proc FN5proc
## condition6B-condition6A       2    1454       8
## condition6C-condition6A       4    1451       2
## condition6D-condition6A      44    1412       1
## condition6E-condition6A     326    1123       1
## condition6C-condition6B       1    1455       3
## condition6D-condition6B      48    1408       2
## condition6E-condition6B     319    1128       1
## condition6D-condition6C      38    1418       2
## condition6E-condition6C     268    1179       1
## condition6E-condition6D      61    1393       2</code></pre>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Goeminne, L. J. E., Argentini, A., Martens, L., and Clement, L. (2015) Summarization vs Peptide-Based Models in Label-Free Quantitative Proteomics: Performance, Pitfalls, and Data Analysis Guidelines. Journal of Proteome Research 14(6), pp 2457-2465.</p>
<p>Goeminne, L. J. E., Gevaert, K., and Clement, L. (2016) Peptide-level Robust Ridge Regression Improves Estimation, Sensitivity, and Specificity in Data-dependent Quantitative Label-free Shotgun Proteomics. Molecular &amp; Cellular Proteomics 15(2), pp 657-668.</p>
<p>Bolstad, B. M., Irizarry, R. A., Astrand, M., and Speed, T. P. (2003) A Comparison of Normalization Methods for High Density Oligonucleotide Array Data Based on Bias and Variance. Bioinformatics 19(2), pp 185-193. <a href="http://bmbolstad.com/misc/normalize/normalize.html" class="uri">http://bmbolstad.com/misc/normalize/normalize.html</a></p>
<p>Navarro P., Kuharev J., Gillet, L. C., Bernhardt, O. M., MacLean, B., Röst, H. L., Tate, S. A., Tsou, C., Reiter, L., Distler, U., Rosenberger, G., Perez-Riverol, Y., Nesvizhskii, A. I., Aebersold, R., and Tenzer, S. (2016) A multicenter study benchmarks software tools for label-free proteome quantification. Nature Biotechnology.</p>
<p>Ramond, E., Gesbert, G., Guerrera, I. C., Chhuon, C., Dupuis, M., Rigard, M., Henry, T., Barel, M., and Charbit, A. (2015) Importance of host cell arginine uptake in Francisella phagosomal escape and ribosomal protein amounts. Molecular &amp; Cellular Proteomics 14, pp 870–881.</p>
<p>Paulovich AG, Billheimer D, Ham A-JL, et al. (2010) Interlaboratory Study Characterizing a Yeast Performance Standard for Benchmarking LC-MS Platform Performance. Molecular &amp; Cellular Proteomic 9(2), pp 242-254.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
